---
title: "Towards an Ecological Trait-data Standard"
author:
- name: Florian D. Schneider \footnotemark[1] \footnotemark[2]
- name: Malte Jochum \footnotemark[4]
- name: Gaetane LeProvost \footnotemark[2]
- name: Andreas Ostrowski \footnotemark[5]
- name: Caterina Penone \footnotemark[4]
- name: Nadja K. Simons \footnotemark[6]
csl: amnat.csl
output:
  word_document:
    fig_caption: yes
    fig_width: 6
  pdf_document:
    fig_caption: yes
    fig_crop: no
    keep_tex: yes
    template: pretty.template
  html_document:
    fig_caption: yes
    fig_width: 4
    highlight: null
    keep_md: yes
    theme: united
bibliography: div05.bib
---

\renewcommand{\thefootnote}{\fnsymbol{footnote}}

\footnotetext[1]{Corresponding author: florian.dirk.schneider@gmail.com}
\footnotetext[2]{Senckenberg Biodiversity and Climate Research Centre (BiK-F), Senckenberganlage 25, 60325 Frankfurt am Main, Germany}
\footnotetext[4]{Institute of Plant Sciences, University of Bern, Bern, Switzerland}
\footnotetext[5]{Department of Mathematics and Computer Science, Friedrich-Schiller-Universität Jena, Jena, Germany}
\footnotetext[6]{Terrestrial Ecology Research Group, Department for Ecology and Ecosystem Management, Center for Life and Food Sciences Weihenstephan, Technische Universität München, Freising, Germany}

\linenumbers

# Abstract

1.	The assessment and analysis of functional traits has been established in all domains of ecological research, promising great advancements for biodiversity and climate research. A wealth of data is available from multiple sources and for many organism groups but is lacking standardisation due to heterogeneous data formats and data types. 
2.	To harmonize and standardise datasets, we present a general vocabulary and structure for an ecological trait-data standard for storing and sharing functional trait data.
3.	We further provide an R-package that assists in transferring own data structures into the data standard and to harmonize and merge trait datasets from heterogeneous sources. 
4.	With these decentralised tools for traitdata management, we intend to facilitate the exchange and analysis of trait data for ecological research. 

# Key-words: 

functional ecology, traits, semantic web, ontologies, data standard

# Introduction

Functional traits are phenotypic characteristics that are related to the fitness and performance of an organism [@mcgill06d; @violle07]. The scope of trait-based research spans from the evolutionary basis of individual-level properties to global patterns of biodiversity and ecosystem functioning. 

Comparative studies of traits inform researchers about physiological and evolutionary trade-offs, and help identifying economic strategies of organisms beyond taxonomy [@menezes10; @forister15;  @siefert15; @gross09; @wright04]. Inferring an unknown functional role or ecological function of individuals from their apparent features [@duarte11; @schrodt15] is a promising venue to bypass taxonomic impediment, i.e. the fact that a majority of species are yet undescribed and little is known of their interactions with the environment and other organisms. 
Mutual 'trait-matching' of co-evolved organisms enables the inference of ecological interactions from observed features of live specimens [@leprovost17; @kalinkat13; @dehling14a] which may be used to inform ecological networks [@weiner14; @miranda13; @digel11; @schleuning15] and parametrise ecological population-dynamic models [@schneider12; @kalinkat13]. These analyses are also a promising venue for a multitrophic community ecology aiming to unravel dependencies and synchrony in trait spaces between interacting functional groups [@soliveres16]. 
Since traits are so closely related to ecological mechanisms, they have proven useful for scaling individual level physiology and behaviour to functional diversity  of communities [@laliberte10a; @petchey09; @villeger08a; @hillerislambers12;@diaz98; @lavorel02], processes at the ecosystem scale [@fontaine05; @jennings09; @simons16; @gossner16; @moretti13; @gamez-virues15; @kuhsel15;  @birkhofer14; @gravel16; @lajoie15], as well as macro-patterns of biodiversity at the global level [@diaz13; @diaz16; @reich99; @reich97]. Traits can be used to predict extinction risk [@fritz10; @pearson14] and susceptibility to environmental toxins [@rubach11].
Most importantly, the trait framework relates key ecosystem services for human well-being back to changes in biodiversity in consequence to global change or local anthropogenic land use [@wood15; @bello10; @diaz07; @lavorel12; @allan15; @diaz11; @jennings09; @gossner16; @mangels17; @simons16; @hopfenmuller14; @mouillot13;  @vandewalle10; @perovic15; @borschig13; @birkhofer15; @birkhofer17]. 

In recent years, functional ecology has thus grown into a major field of ecological research, and that continues to integrate more and more fundamental and applied research questions. 
Given the expected future challenges in dealing with climate change and managing land use,  global trait synthesis is an interdisciplinary, cross-topic field with great potential. 
However, it relies heavily on the broad availability and coverage of factual information and measurement data on all biotic taxa. 
To mobilize data for this enterprise, information must be made available digitally,  harmonized according to data standards and hosted in interlinked online resources. The development of semantic-web standards [@wieczorek12; @page08] and ontologies [@walls12; @garnier17], as well as web-services [@kattge11; @parr16] for trait-data management is under way. 
Nonetheless, individual researchers often neglect the benefit of centralised databases and, to cope with journals' and research funding agencies increasing demand for the publication of data without access restrictions [@swan12; @allison15], instead publish primary data on file hosting services, where no data standards are forced upon the uploaded material (e.g. Figshare, Dryad). Given this prospect, there will be no lack of data, but a lack of data standardisation. 

In this paper, we (1.) discuss the importance of data standards for trait-based research and identify current deficits in standardisation from an pragmatic view of data providers and data users. We further (2.) review existing trait databases and web services, as well as initiatives for standardisation. 
We (3.) describe the common practice for structuring trait data for ecological research. 
Based on these considerations, we (4.) propose a low-threshold structure and vocabulary for describing trait-datasets compatible with existing ontologies. (5.) We provide a tool for an easy adoption of this structure and for the harmonisation of data from different sources. 

With an easy-to-use terminology and toolset, we hope to inform trait-data providers and trait-data users about the general importance of data standardisation and sketch a roadmap towards simple and low-threshold  trait data standardisation.

# 1. Need for data standardisation

Despite the obvious benefit and grand prospects of functional approaches, an integration of several studies across taxa, trophic levels, or organisational scales remains difficult because of different organismal, methodological, and functional focus, which renders data on traits extremely heterogeneous. 

Trait data have therefore been standardised into centralised databases for specific organism groups and research questions. These compiled data are either being published as static data tables or in the form of sophisticated web-services (see below for a review of current initiatives). Services attract data submissions from a focal community of researchers and map heterogeneous data into a common scheme, but more importantly, provide platforms for consensus building in methods. Furthermore, database services offer access control and data usage policies, to claim the rights of the original data providers, while simplifying data queries for synthesis researchers. 

Nonetheless, open-access publications in low-threshold data repositories such as Figshare.com, Dryad (datadryad.org), Researchgate.net or Zenodo.org are gaining importance and foster a decentralised data hosting with low expectations regarding data standardisation and documentation. Although open for manual download, data on low-threshold data platforms might be stored in proprietary (e.g. .xlsx) or binary (e.g. .pdf) data formats which make a programmatical extraction tedious or impossible.

On the other hand, not all centralised databases for functional traits comply with the criteria of open access, which leaves them at a disadvantage compared to low-threshold repositories. 
Open access to science data from publicly funded research is a declared goal of international and national research organisations [@emerson15; @allianceofgermanscienceorganisations10; @theroyalsocietysciencepolicycentre12] and increasingly required by journals and funding agencies to warrant reproducibility of science and foster public benefit. 
Also, factual data such as trait measurements, being not a product of a creative act, do not qualify as intellectual property and copyright does not apply [@carroll15; @creativecommons].
However, there is still no broad consensus achieved in the scientific community whether data from public funding must be generally dedicated to the public domain, or whether the investment in production must be acknowledged by attribution or co-authorship. 
Low-threshold data repositories assume public domain and Creative Commons Attribution licenses compliant with the demands of journals and research funding. 

Given this prospect, there will be no lack of data, but a lack of data standardisation hampering efficient future use of published datasets. 
To cope with this trend towards distributed data sources, data brokering services (e.g. the German Federation on Biological Data; gfbio.org)  or database management systems for scientific projects (e.g. BExIS, XX ) are likely to gain importance. These services simplify and direct the standardized upload of research data and descriptive metadata into reliable and interlinked data infrastructures. One goal of such initiatives is to facilitate data publications and standardization for researchers, for instance by providing terminologies and ontologies for biodiversity data, and by consulting on publication licenses.

After all, to mobilise the broad wealth of data into structured databases, it requires a certain minimum of information from the data providers. Getting accurate and well-defined input requires improving literacy of ecological researchers on data standards and ontologies.  

# 2. Overview of trait-data initiatives

In this section, we list two kinds of initiatives: the first aims at the harmonization of trait data into trait databases; the second develops trait ontologies and data standards. Often ontologies are being developed to serve a particular database project and thus are being developed in concert, as in the case of the TRY plant database [@kattge11] and the TOP Thesaurus of plant traits [@garnier17]. 

## Databases and Datasets 

Plant flower, leaf and stem traits have been measured for decades, and were compiled into comprehensive databases and published as static datasets, e.g. the LEDA traitbase for the Northwest-European flora [@kleyer08]. 
Regarding webservices, most prominently, plant traits from many different sources have been collated into the extensive TRY database [@kattge11]. Other more specialised databases for plant traits exist, for instance flower traits [BiolFlor for German wild flowers, @klotz02] and root measurements [Fine-root Ecology Database; @iversen17]. Traits on crop plants have been collected in agronomy databases like Planteome and BetyDB [https://www.betydb.org/; @lebauer17]. Trait-data for lower plants are less completely assessed. 

Microbial ecology also makes great use of trait-based approaches to assess genomic function and describe functional diversity at the community level [@krause14a; @fierer14;  @fierer12] derived from metagenomic analysis [@torsvik02; @langille13]. Databases are at hand to interpret gene sequences, so called 'operational taxonomic units' (OTUs), in terms of their functional role [e.g. the KEGG orthology, @kanehisa12]. 

In the animal kingdom, n unknown plethora of single datasets has been published along with article publications [e.g., @gossner15, @gossner15c, @renner14; @birkhofer17; @vanderplas17; @sokol11; @price13]. In the past two decades, many datasets have been aggregated and harmonized into greater collections with particular organismal focus [e.g. @ricklefs17; @oliveira17; @iversen17; @myhrvold15; @kissling14; @jones09].
Specialised trait-databases have been created that cover different organism groups, interaction types, or ecosystems. For instance, the BETSI database collects traits on soil invertebrates [http://betsi.cesab.org/; @pey14], and the Carabids.org web portal collects traits of carabid beetles (http://www.carabids.org/). 
Unlike for plants, a single unified webservice and harmonizing scheme for animal trait data is lacking. It has been pointed out that data on animals, covering multiple trophic levels and having evolved very different body plans will not be a simple task and go far beyond the efforts produced for plant traits [@moretti17].

A general database for major physiological and life-history traits of all taxa is the Encyclopedia of Life Trait Bank project [@parr16, see below for more detail]. 

A global overview of existing trait data for all taxa and trait types is difficult to obtain. As an attempt to track  databases and unstandardised datasets from independent publications, we initiated a living spreadsheet (https://goo.gl/QxzfHy) which lists published trait datasets with their their taxonomic focus and extend of traits covered, their location on the internet and the terms of use. We invite data owners and users to register further datasets qualifying as trait data.  

## Thesauri and Ontologies

A glossary or thesaurus of terms would be primarily a listing of definitions. Expanding on this, ontologies are linking individual terms with each other by formally defining the relationship between terms, to the ends of enabling a computational interpretation of data. 

Lists of trait definitions have been developed for plants, for instance in the TOP Thesaurus of plant traits, which is employed in the TRY database [@garnier17], the Planteome ontology for crop plants [@walls12; @jaiswal05; www.plantontology.org], or the Flora Phenotype Ontology [http://obofoundry.org/ontology/flopo.html, @hoehndorf16]. On the side of animal traits, a couple of initiatives published comprehensive trait ontologies, like for instance a Thesaurus for Soil Invertebrate Trait-based Approaches [T-SITA, http://t-sita.cesab.org/, @pey14], the Hymenoptera Anatomy Ontology [@yoder10, http://hymao.org/], or the vertebrate trait ontology [http://obofoundry.org/ontology/vt.html, @park13]. Hosting services for  biological ontologies and terminologies are the OBO foundry (http://obofoundry.org/) or ontobee (http://www.ontobee.org/).

Relating the trait definition to methodological definitions on trait measurements are key for harmonizing trait data already at the physical process of measuring. For instance, handbooks for measurement protocols provide such a methodological standardization for plants  [@cornelissen03; @perez-harguindeguy13] or invertebrates [@moretti17] and should be referenced when defining traits in Thesauri.

While trait thesauri define traits for focal groups of organisms, they are not specifying the  formats or structures in which trait data should be stored. 
The Encyclopedia of Life (EOL) project has advanced on the topic by developing the database framework TraitBank [@parr16], which is to date the most general approach to providing a semantic structure for trait data tables. The framework employs established terms provided by the Darwin Core Standard [DWC, @wieczorek12], relates trait definitions to specific trait ontologies and phenotypic and anatomical ontologies, and maps taxa to global identifiers in taxonomic hierarchies of multiple name service providers (e.g. GBIF Taxonomy) to capture synonyms, misspellings and controversies [@parr16]. Additional layers of information capture bibliographic reference, multimedia archives and ecological interactions. TraitBank invites data submissions to the EOL database in a structured Darwin Core Archive. 
Another initiative aimed for a mobilization of measurements from museal collection data collated in the VertNet database into harmonized traitdata [@guralnick16]. 

However, the Terms of the Darwin Core Standard and available Extensions, TraitBank and VertNet, are responses to a demand in Biodiversity Data Management for structured input from the research communities. 
The practical importance of these frameworks for data providers and data users is less obvious and often not appropriately considered when storing or uploading data to low-threshold  file-publishing services. None of the aggregated and harmonized databases for birds, amphibians, or mammals employs such ontologies or semantic web standards. One reason certainly is compatibility: to date, the existing trait data standards do not cover all of the aspects required in the various use-cases of trait-based research. <!-- examples for blind spots --> 
The other reason is competence: trait-data providers are not trained in the demands of biodiversity data-management and complying with the apparently non-intuitive data structures is an investment without immediate pay-off.

# 3. The structure of trait datasets

As a minimum consensus, trait datasets may be defined as follows: 
A trait dataset contains quantitative measurements or qualitative facts (i.e. trait values) about physical phenotypic characteristics of fitness, behaviour or performance (i.e. traits) observed on an individual organism or parts of an individual (i.e. an entity) that can be assigned to a biological taxon (i.e. a species or higher-level taxon) <!-- or larger organisational unit, e.g. a community -->. 

Quantitative measurements are values obtained either by direct morphological or physiological measurement on single specimens (Fig. 1*a*), by aggregation of replicated measurements on multiple entities (Fig. 1*b*) or by estimation of means or ranges for the species as reported in literature or other published sources (e.g. databases, Fig. 1*c*). Qualitative facts are assignments of an entity to a categorical level, e.g. of a behavioural or life-history trait (Fig. 1.*d*). The entity or observation (i.e. the occurrence) to which the reported measurement or fact applies may differ in resolution -- depending on the scientific question -- and could be a sub-sample or bodypart, an individual specimen, an entire species or higher-level taxon.  
This definition refines the entity-quality model (EQ) as described by @garnier17, where a trait observation is phrased as 'an entity having a quality'. 
<!-- Cat: the difference [between quantitative measurements or qualitative facts] is not totally clear to me - is it the difference between field vs literature gathered data?  Flo: I see measurements as quantitative and facts as qualitative traits, which mostly aligns with field vs. literature, but not entirely. A differentiation that has rarely been specified in the trait-litereature it seems. Garnier et al 2017 just speak of "quality", which is assigned to an "entity", without specifiying it further. Nadja: I like this differentiation, to me this nicely reflects morphometric and literature traits (although its of course not exclusive). Maybe it helps if we add an example after the definition? --> <!--FDS: is this sufficiently defined now? the distinction between measured and literature data is spelled-out further below, when discussing the formats. Quantitative values can be taken from literature, though, like body sizes for example. -->  

\begin{figure}[t]
\centering
\includegraphics[width=0.85\textwidth, angle=0]{figures/fig1.png}
\caption{Types of ecological trait data assume different entities or reported quantities. a) morphometric or morphological measurements of individual body features (lengths, areas, volumes, weights) or other quantities related to life history (e.g. reproductive rates, life spans); b) aggregated traits are reported as means taken on multiple measures of members of a taxon; c) quantities may be extracted from literature or existing databases, referring to the entire taxon (or a subset, e.g. a sex) as the entity of description; d) Qualitative traits are categorical or binary descriptors of the entire taxon.}

\end{figure}

Trait-datasets usually take different formats. For instance, if trait data have been collated at the species level from different literature sources or from expert knowledge, or as aggregated measurements collated from raw data, they usually are reported in a species$\times$trait matrix (a.k.a. wide-table format,  Fig. 2*a*), with a column of trait values for each trait recorded and a row for each species (or taxon) for which data were available. This format is usually reporting missing data as 'NA'. It may store additional information (e.g. on variation of means or literature source) in secondary columns or accompanying datasheets. 
The matrix format is widely used for the production of lookup-tables at the species level, which may either report qualitative facts (from literature or  expert knowledge) or aggregated quantitative traits (averages or ranges for this taxon). Quantitative traits at this resolution for instance may be used for the calculation of community weighted means or functional diversity metrics at the community level [@laliberte10a]. Also at this level of information, missing entries, e.g. about the trait values or behaviour and functional role of a rarely described species, may be inferred from these tables combined with phylogenies ['gap filling', @swenson14] and using Bayesian approaches [@schrodt15]. However, the tables do not allow to estimate within-species variation or assess the quality of the reported measurement or fact [@guralnick16]. 

Phenotypic variation of traits within species would usually be recorded per observation, i.e. each individual specimen would be recorded in a row, with the respective trait values stored in columns. In these occurrence data tables (Fig. 2*b*), each row is centered around a single physical occurrence of a species. Those data are common in investigations of evolutionary trade-offs and trait correlation, or of intra-specific variation along environmental gradients. Occurrence data would also capture phenotypic variation arising from morphotypes and sexual dimorphism. This format is the most intuitive for recording own empirical measurements and therefore is common for measured quantitative data and rarely found for reported qualitative facts obtained from the literature.

Both the matrix format and the occurrence table format have the different trait values which have been collected or measured on one entity stored in multiple columns. Each trait column represents one measured value. The width of the table varies with the number of traits included. Merging data from different sources requires manual harmonization of these structures. 

Computationally more effective and allowing for highest flexibility is the storage of traitdata in 'tidied' long-table formats [@wickham14; @parr16], where each row is reserved for a single measurement or fact of a specific trait, referenced to a single specimen  which is assigned to a specific taxon (i.e., one occurrence). This allows for storing repeated measurements on a single individual by linking the data of separate rows via a unique specimen ID. Also, multivariate trait measurements can be recorded in this format by linking multiple rows via a unique measurement identifier.  Long-table dataset purport multiple advantages for the data manipulation (e.g. filtering, sub-setting and aggregating data), visualization (e.g. plot measured values by factor variable or taxon) and statistical modelling (e.g. ANOVA for testing difference of trait value by sex) [@wickham14]. They further draw from a defined set of columns, which simplifies merging of datasets. 

The latter format thus provides the highest resolution and is ideal for storing raw data of trait measurements in huge databases and for linking them via the respective unique identifiers to additional information, such as occurrenceID or locationID. Each row of the dataset can be interpreted as a statement of an 'entity x having a qualitative/quantitative feature y' (Fig. 2*c*). 

<!-- Cat: here I guess that a figure showing few rows of a species x trait matrix and a long-format one with links to other information (metadata,..) woud help quite a lot. By experience some users are not be familiar with the interest of using the long-table format.-->
<!-- Malte: I agree (see my comment above), such a figure / would definitely help convince people (or at least not to lose them). -->

\begin{figure}[t]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig2.png}
\caption{Formats used for trait datasets.  a) taxon-level trait data compiled from literature or aggregated from measurements are often published as a compiled species $\times$ traits matrix; b) wide-tables report measurements on the specimen-level, i.e. one physical occurrence of a taxon; c) a measurement long-table is a well defined and tidy data format, reporting one single measurement per row and relating it to a trait definition and taxon via unambiguous identifiers. Additional identifiers relate each row to other layers of information on d) the specimen (occurrence) or e) the origin or confidence on the reported measurement or fact.}

\end{figure}

# 4. Proposing an ecological traitdata standard 

With this paper, we propose a versatile, minimal data structure, as well as a defined vocabulary that captures the different degrees of resolution and measurement detail for multiple use cases of trait-based research. We were drawing on the combined expertise of empirical biodiversity researchers (data providers), biodiversity synthesis researchers (data users), and biodiversity informatics researchers (data managers).  
We were particularly considering the work of @kattge11, and @garnier17, as well as @parr16 to ensure compatibility of datasets with major web-services and existing standards for biodiversity data management. 

The online reference glossary of terms and description of the proposed structure is found at https://ecologicaltraitdata.github.io/TraitDataStandard/.

The vocabulary proposed will not capture the entire complexity of trait-based research in ecology and evolution. We invite researchers to comment on and contribute to the next iterations of this standard and develop own applications and ontologies. 

## structure for trait data

As described above, tidy long-table formats are the most predictable and compatible format for the core content of trait datasets and should be used for standalone uploads on low-threshold repositories. 

To enable a better standardisation of nomenclatures and values, the entries of the core dataset should be related to semantic web references using Uniform Resource Identifiers (URIs) keep a duplicate record of accepted taxon names and standard trait names. In these duplicate columns, values should be reported as 'harmonized' entries, which are linked to semantic ontologies and globally valid identifiers, consistent with semantic web standards [@berners-lee01; @wieczorek12; @page08]. 
If data contain multi-layered information on the measurement or occurrence (including geolocation and date information), further columns may be added to the core set (see description for Extensions below). 
For reasons of reproducibility and openness, data should not be uploaded in proprietary spreadsheet formats (like '.xlsx') but rather in comma-separated text files ('.csv' or '.txt').

The metadata of any dataset that employs this data structure should refer to the online version as "Schneider et al. 2017 Ecological Traitdata Standard v1.0, DOI: XXXX.xxxx, URL: https://ecologicaltraitdata.github.io/TraitDataStandard/v1.0/". In addition to the versioned online reference, the dataset should also cite this paper for an explanation of the rationale.  

In very complex datasets repetition of verbose entries and long character strings in a single data table should be avoided [@wickham14]. Therefore we recommend use of  Darwin Core Archives [DwC-A, @robertson09], the structure proposed for GBIF [@gbif17,  http://tools.gbif.org/dwca-assistant/] or EOL TraitBank [@parr16,  http://eol.org/info/cp_archives]. This also integrates a defined framework for metadata in the Ecological Metadata Language [@knb11], which should be used for describing and linking the data tables. 

In some contexts, it may be safer to add such information directly as own columns of the data table to facilitate an analysis of co-factors and correlations further down the road. This way, if datasets of different source are merged, the information is readily available without the risk of breaking the reference to an external datasheet.
The trade-off is data-consistency and readability *vs.* avoidance of content duplication. 

Other cases, where key data columns would be placed in the same table as the core data are traits assessed on a higher level of organisation, e.g. microbial functional traits assessed at the community level taken from a soil sample. Here, location or measurement information are in the primary focus of the investigation (see vocabulary extensions below). 
A general definition, whether a column describes asset data or is part of the central dataset is ill advised. Therefore, our glossary of terms and its extensions should be used to describe the scientific data according to the study context. 

## vocabulary for trait data

Column names and, where possible, field entries should follow the defined vocabulary listed at https://ecologicaltraitdata.github.io/TraitDataStandard/. 
Individual terms should be referenced via their global identifier, which is hosted by the GFBio Terminology Service and also can be accessed programmatically (i.e. via the Application Programming Interface, API). Wherever our glossary refines or duplicates existing terms from other ontologies for biological data, like the Glossary of EOL (http://eol.org/info/516) and Darwin Core (http://rs.tdwg.org/dwc/terms/), we indicate this in the fields 'refines' or 'identical', respectively.

### Core columns

To describe the minimum core content, where each row is the reported measurement or fact for a single observation, the following columns are required: a value (column `traitValue`) and -- for numeric values -- a standard unit (`traitUnit`); a descriptive trait name (`traitName`);  the scientific taxon for which the measurement or fact was obtained (`scientificName`). 
For these core values, we highly recommend to use unambiguous and self-explanatory vocabularies for trait names and taxa. 
However, to have the data compatible with own databases or analytical code, it might be necessary to use abbreviations or identifiers instead.

### Relate to global taxon and trait definitions 

It is indispensable to relate the user-defined vocabulary to a consensus standard of taxon names or or look-up tables of traits. This is achieved by adding globally valid identifiers for taxa (`taxonID`) and trait definitions (`traitID`), complemented by the more readable accepted names (`ScientificNameStd` and `traitNameStd`, respectively), according to a public ontology (e.g. GBIF Backbone Terminology; for *Bellis perennis*, the taxonID would be  'https://www.gbif.org/species/3117424'; in TOP Thesaurus, the traitID for 'fruit mass' would be  'http://top-thesaurus.org/annotationInfo?viz=1&&trait=Fruit_mass').  
By allowing for a double record of both user-specific and standardised entries, we acknowledge the fact that most authors have their own schemes for standardisation and may refer to different scientific community standards  [as practised in TRY; @kattge11]. This redundancy of data enables continuity on the author and data owner side and quality checking and comparability on the data user side. Keeping an additional reference to a kingdom and class (using columns `kingdom`, `class` as described by DWC and suggested for EOL Traitbank) will avoid ambiguities with taxonomic homonyms. 
It is best practice to keep a record of the reported taxon rank (in column `taxonRank`) for filtering purposes for traits recorded at the family or genus level. This information can be used for trait inference of higher taxonomic resolution using hierarchical probabilistic matrix factorisation [@schrodt15]. 

The existing thesauri and ontologies for traits differ widely in terms of hierarchical depths and detail, as well as curation efforts and measures for peer-reviewed quality control. The accessibility by software differs widely. For biodiversity data, the OBO Foundry (http://www.obofoundry.org/) or the GFBio Terminology service (https://terminologies.gfbio.org/browse/), provide hosting for advanced trait ontologies and offer API services for computational access. More compatibility and availability of trait thesauri could greatly improve the possibilities for a comparison of trait data from different sources. 

If no published trait definition is available that can be referenced, trait-datasets should be accompanied by a dataset-specific glossary of traits, or thesaurus. A thesaurus provides a "controlled vocabulary designed to clarify the definition and structuring of key terms and associated concepts in a specific discipline" [@laporte13; @garnier17]. 
Thus, in its simplest form,  a trait thesaurus would assign trait names with A) an unambiguous definition of the trait and B) an expected format (e.g. units or legit factor levels) of measured values or reported facts. It might additionally provide semantic relationships between terms for deriving a hierarchical or tree-based classification of traits. 
By providing a minimal vocabulary for this case (https://ecologicaltraitdata.github.io/TraitDataStandard/#terms-for-traitlists-a-trait-thesaurus), we hope to facilitate the publication of trait thesauri developed for the own project context which always should be referenced in the core trait dataset. The thesaurus might accompany the core data in a Darwin Core Archive, or be published on any other stable webservice. 

See our online reference for recommendations how to publish a minimal Thesaurus compliant with semantic web standards [http://ecologicaltraitdata.github.io/TraitDataStandard/#terms-for-traitlists-a-trait-thesaurus].  We also provide an example thesaurus, that has been developed to capture traits assessed within the German Biodiversity Exploratories project on Github (http://ecologicaltraitdata.github.io/TraitDataList/). 

### Metadata

Information about the authorship and ownership of the data and the terms of use should be considered when sharing and working with trait datasets.
We define a set of terms (https://ecologicaltraitdata.github.io/TraitDataStandard/#metadata-vocabulary) that allows to relate trait data to authorship and define a bibliographic reference and license model. 

In the case of raw measurement data, these information apply to the entire trait dataset, which classifies them as metadata stored in the information structure where the data are published. 
In cases, where individual measurements and facts from different sources are compiled into a trait database, these metadata information must be linked to the measurement level, via an unambiguous `datasetID` and a descriptive `datasetName`. 

<!-- 
(We conceptualize trait-datasets to  not include information about a species' interaction partners/associations of an organism, since they are -- to our understanding -- not within the scope of functional traits. However, such network data may be of interest to be combined with trait data. ) <!-- Malte: which can only be achieved if both types of datasets have been related to an ontology of species names. -->

### Extensions for additional data layers

Beyond measurement units or higher taxon information, further information might be available that may not be core data, but are related to the individual or specimen, or to the reported fact, measurement or sampling event. 

The data standard provides two extensions of the vocabulary that should be used to describe these information:

- the `occurrence` extension (https://ecologicaltraitdata.github.io/TraitDataStandard/#extension-occurrence) contains vocabulary to describe information on the level of individual specimens, such as physiological specifications of the phenotype, such as sex, life stage or age. This also includes the date and geographical location, as well as the method of sampling and preservation, which provides an important resource to analyse trait variation due to differences in space and time.
- the `measurementOrFact` extension (https://ecologicaltraitdata.github.io/TraitDataStandard/#extension-measurement-or-fact) provides terms to describe information at the level of single measurements or reported facts, such as the original literature from where the value is cited, the method of measurement or statistical method of aggregation. It aims at important information to track potential sources of noise or bias in measured data (e.g. variation in measurement method) or aggregated values (e.g. statistical method applied), as well as the source of reported facts (e.g. literature source or expert reference).   

Many terms of these extensions refine terms of the Darwin Core Standard and their own [MeasurementOrFact](http://rs.tdwg.org/dwc/terms/index.htm#measureindex) and [Occurrence](http://rs.tdwg.org/dwc/terms/index.htm#Occurrence) extensions and EOL TraitBank's use of those terms (http://eol.org/info/structured_data_archives). 

Same as for taxon or trait information, these additional layers of information can either be added as extra columns to the core dataset or, to  avoid redundancy and duplication of content, be kept in separate data sheets of a Darwin Core Archive (see above). A unique identifier would link to these other datasheets, encoding each individual occurrence of a species (`occurrenceID`), single measurements or reported facts (`measurementID`), locations of sampling (`locationID`) and sampling campaigns (`eventID`). 
Some data-types may directly refer to existing global identifiers for occurrence IDs, e.g. a GBIF URI or a museum collection code referencing the precise specimen at which the measurement was taken and that is publicly available.

<!-- Malte: For readers not that familiar with trait data, the differences between measurement, occurrence and trait ID might not be that clear. It might be worth adding either a sentence explaining in an example what is what OR a tiny figure that shows how a specimen is sampled, processed, traits are measured etc. -->
 <!-- Nadja: I like the idea of having a figure explaining the different terms -->

# 5. Computational tools for producing compliant data

To access data from public data-bases, the R-package 'traits' [@chamberlain17] contains functions to extract trait data via several open API interfaces including Birdlife, EOL TraitBank or BetyDB. The package  'TR8' provides similar access to plant traits from a list of databases [including LEDA, BiolFlor and Ellenberg values; @bocci15] and aggregates them into a species-trait matrix. However, none of these packages provides harmonization of traits into a unified scheme to facilitate further analysis. 

We provide the R package 'traitdataform' to assist in producing data compliant with the trait data standard proposed here. There are two major use cases for the package:

- preparing of own trait datasets for publication on low-threshold file-servers and project databases, and
- harmonising trait datasets from different sources by moulding them into a unified format. 

A comprehensive documentation of the package is found in the package vignette, and on its Github repository (https://github.com/EcologicalTraitData/traitdataform) and the documentation website (http://EcologicalTraitData.github.io/traitdataform/). 

The package is under continuous open source development and invites participation in development, comments or bug reports via the Github Issue page (https://github.com/fdschneider/traitdataform/issues). 

The key function of the package is `as.traitdata()` which moulds a species-trait-matrix or occurrence table into a measurement longtable format. It also maps column names into terms provided in the trait data standard and adds metadata as attributes to the output object. This example converts an own file 'data.csv' into a dataset of long-table structure that employs the standard vocabulary for core data:

```
library("traitdataform")
dataset <- as.traitdata(read.csv("path/to/data.csv"), 
             traits = c("body_length", "antenna_length", 
                        "metafemur_length"),
             units = "mm",
             taxa = "name_correct", 
             keep = c(locationID = "location")
             )

```

The column containing taxon names is given in parameter 'taxa'. Note that the parameter 'keep' specifies and renames any data that should be maintained in the output. The parameter 'units' is used to specify the units of measurement. 

\begin{figure}[h]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig3.png}
\caption{Process chart of the functions provided within the R package.}

\end{figure}

Scientific taxon names are matched automatically to the GBIF Backbone Taxonomy and added to the core data by calling the function `standardize.taxonomy()`. Similarly, the `standardize.traits()` function matches user-provided trait names onto a given trait vocabulary and links it to an existing trait ontology or a user-provided table of trait definitions, i.e. a thesaurus of traits. The same function harmonises trait values into target units and legit factor levels. 
The following example harmonizes traits based on a minimal thesaurus, referencing trait names with globally valid URIs defined by the BETSI thesaurus of soil invertebrate traits:

```
traitlist <- as.thesaurus(
       body_length = as.trait("body_length", traitUnit = "mm", traitUnitStd = "mm", 
         traitID = "http://t-sita.cesab.org/BETSI_vizInfo.jsp?trait=Body_length"), 
       antenna_length = as.trait("antenna_length", traitUnitStd = "mm", 
         traitID = "http://t-sita.cesab.org/BETSI_vizInfo.jsp?trait=Antenna_length"),
       metafemur_length = as.trait("metafemur_length", traitUnit = "mm",
         traitID = "http://t-sita.cesab.org/BETSI_vizInfo.jsp?trait=Femur_length")
)

datasetStd <- standardize.traits(dataset, thesaurus = traitlist)
```

The function `as.thesaurus()` provides a structured object that is required by the function `standardize.traits()`. Other ways of defining a thesaurus object are documented in the package vignette and function documentation (`?as.thesaurus`). Future iterations of the R package will aim at automatising the generation of trait thesaurus objects from globally available Ontologies. 

The package functions form a toolchain where each function can be piped as an input into the next. A wrapper function `standardize()` applies all functions sequentially, making transferring and harmonising traitdata as simple as:

```
datasetStd <- standardize(read.csv("path/to/data.csv"),
            thesaurus = traitlist,
            taxa = "name_correct", 
            units = "mm"
            )
```

Datasets that have been produced by these functions can easily be appended using the `rbind()` function of R base, while maintaining any available metadata information as separate column entries. 

To merge datasets with additional information on the occurrence or measurement level, secondary data tables can be added as columns of the core dataset according to a unique identifier using the function `merge()`. This enables an easy handling of data sources that have been stored in a Darwin Core Archive format. 

Since the intention of the package is to facilitate upload of own data to hosting services but also simplify the access to published trait data, the package offers direct access to trait datasets that have been published in the public domain or under creative commons licenses [@gossner15; @gossner15c; @ricklefs17; @vanderplas17; @myhrvold15]. See a full list of linked datasets on the package  website or by calling `data(package = "traitdataform")`. We invite users and authors of datasets to add further data to the package and thereby contribute to this registry for distributed trait datasets. 

# Conclusion

In this paper, we propose a general, low-threshold structure and vocabulary for ecological trait data. 
We also provide a toolbox in R to transfer and harmonize data into this scheme. 

To date, existing initiatives and centralised infrastructures for standardisation and data  hosting are focused on a constrained organism group, ecosystem type or region. To make these heterogeneous sources of data more accessible and interlinkable, we propose a traitdata standard terminology and formatting scheme that can be applied in multiple contexts, for instance for building an own taxon-specific database or in user-side compilations of trait data from different sources. 

The tools proposed here help to standardise trait datasets before upload to central as well as distributed data repositories. In times of increasing demand for open research data and international platforms for biodiversity data management, the development of meaningful terminologies for the standardisation of biodiversity data is more than essential: defined ontologies enable researchers to relate published datasets to each other to achieve a greater synthesis, thereby opening the doors for new insights on global biodiversity patterns. In terms of data science, semantic web ontologies are a prerequisite for advanced big data mining using machine learning techniques.

We therefore encourage the development of trait thesauri in semantic ontologies. The biggest challenge in community efforts of standardisation of traits may be the investment in consensus building which leads to an acceptance and establishment of the methodological and conceptual definitions of traits. This effort is time consuming and lengthy, but it returns synergies and great scientific benefit by enabling a global synthesis, as the example of the global plant trait database TRY shows intricately [@diaz16]. 

Furthermore, the tools provided here can be applied to harmonise and combine data from heterogeneous sources. Such tools to handle the extensive raw data resources on traits are a prerequisite for research on trait variation along environmental gradients or intra-specific variation. It also opens the ground for a development of best practice principles for the filtering of raw measurements when creating species-level mean values. 
We hope this aids research on the response of traits to environmental and anthropogenic drivers, synthesis on trait patterns on all ecological scales [@messier10], as well as multi-trophic trait analysis [@soliveres16; @allan15]. 

The trait data standard as well as the R package are published for open source community development. Contributions are invited and vital for the future development of the data standard and an extension and improvement of the R package. 


# Acknowledgements

Thanks to all respondents to an internal online survey on trait data for the Biodiversity Exploratories project and to Matthias Biber, Diana Bowler, Klaus Birkhofer, Runa Boeddinghaus, Martin Gossner, Catrin Westphal  for comments on the manuscript, the trait data standard and the R-package. 

We thank the managers of the three Exploratories, Kirsten Reichel-Jung, Juliane Vogt, Miriam Teuscher, and all former managers for their work in maintaining the plot and project
infrastructure; Christiane Fischer and Jule Mangels for giving
support through the central office, Michael Owonibi for
managing the central data base, and Markus Fischer, Eduard Linsenmair,
Dominik Hessenmöller, Daniel Prati, Ingo Schöning, François Buscot,
Ernst-Detlef Schulze, Wolfgang W. Weisser and the late Elisabeth Kalko
for their role in setting up the Biodiversity Exploratories project. 

The work has been partly funded by the DFG Priority Program
1374 "Infrastructure-Biodiversity-Exploratories" (DFG-Refno.). Field work
permits were issued by the responsible state environmental offices of
Baden-Württemberg, Thüringen, and Brandenburg (according to § 72
BbgNatSchG).

M.J. was supported by the German Research Foundation within the framework of the Jena Experiment (FOR 1451) and by the Swiss National Science Foundation.

# Authors' contributions

FDS, MJ, GP, AO, CP, and NKS conceived the ideas for the manuscript and critically revised the glossary of terms of the trait data standard; FDS led the writing of the manuscript; NKS authored the list of traits; FDS developed the R package. All authors contributed critically to the drafts and gave final approval for publication.

# References 

\footnotesize