---
title: "Introducing an Ecological Trait-data Standard"
author: Florian D. Schneider, Malte Jochum, Gaetane LeProvost, Andreas Ostrowski,
  Caterina Penone, Nadja K. Simons
csl: amnat.csl
output:
  pdf_document: default
  word_document:
    keep_md: yes
geometry: margin=1.5in
header-includes: \usepackage{lineno}
fontsize: 12pt
bibliography: div05.bib
---

\linenumbers

# Abstract

1.	The assessment and analysis of functional traits has been established in all domains of ecological research promising great advancements for biodiversity and climate research. A wealth of data is available from multiple sources and for many organism groups but is lacking standardisation due to heterogeneous data formats and data types. 
2.	To harmonize and standardize trait data, we present a first version of an ecological trait-data standard for storing and sharing functional trait data.
3.	We further provide an R-package that assists in transferring own data structures into the data standard and to harmonize and merge trait datasets from heterogeneous sources. 
4.	With these decentralized tools for traitdata management, we intend to facilitate the exchange and analysis of trait data for ecological research. 

# Key-words: 

functional ecology, traits, semantic web, ontologies, data standard

# Introduction

Functional traits are phenotypic characteristics that are related to the fitness and performance of an organism [@mcgill06d; @violle07] and have been used to describe the role of species in ecosystems [@fontaine05] or their ability to persist under certain environmental conditions [@Forsman2015] (@refs).

<!-- more refs:  Reich et al., 1997; Wright et al., 2004; Chave et al., 2009; Schrodt et al., 2015-->

Traits have proven useful for bridging individual level physiology and behaviour to processes at the ecosystem scale and patterns of biodiversity at the global level [@diaz13; @diaz16]. Comparative study of traits informs researchers about physiological and evolutionary trade-offs, and helps identifying economic strategies of organisms beyond taxonomy [@menezes10; @forister15;  @siefert15; @gross09]. Inferring an unknown functional role or ecological function of individuals from their apparent features [@duarte11; @schrodt15] is a promising venue to bypass taxonomic impediment, i.e. the fact that a majority of species are yet undescribed and little is known of their interactions with the environment and other organisms. For instance, defense traits and predator feeding traits have co-evolved, just as pollinator and flower traits. This 'trait-matching' enables the inferrence of ecological interactions from observed features of live specimens [@leprovost17; @kalinkat13] which may be used to inform ecological networks [@munoz17; @weiner14; @miranda13; @digel11] and parameterize ecological population-dynamic models [@schneider12].   
On the level of species assemblages, community weighted means or variance of functional traits [@laliberte10a] or multivariate measures of 'functional diversity' [@petchey09; @villeger08a] are used to understand changes in functional composition or the loss of functional diversity in consequence to global change or local anthropogenic land use [environmental filtering; @jennings09; @gossner16; @mangels17; @simons16; @hopfenmuller14; @mouillot13;  @vandewalle10; @perovic15; @borschig13; @birkhofer15; @birkhofer17]. Since traits are closely related to ecological mechanisms, such patterns in functional community composition are promising versatile statistical tools to assess the current state of an ecosystem [ @moretti13; @kuhsel15;  @gamez-virues15; @birkhofer14; @gravel17; @lajoje15] or of ecosystem services [@wood15b; @bello10; @diaz07; @lavorel12; @allan15].  Due to the abovementioned trait-matching, these analyses are also a promising venue for a multitrophic community ecology aiming to unravel dependencies and synchrony in trait spaces between interacting functional groups [@soliveres16].  
In recent years, functional ecology has thus grown into a major field of ecological research, linking basic physiological knowledge to functions and services as well as applied matters in land use management and biodiversity conservation, as well as risk assessment of anthropogenic stressors and climate change.  

Given the expected future challenges in dealing with climate change and managing land use, such global trait synthesis is an interdisciplinary/cross-topic field with great potential. 
<!-- Nadja: Some thoughts/snippets based on other publications: " a relevant amount of data is available to quantify and understand trait variation beyond aggregation on species level. The analysis presented here is only a first step to disentangle within- and between-species variability. [...] A more empirically based representation of functional diversity is expected to contribute to an improved prediction of biome boundary shifts in a changing environment. [...] Empirical inference of this kind is expected to stimulate and strengthen hypothesis-driven research (Golub, 2010; Weinberg, 2010) towards a unified ecological assessment of plant traits and their role for the functioning of the terrestrial biosphere." [@kattge11] / "Our traitbased approach to modelling competition makes generalization possible across the forest ecosystems of the world and their highly diverse species composition. [...] A challenge for the future is to move beyond growth to analyse all key demographic rates and life-history stages, and analyse how traits influence competitive outcomes and stable coexistence at the population level."[@Kunstler2016] / "Traits include those that in combination allow a first-order assessment of species' similarity in "ecological niches" or the overlap in their functional role. The standardized interpretation and adherence to source information and clear documentation aims to provide a replicable and transparent basis for research. [...] The trait variables in the dataset should support hypothesis testing or applied inference in ecology, conservation and evolution." [@Wilman2014] -->
<!-- Prospect: what will trait analysis acheive, if global data across taxa are available, 

- intra-specific variation and evolutionary trade-offs
- global variation
- community functional pattern, comparative study across continents
- big data challenges
-->

## Overview of trait-data initiatives

Despite these obvious benefit and grand prospects, an integration of several studies across taxa, trophic levels, or organisational scales remains difficult because of different organismal, methodological, and functional focus, which renders data on traits extremely heterogeneous. 
Trait data have therefore been standardised into centralised databases for specific organism groups and research questions. These services provide platforms for consensus building in methods, attract data submissions from a focal community of researchers and map them into a common scheme. Furthermore, the databases provide access management services and provide data usage policies, while simplifying data queries for synthesis researchers.

Most prominently, plant traits from many different sources  have been collated into the quickly growing TRY database [@kattge11; e.g. including the extensive LEDA traitbase for the Northwest-European flora, @kleyer08].  Other more specialised databases for plant traits exist, for instance flower traits [BiolFlor for German wild flowers, @klotz02] and root measurements [Fine-root Ecology Database; @iversen17]. Traits on crop plants have been collected in agronomy databases like Planteome and BetyDB [https://www.betydb.org/; @lebauer17]. Trait-data for lower plants are less completely assessed. <!-- TRY collates trait data from other published databases with individually owned datasets, provides an extended access management, data query services and tracking of data usage in studies and secondary publications. The great achievement of TRY is to harmonize these data into a unified scheme and provide a single platform of global coverage, which takes a huge effort to attract data submissions and map them into a common scheme. Repeating this exercise for other organism --> 
<!-- Nadja: This is of course the major achievement of TRY, but I think we don't have to mention it because this not our immediate goal with the trait standard. The important bit for us is that we provide tools to harmonize datasets. -->

Microbial ecology also makes great use of trait-based approaches to assess genomic function and describe functional diversity at the community level in community-aggregated traits [@krause14a; @fierer14; CAT, @fierer12] derived from metagenomic analysis. <!-- databases or datasets on microbial traits? --> 

In the animal kingdom, specialised trait-databases have been created that cover different organism groups, interaction types, or ecosystems. For instance, the BETSI database collects traits on soil invertebrates [http://betsi.cesab.org/; @pey14], and the Carabids.org web portal collects traits of carabid beetles (http://www.carabids.org/). Besides these databases, an unknown plethora of single datasets has been published along with article publications [e.g., @gossner15, @gossner15c, @renner14; @birkhofer17; @vanderplas17  <!-- FDS: add some classics of trait data --> @Sokol2011, @Price2013 <!-- Nadja: oldest relevant pub. in Dryad-->]. In the past years, many datasets have been aggregated and harmonized into greater collections with particular organismal focus [e.g. @ricklefs17; @amphibio17; @iversen17; @myhrvold15; @kissling14; @jones09]. Unlike for plants, a single unified platform and harmonizing scheme for animal trait data is lacking. It has been pointed out that data on animals, covering multiple trophic levels and having evolved very different body plans will not be a simple task and go far beyond the efforts produced for plant traits [@refs].

A global overview of existing trait data for all taxa and trait types is difficult to obtain. As an attempt to track  databases and unstandardized datasets from indipendent publications, we initiated a living spreadsheet (https://goo.gl/QxzfHy) which lists published trait datasets with their their taxonomic focus and extend of traits covered, their location on the internet and the terms of use. We invite data owners and users to register further datasets qualifying as trait data.  

## Need for ontologies to enable data standardisation

Public research funding agencies increasingly demand the publication of data without access restrictions [@swan12; @allison15]. However, although the number of trait-based studies grows constantly, not all scientific databases for functional traits comply with this demand for open data. While it seems clear that factual data, being not a product of a creative act, do not qualify as intellectual property and copyright does not apply [@ref], there is still no broad consensus achieved in the scientific community whether such data from public funding must be generally dedicated to the public domain, or whether the investment in production must be acknowledged by attribution or co-authorship.  <!--The training on compliance with open data standards within the community of environmental sciences can still be much improved [@schmidt16]. legal standpoint in most countries: descriptive data are not copyright protected. Publicly fundet research must provide products. is still debated by the scientific community. -->

Open-access publications in low-threshold data repositories such as Figshare, dataDryad, Researchgate or Zenodo.org are gaining importance and foster a decentralised data hosting with low expectations regarding data standardisation and documentation. Although open for manual download, data on those platforms might be stored in proprietary (e.g. .xlsx) or binary (e.g. .pdf) data formats which make a programmatical extraction tedious or impossible. Since the platforms are open for any type of scientific data, no data standards are forced upon the uploaded material. 
Wheter open or not, the publication of original data is increasingly required by journals to warrant reproducibility of science. Given this prospect, there will be no lack of data, but a lack of data standardization hampering efficient future use of published datasets.

To facilitate the standardized upload of research data to reliable and interlinked data infrastructures, data brokering services such as the German Federation on Biological Data (gfbio.org) are likely to gain importance. One goal of such initiatives is to facilitate data publications and standardization for researchers, for instance by providing terminologies and ontologies for biodiversity data.

Ontologies have been developed for plant traits, for instance in the TOP Thesaurus of plant traits, which is employed in the TRY database [@garnier17], the Planteome ontology for crop plants [@walls12; @jaiswal05; www.plantontology.org], or the Flora Phenotype Ontology [http://obofoundry.org/ontology/flopo.html @hoehndorf16]. On the side of animal traits, a couple of initiatives published comprehensive trait standards and method catalogues, like for instance a Thesaurus for Soil Invertebrate Trait-based Approaches [T-SITA, http://t-sita.cesab.org/, @pey14] or the vertebrate trait ontology [http://obofoundry.org/ontology/vt.html; @park13]. A resource for biological ontologies and terminologies is the OBO foundry (http://obofoundry.org/).

While these ontologies define traits for focal groups of organisms, they do not specify in which formats or structures trait data should be stored. 
The Encyclopedia of Life (EOL) project has advanced on the topic by developing the database framework TraitBank [@parr15], which is to date the most general approach to providing a semantic structure for trait data tables. The framework employs established terms provided by the Darwin Core Standard (DWC, @wieczoreck12), relates trait definitions to specific trait ontologies and phenotypic and anatomical ontologies (<!-- define --> PATO, UBERON), and maps taxa to global identifiers in taxonomic hierarchies of multiple name service providers (e.g. GBIF Taxonomy) to capture synonyms, misspellings and controversies [@parr15]. Additional layers of information capture bibliographic reference, multimedia archives and ecological interactions. TraitBank invites data submissions to the EOL database in a structured Darwin Core Archive. 
Another initiative aimed for a mobilization of measurements from museal collection data collated in the VertNet database into harmonized traitdata [@guralnick16]. 

However, the Terms of the Darwin Core Standard and available Extensions, TraitBank and VertNet, are responses to a demand in Biodiversity Data Management for structured input from the research communities. The practical importance of these frameworks for data providers and data users is less obvious and often not appropriately considered when storing or uploading data at file-publishing services. None of the aggregated and harmonized databases for birds, amphibians, or mammals employs such ontologies or semantic web standads. One reason certainly is compatibility: to date, the existing trait data standards do not cover all of the aspects required in the various use-cases of trait-based research. <!-- examples for blind spots --> 
The other reason is competence: trait-data providers are not trained in the demands of biodiversity data-management and complying with the apparently non-intuitive data structures is an investment without immediate pay-off.

# Proposing an ecological traitdata standard 

With this paper, we propose a versatile, minimal data structure, as well as a scheme of a defined vocabulary that captures the different degrees of resolution and measurement detail for multiple use cases of trait-based research. We also provide computational tools in an R package to help users transfer their original data into the proposed scheme.

We build upon the structure proposed for EOL TraitBank [@parr16], which applies and expands terms of the Darwin Core standard [DWC; @wieczorek12]. We further expand the definitions of these terms  and add new key terms for the use case of trait data, particularly considering the work of @kattge11, and @garnier17, and drawing on the combined expertise of empirical biodiversity researchers (data providers), biodiversity synthesis researchers (data users), and biodiversity informatics researchers (data managers).  

The vocabulary proposed will not capture the entire complexity of trait-based research in ecology and evolution. We invite researchers to comment on and contribute to the next iterations of this standard and develop own applications and ontologies. 

## Common structure of trait datasets

As a minimum consensus, trait datasets may be defined as follows: 
A trait dataset contains quantitative measurements or qualitative facts (i.e. trait values) about physical phenotypic characteristics of fitness, behaviour or performance (i.e. traits) observed on an individual organism or parts of an individual (i.e. an entity) that can be assigned to a biological taxon (i.e. a species or higher-level taxon) <!-- or larger organisational unit, e.g. a community -->. 

Quantitative measurements are values obtained either by direct morphological or physiological measurement on single specimens (Fig. 1*a*), by aggregation of replicated measurements on multiple entities (Fig. 1*b*) or by estimation of means or ranges for the species as reported in literature or other published sources (e.g. databases, Fig. 1*c*). Qualitative facts are assignments of an entity to a categorical level, e.g. of a behavioural or life-history trait (Fig. 1.*d*). The entity or observation (i.e. the occurrence) to which the reported measurement or fact applies may differ in resolution -- depending on the scientific question -- and could be a subsample or bodypart, an individual specimen, an entire species or higher-level taxon.  
This definition refines the entity-quality model (EQ) as described by @garnier17, where a trait observation is phrased as 'an entity having a quality'. 
<!-- Cat: the difference [between quantitative measurements or qualitative facts] is not totally clear to me - is it the difference between field vs literature gathered data?  Flo: I see measurements as quantitative and facts as qualitative traits, which mostly aligns with field vs. literature, but not entirely. A differentiation that has rarely been specified in the trait-litereature it seems. Garnier et al 2017 just speak of "quality", which is assigned to an "entity", without specifiying it further. Nadja: I like this differentiation, to me this nicely reflects morphometric and literature traits (although its of course not exclusive). Maybe it helps if we add an example after the definition? --> <!--FDS: is this sufficiently defined now? the distinction between measured and literature data is spelled-out further below, when discussing the formats. Quantitative values can be taken from literature, though, like body sizes for example. -->  

\begin{figure}[t]
\centering
\includegraphics[width=0.85\textwidth, angle=0]{figures/fig1.png}
\caption{Types of ecological trait data assume different entities or reported quantities. a) morphometric or morphological measurements of individual body features (lengths, areas, volumes, weights) or other quantities related to life history (e.g. reproductive rates, life spans); b) aggregated traits are reported as means taken on multiple measures of members of a taxon; c) quantities may be extracted from literature or existing databases, refering to the entire taxon (or a subset, e.g. a sex) as the entity of description; d) Qualitative traits are categorical or binary descriptors of the entire taxon.}

\end{figure}

Trait-datasets usually take different formats. For instance, if trait data have been collated at the species level from different literature sources or from expert knowledge, or as aggregated measurements collated from raw data, they usually are reported in a species$\times$trait matrix (a.k.a. wide-table format,  Fig. 2*a*), with a column of trait values for each trait recorded and a row for each species (or taxon) for which data were available. This format is usually reporting missing data as 'NA'. It may store additional information (e.g. on variation of means or literature source) in secondary colums or accompanying datasheets. 
The matrix format is widely used for the production of lookup-tables at the species level, which may either report qualitative facts (from literature or  expert knowledge) or aggregated quantitative traits (averages or ranges for this taxon). Quantitative traits at this resolution for instance may be used for the calculation of community weighted means or functional diversity metrics at the community level [@laliberte10]. Also at this level of information, missing entries, e.g. about the trait values or behaviour and functional role of a rarely described species, may be inferred from these tables combined with phylogenies ['gap filling', @swenson14] and using bayesian approaches [@schrodt15]. However, the tables do not allow to estimate within-species variation or assess the quality of the reported measurement or fact [@guralnik15]. 

Phenotypic variation of traits within species would usually be recorded per observation, i.e. each individual specimen would be recorded in a row, with the respective trait values stored in columns. In these occurrence data tables (Fig. 2*b*), each row is centered around a single physical occurrence of a species. Those data are common in investigations of evolutionary trade-offs and trait correlation, or of intra-specific variation along environmental gradients. Occurrence data would also capture phenotypic variation arising from morphotypes and sexual dimorphism. This format is the most intuitive for recording own empirical measurements and therefore is common for measured quantitative data and rarely found for reported qualitative facts obtained from the literature.

Both the matrix format and the occurrence table format have the different trait values which have been collected or measured on one entity stored in multiple columns. Each trait column represents one measured value. The width of the table varies with the number of traits included. Merging data from different sources requires manual harmonization of these structures. 

Computationally more effective and allowing for highest flexibility is the storage of traitdata in 'tidied' long-table formats [@wickham14; @parr15], where each row is reserved for a single measurement or fact of a specific trait, referenced to a single specimen  which is assigned to a specific taxon (i.e., one occurrence). This allows for storing repeated measurements on a single individual by linking the data of separate rows via a unique specimen ID. Also, multivariate trait measurements can be recorded in this format by linking multiple rows via a unique measurement identifier.  Longtable dataset purport multiple advantages for the data manipulation (e.g. filtering, subsetting and aggregating data), visualization (e.g. plot measured values by factor variable or taxon) and statistical modeling (e.g. ANOVA for testing difference of trait value by sex) [@wickham14]. They further draw from a defined set of columns, which simplifies merging of datasets. 

The latter format thus provides the highest resolution and is ideal for storing raw data of trait measurements in huge databases and for linking them via the respective unique identifiers to additional information, such as occurrenceID or locationID. Each row of the dataset can be interpreted as a statement of an 'entity x having a qualitative/quantitative feature y'. 
To comply with the ETS, datasets must be reformatted to the longtable format (Fig. 2*c*). 

<!-- Cat: here I guess that a figure showing few rows of a species x trait matrix and a long-format one with links to other information (metadata,..) woud help quite a lot. By experience some users are not be familiar with the interest of using the long-table format.-->
<!-- Malte: I agree (see my comment above), such a figure / would definitely help convince people (or at least not to lose them). -->

\begin{figure}[t]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig2.png}
\caption{Formats used for trait datasets. Identifiers relate the measurement to other layers of information on the specimen, the taxon or the trait measured. a) taxon-level traits are often published as a compiled species $\times$ traits matrix; b) measured data report measurements on the specimen-level, i.e. one physical occurrence of a taxon; c) a measurement longtable is a well defined and tidy data format, reporting one single measurement per row and relating it to an occurrence and taxon via unambiguous identifiers.   }

\end{figure}

## Common terms for trait data

In the subsequent sections we propose a set of terms that describe the columns composing a trait dataset, starting with the ones required for our minimal definition of 'core' trait-datasets proposed above. To enable a better standardization of nomenclatures and values, we add columns that contain 'harmonized' entries, which are linked to semantic ontologies and globally valid identifiers. 
We furthermore provide extensions to add information on the level of the occurrence, the measurement or fact, and the entire dataset (i.e. metadata). 

Additionally, a set of terms is proposed to facilitate the definition of traits in trait ontologies or a thesaurus of traits, consistent with semantic web standards (@ref, see below for further detail). 

The online reference glossary of terms is found at https://ecologicaltraitdata.github.io/TraitDataStandard/ and should be referred to in the metadata of any dataset that employs the ETS as "Schneider et al. 2017 Ecological Traitdata Standard v1.0, DOI: XXXX.xxxx, URL: https://ecologicaltraitdata.github.io/TraitDataStandard/v1.0/". In addition to the versioned online reference, the dataset should also cite this publication for an explanation of the rationale.  
Individual terms should be referenced via their global Identifier, which is hosted by the GFBio Terminology Service and also can be accessed programmatically via an API (@ref). Where our glossary refines or duplicates existing terms from other ontologies for biological data, like the Glossary of EOL (http://eol.org/info/516) and Darwin Core (http://rs.tdwg.org/dwc/terms/), we indicate this in the fields 'refines' or 'identical', respectively.

<!--
Table 1: Minimal set of terms for the traitdata standard. valueType can be of type character, factor, boolean or numerical. vocabulary is only needed for factorial trait data and defines the factor levels which are allowed. The identifier is a unique and globally valid URI. 

| Term | valueType | vocabulary | Identifier | Definition | 
|------|-----------|------------|------------|------------|
|scientificName | character | |http://fdschneider.de/bexis_traits/traitdatastandard.html#scientificname |Original character string provided as species name by the data owner (kept for reference only)| 
|taxonID | factor | | http://fdschneider.de/bexis_traits/traitdatastandard.html#taxonid | Verified name identifier of the species or subspecies (or higher taxon) for which this value was collected. Each entry should link to a published reference species list/taxonomy. Avoid synonyms.|
|taxonRank | factor | domain; kingdom; subkingdom; superhpylum [...] |http://fdschneider.de/bexis_traits/traitdatastandard.html#taxonrank | The taxonomic rank of the most specific name in the scientificName. Recommended best practice is to use a controlled vocabulary.|
|kingdom | factor | |http://fdschneider.de/bexis_traits/traitdatastandard.html#kingdom | The full scientific name of the kingdom in which the taxon is classified.| 
|traitName | character | |http://fdschneider.de/bexis_traits/traitdatastandard.html#traitname |Name of the trait as used by the data provider (kept for compatibility on provider side).|
|traitID|factor|see traitlist|http://fdschneider.de/bexis_traits/traitdatastandard.html#traitid ||
|traitValue|character||http://fdschneider.de/bexis_traits/traitdatastandard.html#traitvalue |Measured raw value or factor level for this species trait as measured and provided by the author.|
|traitUnit|character||http://fdschneider.de/bexis_traits/traitdatastandard.html#traitunit |Reports the unit that the author's raw data were measured in, if applicable (only for numeric values).|
|scientificNameStd|character|	|http://fdschneider.de/bexis_traits/traitdatastandard.html#scientificnamestd |Full scientific name (with authorship and date information if known).|
|traitNameStd|factor||http://fdschneider.de/bexis_traits/traitdatastandard.html#traitnamestd |Descriptive name of the trait reported. This should follow the controlled vocabulary of a thesaurus of traits available online or published along with the dataset.|
|traitValueStd|character||http://fdschneider.de/bexis_traits/traitdatastandard.html#traitvaluestd |Standardised measured value or factor level for this species trait. Numerical values must use the correct unit. Factor levels must be compliant with the thesaurus of traits.|
|traitUnitStd|character||http://fdschneider.de/bexis_traits/traitdatastandard.html#traitunitstd |The units associated with the measurementValue. The expected standard unit for this trait as defined in the traitlist/ thesaurus.|
|measurementID|	character||	http://fdschneider.de/bexis_traits/traitdatastandard.html#measurementid |A unique, dataset-specific identifier. Links multi-value trait measurements, e.g. x-y-z coordinates of a morphometric landmark, biochemical compound quantities for different chainlengths. In this case, the trait names must specifiy the sub-measurement, e.g. "landmark32_x", and must be specified in a reference trait list, given in the field "measurementMethod".|
|occurrenceID|character|	|http://fdschneider.de/bexis_traits/traitdatastandard.html#occurrenceid|A character string user defined by the data author. It is recommended best practice to use a globally unique identifier, if available, e.g. a museum ID referring to the individual specimen from which the data were measured. Usually does not apply for literature data.|
|warnings|character|	|	|Warnings from the R-script will be stored here, e.g. regarding a lack of match between the provided taxonID and the ontology, or the trait names or values, a mismatch in the units provided and the unit expected according to the trait table. User defined warnings and flags can be added as well, e.g. 'NOTUSE'.|

<!-- Nadja: Is this the kind of table you had in mind? -->

### core elements of a trait-dataset

The core content of a row in the trait-dataset is the reported measurement or fact for a single observation, which is composed of a value (`traitValue`) and -- for numeric values -- a standard unit (`traitUnit`). To associate the measurement or fact to a clear trait definition, a unique identifier (`traitID`) links each row to a descriptive trait name (`traitName`). 
 
The core data are also keeping a record of the scientific taxon for which the measurement or fact was obtained in the form of a descriptive taxon name (`scientificName`). 

For these core values, we highly recommend to use unambiguous and self-explanatory vocabularies for trait names and taxa. However, to have the data compatible with own databases or analytical code, it might be necessary to use abbreviations or identifiers instead. It is indispensible to relate the user-defined vocabulary to a consensus standard of taxon names or or look-up tables of traits. 

### relating to global taxonomy standards 

Taxonomic standardisation, with respect to ongoing renamings and refinemenets, is a challenging issue in its own right. To enable compatibility across datasets, conventions on trait definitions and taxon names are inevitable. We consider the fact that most authors have their own schemes for standardisation and may refer to different scientific community standards by allowing for a double record of both user-specific and standardised entries [as practiced in TRY; @kattge11]. This duplication of data enables continuity on the author and data owner side and quality checking and comparability on the data user side. We therefore propose to add a second set of columns that contain standardised entries of the taxon name (`scientificNameStd`) which map synonyms to an accepted species name according to a published taxonomic ontology (e.g. GBIF Backbone Terminology) and links to it using a globally unique identifier (`taxonID`). The field `taxonID` therefore should contain an public,  globally unique Uniform Resource Identifier (URI) that resolves to the precise taxon definition at the time of the description or measurement (e.g. for *Bellis perennis*, the taxonID would be  'https://www.gbif.org/species/3117424'). 
Keeping a reference to a kingdom and class (using columns `kingdom`, `class` as described by DWC and suggested for EOL Traitbank) will avoid ambiguities with taxonomic homonyms. 
Additionally, it is best practice to keep a record of the reported taxon rank (in column `taxonRank`) for filtering purposes for traits recorded at the family or genus level. This information can be used for trait inference of higher taxonomic resolution using hierarchical probabilistic matrix factorisation [@schrodt15]. 
Alternatively to expanding the number of columns, these data could be contained in a separate data file which relates to the core dataset via the field `ScientificName`. 

### relating data to trait definitions 

A similar approach to standardised trait names and harmonised reported values in compliance with published ontologies is advised. Here, column `traitNameStd` should report the accepted trait name in an online ontology, while the value and unit are reported in columns `traitValueStd` and `traitUnitStd`.
Similar as for `taxonID`, the field `traitID` should refer to globally unique identifiers (URIs) of trait definitions or a measurement methodology. 

<!-- The initiatives to standardise traits into consensus terminology are usually formed around a specific research question, methodology, or organism group and therefore vary in the traits that they consider important.  -->
To date, semantic ontologies exist for plant traits [e.g. TOP Plant Trait Ontology , @garnier17], soil invertebrates [T-SITA, @pey14], or biomedical traits of vertebrates [Vertebrate Trait Ontology, @park13] and more use cases (see OBO foundry for further examples). However, for most organism groups and research contexts we attest a deficit of public ontologies with publicly accessible URIs which would be used as `taxonID`. 

<!--### Defining own trait thesauri and ontologies-->

If no published trait definition is available that can be referenced, trait-datasets should be accompanied by a dataset-specific glossary of traits, or thesaurus. In its simplest form, a trait thesaurus would provide a "controlled vocabulary designed to clarify the definition and structuring of key terms and associated concepts in a specific discipline" [@laporte13; @garnier17]. 
This listing or table should at minimum provide a human readable, informative trait name (`traitName`) and define a unique dataset-specific identifier (`traitID`) as well as a short, unambiguous verbal definition (`traitDescription`) which may make use of standard terms provided in other Ontologies. As an Example,  the definition for 'fruit mass' in TOP reads: "the mass (PATO:mass), either fresh or dried, of a fruit (PO:fruit)", referring to Phenotypic Characeristics Ontology PATO and Planteome Plant Ontolgy, PO (http://top-thesaurus.org/annotationInfo?viz=1&&trait=Fruit_mass). 
The thesaurus should also constrain the legit factor levels (for categorical data, `factorLevels`) or expected standard units (`traitUnitStd` for numerical data). The type of values should be differentiated in the field `valueType` by specifying 'numerical', 'logical', 'integer', 'categorical' traits.

Relating the trait definition to methodological definitions on trait measurements help standardize the physical process of measurements. The Handbook for Measurement of traits [@moretti17] is a model example of such a methodological standardization approach.

Each trait definition may link to a broader or narrower term (`broaderTerm`, `narrowerTerm`), related terms (`relatedTerm`) or synonyms (`synonym`). For instance, the definition of 'femur length of first leg, left side' is narrower than 'femur length' which is narrower than 'leg trait' which is narrower than 'locomotion trait' (@ref semantic database methods). This extends the trait list into a semantic web resource, facilitates the classification of traits, and allows for cross-taxon comparative studies at the level of broader terms. (more detail on semantic web ontologies can be found in the online reference for the ETS at @refs...).

The existing thesauri and ontologies differ widely in terms of hierarchical depths and detail, as well as curation efforts and measures for peer-reviewed quality control. More compatibility of trait thesauri could greatly improve the possibilities for a comparison of trait data from different sources. 
We encourage hosting such semantic web resources developed for own projects as public resources, e.g. on OBO Foundry, the GFBio Terminology service, or on GitHub. We provide an example thesaurus, that has been developed to capture traits assessed within the German Biodiversity Exploratories project on Github (). 

<!-- Flo: For now removing this block on the steps necessary to create web ontologies of traits. The section above is covering the essentials and implies that further work is required. We might add this following section as a supplement or within the traitdata standard. -->
<!--

## towards a distributed semantic web ontology of traits

Thus, a trait thesaurus or ontology would assign trait names with A) an unambiguous definition of the trait and B) an expected format of measured values or reported facts, and might additionally provide C) semantic relationships for deriving a hierarchical or tree-based classification of traits. 

An ontology is going one step further by providing an explicit definition of a term including a formal model of the conceptual objects and relationships, or entities and qualities [@garnier17, @gruber95], ideally structured following the guidelines for a semantic web standard [ref]. 

Traits are not only defined in terms of their interpretation, but are ideally also standardised in terms of numerical units and, even more important, the use of factor levels. This is challenging, given the range of data types that fall within datasets of functional traits. 

Numerical values represent measurements of lengths, volumes, ratios, rates or timespans. Integer values may apply to count data (e.g. eggs per clutch). 
Binary data (encoded as 0 or 1) or logical data (coded as TRUE or FALSE) may apply to qualitative traits such as specific behaviour during mating (e.g. are territories defended) or specialisation to a given habitat (e.g. species restricted to relicts of primeval forests). Many traits are categorical and allow for a constrained set of factor levels, such as sex differences in wing morphology (both sexes winged, both sexes unwinged, only males winged, only females winged) or unconstrained entries such as color. Categorical traits often are ordinal, i.e. they have a logical sequence as in the case of life stages or hibernation stages, or habitat preference traits such as horizontal stratum use. 
Finally, there are specific formats of multivariate trait values, e.g. x-y-z coordinates of a landmark measured in a standardised 3D space [ref] or relative abundance of chain-lengths in biochemical compounds [].
To cope with this variety of data types, definitions should refer to other well-defined terms of existing ontologies that describe standard units, morphological body parts, protein characteristics (Protein Ontology) or chemical terms (e.g. the ChEBI, http://www.obofoundry.org/ontology/chebi.html).

Multiple approaches have spawned around the initialisation of trait databases. Such reference definitions are currently being published in methodological handbooks (Moretti et al , ) or lists of accepted traits (databases), or an online ontology or thesaurus (TOP, T-Sita, ), most advanced certainly for plant traits compiled in the Thesaurus of Plant characteristics (TOP, @garnier17).

Online ontologies extend into (machine readable) semantic web resources by providing a hierarchical classification of traits (TOP) or a relational tree of functional traits (e.g. TOP or T-SITA). . 

Each trait definition may link to a broader or narrower term. For instance, the definition of 'femur length of first leg, left side' is narrower than 'femur length' which is narrower than 'leg trait' which is narrower than 'locomotion trait'. (Ref semantic database methods)
This links traits of similar functional meaning and allows cross-taxon comparative studies at the level of broader terms. 

These systematic approaches to traits will be very useful for comparing similar traits measured in different taxa at higher hierarchical levels.
Furthermore, trait definitions may refer to related terms that describe a similar feature but with other means, or synonyms defined in other trait ontologies. By providing this semantic interlinkage of trait ontologies, a web of definitions is spun across the internet which allows researchers and search engines to relate independent trait measurements with each other.  

The benefit of such classifications will increase if open Application Programming Interfaces (APIs) provide a way to extract the definitions and higher-level trait hierarchies programmatically via software tools. To harmonize trait data across databases, future trait standard initiatives should provide this functionality. 
Online ontologies hosted with accredited ontology servers have the advantage of providing a persistent and direct link of the term on the internet (a *Uniform Resource Identifier*, URI).
Terminology portals or registries, such as the GFBio Terminology Service or ontobee, may provide a central host for trait ontologies. 

Thus, a trait thesaurus or ontology would assign trait names with A) an unambiguous definition of the trait and B) an expected format of measured values or reported facts, and might additionally provide C) semantic relationships for deriving a hierarchical or tree-based classification of traits. It additionally might relate to a more specific method of measurement.

The biggest challenge in community efforts of standardization of traits may be the investment in consensus building which leads to an acceptance and establishment of the standards. This effort is time consuming and lengthy, but is returns synergies and great scientific benefit by enabling a global synthesis, as the example of a global plant trait database TRY shows intricately [@diaz16]. 
-->
<!-- Nadja: I agree that this section is not needed in the paper, but it would certainly be nice to publish it in a supplement or with the traidata standard. Malte: Agreed. -->

### Metadata

Information about the authorship and ownership of the data and the terms of use should be considered when sharing and working with trait datasets. 

Individual measurements and facts may already be part of a larger set of traits or a separate trait database before they are added to a combined trait dataset. To retain the rights of the original data contributor, the field `rightsHolder` states the person or organization that owns or manages the rights to the data; `bibliographicCitation` states a bibliographic reference which should be cited when the data is used; and license specifies under which terms and conditions the data can be used, re-used and/or published. This information always applies to one single fact or measurement, further information on the larger dataset which originally contained this entry can be stored in `datasetID`, `datasetName`, `author` <!-- -->. These columns should hence give credit to the person who compiled the original dataset and signs responsible for the correct identification and reporting of the rights holder.
These information usually may be kept in the metadata of the dataset, but if datasets from different sources are merged, those should be referred to by a unique identifier (`datasetID`) or be reported as additional columns in the merged dataset (`author`, `license`, ...; see Dublin Core Metadata standards, Ref). 

<!-- 
(We conceptualize trait-datasets to  not include information about a species' interaction partners/associations of an organism, since they are -- to our understanding -- not within the scope of functional traits. However, such network data may be of interest to be combined with trait data. ) <!-- Malte: which can only be achieved if both types of datasets have been related to an ontology of species names. -->

## Extensions for additional data layers

Beyond measurement units or taxon rank resolution, further information might be available that are not recorded in the core data table, but are related to the trait type, the taxon, the individual or specimen, or to the reported fact, measurement or sampling event. 

To avoid redundancy and duplication of content, those information would usually be kept in seperate data sheets of a spreadsheet document. Examples are the place were a specimen has been sampled, the measurement method or statistical precision of the reported value, or the literature source from where a species value has been cited. A unique identifier would link to this other datasheet, encoding each individual occurrence of a species (`occurrenceID`), single measurements or reported facts (`measurementID`), locations of sampling (`locationID`) and sampling campaigns (`eventID`). 

For reasons of reproducibility and openness, data should not be uploaded in proprietary spreadsheet formats (like '.xlsx') but rather in comma-separated text files ('.csv' or '.txt'). EOL TraitBank proposes the use of Darwin Core Archives [http://eol.org/info/structured_data_archives, DwC-A; @robertson09] to upload such related data along with the core trait dataset, which is also the accepted format for GBIF [@braak11]. These are .zip archives containing data table txt-files along with a descriptive metadata file (in .xml format). Detailled descriptions and tools for validation can be found on the website of EOL (http://eol.org/info/cp_archives) and GBIF (http://tools.gbif.org/dwca-assistant/). 

Having separate data sheets linked by identifiers has the advantage of tidy datasets, avoiding duplication of information (@wickham14).
However, for standalone dataset publications on a hosting service without such file format recommendations (e.g. Figshare), data authors may want to avoid the expansion into multiple data tables and rather keep all columns in the same file, which is legitimate. 
It also may be safer to add such information directly as own columns of the data table to facilitate an analysis of cofactors and correlations further down the road. This way, if datasets of different source are merged, the information is readily available without the risk of breaking the reference to an external datasheet.
The trade-off is data-consistency and readability vs. avoidance of duplication. 

Other cases, where key data columns would be placed in the same table as the core data are traits assessed on a higher level of organisation, e.g. microbial functional traits assessed at the community level taken from a soil sample. Here, location or measurement information are in the primary focus of the investigation. 
A general definition, whether a column describes asset data or is part of the central dataset is ill advised. Therefore, our glossary of terms should be used to describe the scientific data according to the study context. 

The ETS provides two extensions of the namespace that should be used to describe these datasets:

- the `occurrence` extension contains information on the level of individual specimens, such as date and location and method of sampling and preservation, or physiological specifications of the phenotype, such as sex, life stage or age. 
- the `measurementOrFact` extension takes information at the level of single measurements or reported values, such as the original literature from where the value is cited, the method of measurement or statistical method of aggregation. 

Many terms of this section refine terms of the Darwin Core Standard and their own [MeasurementOrFact](http://rs.tdwg.org/dwc/terms/index.htm#measureindex) and [Occurrence](http://rs.tdwg.org/dwc/terms/index.htm#Occurrence) extensions  and EOL TraitBank use of those terms (http://eol.org/info/structured_data_archives). 

Some data-types may benefit from existing global identifiers for occurrence data, e.g. a GBIF URI or a museum collection code referencing the precise specimen at which the measurement was taken and that is publicly available.

 <!-- Malte: For readers not that familiar with trait data, the differences between measurement, occurrence and trait ID might not be that clear. It might be worth adding either a sentence explaining in an example what is what OR a tiny figure that shows how a specimen is sampled, processed, traits are measured etc. -->
 <!-- Nadja: I like the idea of having a figure explaining the different terms -->

### additional information on measurement or fact

As a high-level discrimination of the source of the measurement or fact, the Darwin Core Term `basisOfRecord` takes an entry about the type of trait data recorded: Were they taken by own measurement (distinguish "LivingSpecimen", "PreservedSpecimen", "FossilSpecimen") or taken from literature ("literatureData"), from an existing trait database ("traitDatabase"), or is it expert knowledge ("expertKnowledge"). It is highly recommended to provide further detail about the source in the column `basisOfRecordDescription`. 

To keep track of potential sources of noise or bias in measured data, the method of measurement (`measurementMethod`), the person conducting the measurement (`measurementDeterminedBy`), and the date at which the measurement was obtained (`measurementDeterminedDate`) are recorded. 
Authors would often report aggregate data of repeated or pooled measurements, e.g. by weighing multiple individuals simultaneously and calculating an average. In these cases, recording the number of individuals (`individualCount`) along with a dispersion measure (e.g. variance or standard deviation, `dispersion`) or range of values (e.g. min and max of values observed in the field `measurementValueMin`, `measurementValueMax`) is adviced. The field `statisticalMethod` names the method for data aggregation (e.g. mean or median) as well as the variation or range (e.g. reporting variance or standard deviation).

For data not obtained from own measurement, the field `references` provides a precise reference to the source of data (e.g. a book or existing database) or the authority of expert knowledge. 
For literature data, the original source might report trait values on the family or genus level, but the dataset author infers and reports the trait data at species level (e.g. if the entire genus reportedly shares the same trait value). To preserve this information, the column `measurementResolution` should report the taxon rank for which the reported value was originally assessed.

### additional information on observation context (occurrence) 

This category of terms contains further information about the individual specimen or occurrence that has been observed and measured. 

For both literature and measured data, trait values may be recorded for different sub-categories of individuals of a taxon to capture polymorphisms, for instance differentiated by sex or life stage. The template provides the fields `sex`, `lifeStage`, `age`, and `morphotype` for this distinction.

Seasonal variation of traits may be recored by assigning a date and time of sampling to the occurrence, using the fields `year`,	`month` and	`day`, depending on resolution. The field definitions of the Darwin Core Standard can be applied instead, to refer to a geological stratum, for instance. 
Sampling may be further specified using a unique identifier for the sampling event (`eventID`) which references to an external dataset. The record of a `samplingProtocol` may capture bias in samling methods. Further procedures and methods of preservation should be reported in `preparations`.  

To capture geographic variation of traits, a set of fields for georeferencing can put the observation into spatial and ecological context (`habitat`, 	`decimalLongitude`,	`decimalLatitude`, `elevation`,	`geodeticDatum`, `verbatimLocality`, `country`, `countryCode`). The field `locationID` may be used to reference the occurrence to a dataset-specific or global identifier. This allows the trait data to double as observation data, e.g. for upload to the GBIF database. 

<!-- ## The Biodiversity Exploratories Extensions and template 

Data will be uploaded to BExIS using the template to join data to the trait dataset while keeping full data authority: author information and ownership, access right management, as provided by BExIS. 

If applicable, trait values can be assigned to regions, plot IDs or sampling events which will then allow users to combine trait data with abundance biodiversity, ecosystem functioning, other observational data or the management regime on the sampled plots.

Metadata:
Each trait dataset uploaded to BExIS and added to the trait database will be accompanied with a metadata file containing additional information on data origin, method, authors and owners (in redundancy with primary data, see above) and possibly more detailed definition of the traits recorded. 
--> 
<!-- 
## information on attribution and permissions to use 

Individual measurements and facts will likely already belong to a larger set of traits or a separate trait database before they are added to a combined trait database. To retain the rights of the original data contributor, the field `rightsHolder` states the person or organisation who owns or manages the rights to the data; `bibliographicCitation` states a bibliographic reference which should be cited when the data is used; and `license` specifies under which terms and conditions the data can be used, re-used and/or published. 
This information always applies to one single fact or measurement, further information on the larger dataset which originally contained this entry can be stored in `datasetID`, `datasetName`,`authorLastname` and `authorFirstname`. These columns should hence give credit to the person who compiled the original dataset and signs responsible for the correct identification and reporting of the rights holder.



Finally, there is the set of information that applies to the entire trait-dataset, which classifies them as metadata. Since trait data are of great use for synthesis studies, information about how the data may be distributed, re-used and attributed to are of particular importance for trait datasets. Most researchers encourage re-use of their published datasets while making sure they are sufficiently credited. The use of permissive licenses for traitdata publications, such as Creative Commons Attribution or Creative Commons Zero/Public Domain release, has been established as the gold standard. 

<!-- Where is this data record available online? Source in the Measurements or Facts extension file is for a url to be displayed alongside the data point, so EOL visitors can click through to an online resource where the data originated. This might be a dataset stored at an online repository, a taxon page on a website, or any other online location.
What shall I cite if I use this data record in a publication? BibliographicCitation in the Measurements or Facts extension file is for the suggested citation format for a user of this dataset to include in a references list or bibliography. Even if the data record or dataset have never been published elsewhere, we recommend that you craft an appropriate citation so others can provide proper credit.
If the dataset is from a literature review, and one or more references are available for individual data points, these references should be recorded in the References extension and listed by their Reference IDs in the ReferenceID field of the Measurements or Facts extension file, see below.  -->


# Computational tools for producing compliant data

We provide an R package to assist in producing data compliant with the trait data standard proposed above. The package is being developed as an open source project on GitHub (https://github.com/fdschneider/traitdataform).

There are two major use cases for the package:

- preparation of own trait datasets for upload into public data bases, and
- harmonising trait datasets from different sources by moulding them into a unified format. 

The key function of the package is `as.traitdata()` which moulds a species-trait-matrix or occurrence table data into a measurement longtable format. It also maps column names into terms provided in the trait data standard. 

\begin{figure}[h]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig3.png}
\caption{Process chart of the functions provided within the R package.}

\end{figure}

Scientific taxon names are matched automatically to the GBIF Backbone Taxonomy and added as duplicate columns by calling the function `standardize.taxonomy()`

The `standardize.traits()` function matches user-provided trait names onto public trait ontologies and links it to an existing trait ontology or a user-provided table of trait definitions, i.e. a thesaurus of traits. The same function harmonises trait values into target units and legit factor levels. 

<!-- Add quick examples here -->

Several datasets that have been produced by these functions can easily be merged using the `rbind()` function of R base. 

The package functions comprise a toolchain where each function can be piped as an input into the next. A wrapper function `standardize()` applies all functions sequentially, making transferring and harmonising traitdata as simple as:

```r
library("traitdataform")
traitdataset1 <- standardize(read.csv("path/to/data.csv"),
            thesaurus = as.thesaurus("http://url.of/thesaurus.csv"),
            taxa = "name_correct", units = "mm")
```

For a detailed description on how to use the package see the package vignette (https://github.com/fdschneider/traitdataform/raw/master/docs/vignette.pdf). 

The package is under continuous open source development and invites participation in development, comments or bug reports via the Github Issue page (https://github.com/fdschneider/traitdataform/issues). 

Since the intention of the package is to facilitate upload of own data to hosting services but also simplify the access to published trait data, the package includes data objects that are compiled from datasets listed in our online spreadsheet that have been published in the public domain [@gossner15; @gossner15a; @ricklefs17; @vanderplas17]. 
To access data from public data-bases, the R-package 'traits' [@chamberlain17] contains functions to extract trait data via several open API interfaces including Birdlife, EOL TraitBank or BetyDB. The package  'TR8' provides similar access to plant traits from a list of databases (including LEDA, BiolFlor and Ellenberg Values)[@bocci15] and aggregates them into a species-trait matrix. However, none of these packages provides harmonization of traits into a unified scheme to facilitate further analysis.  
We provide a script (see online appendix and R-package vignette) to demonstrate the harmonization of these different data sources into our data standard. 

# Conclusion

In this paper, we propose a universal scheme for ecological trait data. 
We also provide a toolbox in R to transfer and harmonize data into this scheme. 

To date, existing initiatives and centralised infrastructures for standardisation and data  hosting are focused on a constrained organism group, ecosystem type or region. To make these heterogeneous sources of data more accessible and interlinkable, we propose a traitdata standard terminology and formatting scheme that can be applied in multiple contexts, for instance for building an own taxon-specific database or in user-side compilations of trait data from different sources. 

We further encourage the development of trait definitions in semantic ontologies. The biggest challenge in community efforts of standardisation of traits may be the investment in consensus building which leads to an acceptance and establishment of the standards. This effort is time consuming and lengthy, but it returns synergies and great scientific benefit by enabling a global synthesis, as the example of the global plant trait database TRY shows intricately [@diaz16]. 

These tools help to standardize trait datasets before upload to central as well as distributed data repositories. In times of increasing demand for open research data and international platforms for biodiversity data management, the development of meaningful ontologies for the standardization of biodiversity data is more than essential: defined ontologies enable researchers to relate published datasets to each other to achieve a greater synthesis, thereby opening the doors for new insights on global biodiversity patterns. In terms of data science, semantic web ontologies are a prerequisite for advanced big data mining using machine learning techniques.

Furthermore, the tools provided here can be applied to harmonise and combine data from heterogeneous sources. <!-- enables quality control on aggregated trait measures, verification procedures and opens the ground for a development of best practice principles for the filtering of raw measurements when creating species-level mean values --> We hope this aids research on the response of traits to environmental and anthropogenic drivers, synthesis on trait patterns on all ecological scales [@messier10], as well as multi-trophic trait analysis [@soliveres16; @allan15]. 

The trait data standard as well as the R package are published for open source community development. Contributions are invited and vital for the future development of the data standard and an extension and improvement of the R package. 


# Acknowledgements

Thanks to all respondents to an internal online survey on trait data for the Biodiversity Exploratories project and to Matthias Biber, Diana Bowler, Klaus Birkhofer, Runa Boeddinghaus, Martin Gossner, Catrin Westphal  for comments on the manuscript, the trait data standard and the R-package. 

We thank the managers of the three Exploratories, Kirsten Reichel-Jung, Juliane Vogt, Miriam Teuscher, and all former managers for their work in maintaining the plot and project
infrastructure; Christiane Fischer and Jule Mangels for giving
support through the central office, Michael Owonibi for
managing the central data base, and Markus Fischer, Eduard Linsenmair,
Dominik Hessenmöller, Daniel Prati, Ingo Schöning, François Buscot,
Ernst-Detlef Schulze, Wolfgang W. Weisser and the late Elisabeth Kalko
for their role in setting up the Biodiversity Exploratories project. 

The work has been partly funded by the DFG Priority Program
1374 "Infrastructure-Biodiversity-Exploratories" (DFG-Refno.). Field work
permits were issued by the responsible state environmental offices of
Baden-Württemberg, Thüringen, and Brandenburg (according to § 72
BbgNatSchG).

M.J. was supported by the German Research Foundation within the framework of the Jena Experiment (FOR 1451) and by the Swiss National Science Foundation.

# Authors' contributions

FDS, MJ, GP, AO, CP, and NKS conceived the ideas for the manuscript and critically revised the glossary of terms of the trait data standard; FDS led the writing of the manuscript; NKS authored the list of traits; FDS developed the R package. All authors contributed critically to the drafts and gave final approval for publication.

# References 

\footnotesize