---
title: "Towards an Ecological Trait-data Standard"
author:
- name: Florian D. Schneider \footnotemark[1]  \footnotemark[2]
- name: David Fichtmüller \footnotemark[5]
- name: Martin M. Gossner \footnotemark[8]
- name: Anton Güntsch \footnotemark[5]
- name: Malte Jochum \footnotemark[4]
- name: Birgitta König-Ries\footnotemark[6]
- name: Gaëtane Le Provost \footnotemark[2]
- name: Pete Manning \footnotemark[2]
- name: Andreas Ostrowski \footnotemark[6]
- name: Caterina Penone \footnotemark[4]
- name: Nadja K. Simons \footnotemark[7]
csl: amnat.csl
fontsize: 11pt
geometry: margin=1.4in
documentclass: scrartcl
output:
  pdf_document:
    fig_caption: yes
    fig_crop: no
    keep_tex: yes
    template: pretty.template
  html_document:
    fig_caption: yes
    fig_width: 4
    highlight: null
    keep_md: yes
    theme: united
  word_document:
    fig_caption: yes
    fig_width: 6
bibliography: div05.bib
---

<!--
Title Page should include:

    A concise and informative title. Do not include the authorities for taxonomic names.
    A list of all authors' names with names and addresses of Institutions.
    The name, address and e-mail address of the correspondence author.
    A running headline of not more than 45 characters.
--> 

\renewcommand{\thefootnote}{\fnsymbol{footnote}}

\footnotetext[1]{Corresponding author: florian.dirk.schneider@gmail.com, c/o Birgitta König-Ries
Friedrich-Schiller-Universität Jena, Institut f. Informatik, Ernst-Abbe-Platz 2, 07743 Jena
}
\footnotetext[2]{Senckenberg Biodiversity and Climate Research Centre (BiK-F) Frankfurt am Main, Germany}
\footnotetext[5]{Botanic Garden and Botanical Museum Berlin, Freie Universität Berlin, Berlin, Germany}
\footnotetext[8]{Forest Entomology, Swiss Federal Research Institute WSL, Birmensdorf, Switzerland} <!--  Zürcherstrasse 111, CH-8903 -->
\footnotetext[4]{Institute of Plant Sciences, University of Bern, Bern, Switzerland}
\footnotetext[6]{Department of Mathematics and Computer Science, Friedrich-Schiller-Universität Jena, Jena, Germany}
\footnotetext[7]{Department of Ecology and Ecosystem Management, Technische Universität München, Freising, Germany}

running title: Towards an Ecological Trait-data Standard

\newpage
\linenumbers
\renewcommand\linenumberfont{\normalfont\tiny}

# Abstract

<!-- 
The Abstract must not exceed 350 words and should list the main results and conclusions, using simple, factual, numbered statements:

Point 1: set the context for and purpose of the work;

Point 2: indicate the approach and methods;

Point 3: outline the main results;

Point 4: identify the conclusions and the wider implications.
-->

1. Trait-based approaches are widespread throughout ecological research, as they offer great potential to achieve a general understandnig of a wide range of ecological and evolutionary mechanisms. Accordingly, a wealth of trait data is available for many organism groups, but this data is underexploited due to a lack of standardisation and heterogeneity in data formats and definitions.
2. We review current initiatives and structures developed for standardising trait data and discuss the importance of standardisation for trait data hosted in distributed open-access repositories. 
3. In order to facilitate the standardisation and harmonisation of distributed trait datasets by data providers and data users, we propose a standardised vocabulary that can be used for storing and sharing ecological trait data. We discuss potential incentives and challenges for the wide adoption of such a standard by data-providers. 
4. The use of a standard vocabulary allows trait datasets from heterogeneous sources to be compiled more easily into databases and facilitates the creation of interfaces between software tools for trait-data handling and analysis. By aiding decentralised trait-data standardisation, our data standard may facilitate the use of trait data for a broader ecological research community and enable global syntheses across a wide range of taxa and ecosystems. 

# Key-words: 

functional ecology, species traits, semantic web, ontologies, data standard, ecoinformatics

\newpage

\tiny

Term | Definition 
----- | ---------
Term | A word that names or labels a particular *concept* as part of the specialised vocabulary of a field. 
Concept | An idea, notion or object that is made explicit in an information context by a *term* definition, and referenced to a *URI* or other accessible reference (https://www.w3.org/TR/skos-reference/)
Controlled vocabulary | A list of *terms* that gives all valid consensus terms for a prarticular context, while no unlisted entries are accepted
Terminology | The  body of *terms* and *concepts* used with a particular application in a subject of study, usually formalised in a *thesaurus* or *ontology*
Data standard | A published set of instructions and *terminologies* for storing and exchanging data content of a particular type (e.g. trait data), that is recognised by a large proportion of members of the application context
Thesaurus | *Controlled vocabulary* that provides key *terms* with their associated *concepts* and relations for a specific field or domain of interest [@laporte13]; e.g. may define a hierarchy of broader or narrower terms
Ontology | *Controlled vocabulary* that defines *concepts* in relation to each other in a formal language, e.g. provides cross-references and semantic meaning; a semantic model of the objects and their relationships in a domain of interest [@gruber95]; commonly published in OWL format for machine readability.
Semantic web | An extension of the World Wide Web that aims for machine-readable meaning of information via well-defined *data standards*, *ontologies* and exchange protocols [@berners-lee01]; the World Wide Web Consortium (W3C) defines standards, i.e. specifications of protocols and technologies for the semantic web (http://www.w3.org/standards/semanticweb/)
Dataset | A set of measurements and observations, often stored in a data-table and originating from a single experimental set-up or study context; can be considered as being internally homogeneous across all data entries
Data table | A two-dimensional spread-sheet containing data organised in rows and columns; in most cases these data are considered 'static', i.e. they are not altered or filtered across time
Database | A suite of data compiled from multiple *datasets*, i.e. from multiple study contexts or observation types; may take the form of a single two-dimensional *data tables*, but mostly is organised as multiple data tables linked via *identifiers* into *relational databases*; may be built using a specific database software or language (e.g. SQL)
Relational database | Two or more *data tables* that are related by common information contained in one or more columns; common information is usually labelled by *identifiers* (IDs)
Online database | A *database* that is made accessible on the internet; offering forms for filtering and downloading subsets of the data; some online databases offer access via a *webservice* and an *API* that can be addressed computationally 
Identifier (ID) | A unique label that relates row entries within and across *datasets*; may be used to connect multiple *data tables* into a *relational database*; can be user-specific or, in form of a *URI*, point to a globally valid *ontology* or *thesaurus*  
Uniform Resource Identifier (URI) | An unambiguous pointer to a unique resource on the internet; used to refer to single terms of a *thesaurus* or *ontology*; Example: 'http://t-sita.cesab.org/BETSI_vizInfo.jsp?trait=Body_length'
Web Service | A service to access *online databases* directly and computationally, i.e. by calls from a software tool; returns content of the database according to the request in different formats and *data standards*
Application Programming Interface (API) |  A set of clearly defined methods of communication between  software components, e.g. client software and the *web service* of an *online database*; APIs are usually documented on the website of the service provider
Online Portal | A website designed as a platform for the exchange of information, e.g. trait data; a portal may include a communication forum, data upload forms, a database access point, and advanced user management for data access
File repository | A short-term storage of datasets or long-term archiving on *file-hosting services*; online repositories make data available for public access, provide *metadata* and (not always) facilitate citations via persistent identifiers, e.g. DOIs (Digital Object Identifiers) 
File-hosting service | An online platform that hosts *datasets* or entire *file repositories* and provides access to a wide audience on the internet; Examples: Figshare.com, Dryad (datadryad.org), Researchgate.net, or Zenodo.org
Metadata | Data documentation of the higher level information or instructions; describe the content, context, quality, structure, provenance and accessibility of a data object [@michener97]
Darwin Core Standard (DwC) | Body  of *terminologies* providing terms intended to facilitate the sharing of information about biological diversity (http://rs.tdwg.org/dwc/)
Darwin Core Archive (DwC-A) | A file archive, or *repository* structure, that contains *metadata* (specified e.g. using Ecological Metadata Language, EML) and primary data combined into a *relational database* via *identifier* columns [@gbif17,  http://tools.gbif.org/dwca-assistant/]
Method handbook | A listing of consensus methodology that is to be applied to acquire a particular measure, thus formalising the precise *concepts* of measures; Example: Handbook of protocols for standardized measurement of terrestrial invertebrate functional traits @moretti17
Occurrence | The observation context of a single entity of a taxon, i.e.  an individual observed at a particular place and time (http://rs.tdwg.org/dwc/terms/Organism); Sometimes used as synonym of 'observation' in data management context

Table: Glossary of terms from the biodiversity data-management context as they are used in this paper; draws from @garnier17. 

<!--
Measurement |  
Aggregated Value |
Value | 
Fact | 
-->

\normalsize

\newpage
<!-- CP:  I think we should develop more on what are the current problems/limitations (quickly because it’s then expanded in the paper) so it becomes evident that the standard+package fill a real need.

That’s the structure I try to follow:
1. Traits are important in ecology, evolution and conservation (i.e.  It is important to mention that traits are important in research and give some examples of questions that traits can help to solve but maybe not spend too long on it (most people know that they are important).

2. To increase the scale of studies (spatial, temporal, taxonomical) we need to put together trait data from different sources. When we do this we can answer new questions and find generalities. So there is a growing interest in assembling trait data from different sources and taxa (e.g. give the number of papers using TRY).

3. There is a problem of standardization that makes it difficult to match trait datasets from different sources. Different reasons: no traits definitions, no common structure.

4. One solution are databases. But these are costly and depend on the will of few persons/funding. They also cover few taxonomic groups. Matching between different databases is still difficult because of different definitions and structures. Most data are still uploaded separately.
We need a solution to make their use easy. 
-->
<!-- GLP: 
 Add a general paragraph on the fact that there have been some initiatives for the development of trait data base for plants (e.g. TRY), why is it important (provide quick access to plant trait values, allowing comparisons even between ecosystems and biomes), now the development of trait data base for other organisms (from microbes to mammals). Say also something on the development of general protocols for measuring traits (e.g. Moretti et al. 2016; Freschet et al. 2017) ?

Importance of data standardization. This is the main idea here and so I wonder if some idea of the first point (Need for data standardization) should not be merge here with the last paragraph of the introduction ? As there is also some repetitions between those 2 parts.

I wonder if you should speak about ‘Functional traits’here or just Traits ? As you are not only considered Functional traits in this paper (e.g. life history traits) ? Maybe remove the definition no ? 
-->
<!-- PM: Thuis is quite complex- e.g. ‘taxonomic impediment’etc. I actually think you can simplify this and built these references into the more general text above by adding a copule of more examples. I don’t think you need to say much about the merits of the trait based approach here- it’s covered elsewhere. 

Huge number of references here- I think you can strip it down to a few key ones, and reviews. Overall I think this section is quite a lot more complex than it needs to be – I think all you need to say is that it’s very popular, general and mechanistic and so adopted throughout eco and evo, with no sign of slowing down- the TRY database download stats can provide a quantitative measure of this- see their site. 
-->

# Introduction
<!-- 1. Traits are important in ecology, evolution and conservation -->

Functional traits are phenotypic (i.e. morphological, physiological, behavioral) characteristics that are related to the fitness and performance of an organism [@mcgill06; @violle07]. 
Recent years have seen a proliferation of trait-based research in a wide range of fields and have been used to understand the evolutionary basis of individual-level properties [@salguero-gomez16], global patterns of biodiversity [@diaz16] and the relationship between ecosystem functions and the functional composition of species assemblages [@mouillot13; @perovic15; @bello10]. This research provides the mechanistic framework for linking climate change or anthropogenic land use to biodiversity and its related functions [@diaz11; @lavorel12; @allan15].
Species traits have been suggested as indivator variable for monitoring  ecosystem health at the individual level [@kissling18].  
 <!-- ;@wood15;  @diaz11; @jennings09; @hopfenmuller14; @mouillot13;  @vandewalle10; @perovic15-->
Using traits is also a promising means of bypassing taxonomic impediment, i.e. the fact that most species are yet undescribed and little is known of their interactions with other organisms and their environment. This is because functional traits allow us to infer the ecological role of organisms  from their apparent features, regardless of their taxonomic identity [@leprovost17; @duarte11; @schrodt15].

<!-- Comparative studies of traits inform researchers about physiological and evolutionary trade-offs, and help identifying economic strategies of organisms beyond taxonomy [@menezes10; @forister15;  @siefert15; @gross09; @wright04]. Inferring an unknown functional role or ecological function of individuals from their apparent features [@duarte11; @schrodt15] is a promising venue to bypass taxonomic impediment, i.e. the fact that a majority of species are yet undescribed and little is known of their interactions with the environment and other organisms. 
Mutual 'trait-matching' of co-evolved organisms enables the inference of ecological interactions from observed features of live specimens [ @kalinkat13; @dehling14a] which may be used to inform ecological networks [@weiner14; @miranda13; @digel11; @schleuning15] and parametrise ecological population-dynamic models [@schneider12; @kalinkat13]. These analyses are also a promising venue for a multitrophic community ecology aiming to unravel dependencies and synchrony in trait spaces between interacting functional groups [@soliveres16]. <!-- GLP: Maybe not the best example as they are not looking at trait diversty and species interactions ? 
Put instead:
Gravel et al. 2016 The meaning of functional trait composition of food webs for ecosystem functioning
And Coux et al. 2016 Linking species functional roles to their network roles
? -->
<!--
Since traits are so closely related to ecological mechanisms, they have proven useful for scaling individual level physiology and behaviour to functional diversity  of communities [@laliberte10a; @petchey09; @villeger08a; @hillerislambers12;@diaz98; @lavorel02], processes at the ecosystem scale [@fontaine05; @simons16; @gossner16; @moretti13;  @kuhsel15;  @birkhofer14], as well as macro-patterns of biodiversity at the global scale [@diaz16; e.g. @reich97]. <!-- Traits can be used to predict extinction risk [@fritz10; @pearson14] and susceptibility to environmental toxins [@rubach11].-->
<!--
Most importantly, the trait framework relates key ecosystem services for human well-being [@diaz11; @lavorel12; @allan15] back to changes in biodiversity in consequence to global change or local anthropogenic land use [ @diaz07; @gossner16; @mangels17; @jennings09; @simons16;  @mouillot13;  @vandewalle10;  @birkhofer15;@gamez-virues15; @birkhofer17]. 
--> 

<!-- 2. trait databases have been developed, with standard structure and terms --> 
Despite the importance of functional trait data, fully exploiting the potential of trait-based approaches relies heavily on the broad availability and compatibility of trait data to achieve sufficient taxonomic and regional coverage, both of present-day taxa as well as in evolutionary deep-time. 
However, the heterogeneity of data arising from different research contexts render trait data extremely heterogeneous and make the task of data compilation time-consuming and error-prone.
<!-- i.e. differences in taxonomic nomenclature and resolution (e.g. reported on species level or aggregated on higher taxonomical orders), the scale and place of the study context, or the accurracy of the methodology applied in measurements--> 
To date, trait data have traditionally been harmonised and compiled into centralised databases only for specific organism groups and regional scope, often centred around particular research questions [e.g. PanTHERIA, @jones09; TRY, @kattge11; AmphiBio, @oliveira17]. Less well-studied taxa and specialised research questions lack the resources for such an endeavour.
Besides initiatives aiming at assembling data, tools to enable the compatibility of data across databases are being developed. These include software to access trait data from the internet [e.g. @chamberlain17; @ankenbrand18], semantic-web standards [@wieczorek12; @page08] and thesauri of standard terms [@walls12; @garnier17].

<!-- 4. distributed trait data must be integrated --> 
Meanwhile, open and reproducible science has become mainstream: publication of research data without access restrictions, with structured metadata and in accordance to data standards to enable their reuse, has become the declared goal of an open biodiversity knowledge management (http://www.bouchoutdeclaration.org/) and is increasingly demanded by journals and public research funding agencies [@emerson15; @allianceofgermanscienceorganisations10; @royalsocietysciencepolicycentre12; @swan12; @allison15]. 
As a result, an increasing number of individual research projects publish their primary data on general-purpose file hosting services (e.g. Figshare.com, see Table 1), where no data standards areenforced upon the uploaded material [@wilkinson16]. 
It is thus likely that trait data will become increasingly available, but a lack of data and metadata standardisation will hamper the efficient reuse and synthesis of published datasets. 

In this paper, we review existing trait databases and online portals, and initiatives for data standardisation from the pragmatic view of data providers, data curators and data users, as well as database managers. We discuss current efforts to making trait data visible, accessible, interoperable and re-useable in downstream data analysis, as demanded by the FAIR guiding principles for scientific data [@wilkinson16]. 
Furthermore, we show how the current deficit in the standardisation of primary data hampers the implementation of interoperability and reuse of trait data. 
Based on these considerations, we propose a versatile vocabulary for describing trait datasets that builds upon, and is compatible with , existing terminology standards for biodiversity data. 
Since this standard vocabulary relies on the adoption of a broad research community, we discuss incentives for its use and lay out mechanisms for a future consensus-building and community development towards an accessible ecological trait data standard.
<!--Finally, we present an R package that assists the harmonisation of trait data from distributed sources. -->
<!-- FDS: Why 'ecological' trait data? what is special about this subject? Environmental context, community context, space and time, attributes of functional role in the ecosystem and energetic demands, interactions (not covered right now?), correlation and economic trade-offs in trait evolution -->

# Initiatives for trait-data standardisation

The need for standardising trait-data arises from the prospective gain of compiling heterogeneous trait-datasets for data synthesis. 
Often, the scientific scope and focus differs between data providers measuring and assessing the trait data in the first place and data users who re-use published data for a broader synthesis application. Data curators are taking up the task of providing compiled and harmonised datasets or databases. Data managers are concerned with the development of complex digital infrastructures for handling and analysing large amounts of data. 
These idealised roles of researchers cooperatively are dealing with trait-data standardisation throughout the data life-cycle. In this chapter, we review four types of initiatives that are of relevance for trait-data standardisation (see Glossary in Table 1 for italicised terms):

1. Initiatives that provide trait *datasets* which have been assembled out of a particular research interest, either by measurement or collated from the literature.
2. Initiatives that aim to harmonise trait data from the literature or from direct measurements into trait *databases* and make those data widely available. 
3. Initiatives that aim at the standardisation and development of consensus measurement methods and definitions for traits and provide standard *terminologies* in the form of *thesauri* and *ontologies*.
4. Initiatives that aim to leverage *relational database* structures and *semantic web* technology to link trait data to a wider set of biodiversity data. 

We consider these initiatives separately although they are often developed in conjunction to serve a particular database project, such as the TRY plant database [@kattge11; @kattge11a] and the  Thesaurus of Plant Traits [TOP; @garnier17]. 
We show how the degree of trait-data standardisation in existing datasets is highly variable and which tools and standards are currently applied to achieve harmonisation of data from multiple, distributed sources. 
The objective of this review is to raise awareness of the generic structure of trait data and aid researchers in how to share and publish their own datasets in an appropriate form. 

## Trait datasets 

In the field of comparative biology, morphological traits, such as traits  related to flower shape, leaf and stem structures for plants or wing and beak measurements for birds, as well as life-history traits such as Ellenberg values for plants or physiological and reproductive traits for animals (e.g. reproductive traits, feeding biology, dispersal, metabolic rates and body size) have been assessed for decades and have been published in regular journal articles or books. 
With the rise of ecological trait-based research, measurements and information available from species descriptions have been compiled into project-specific datasets that typically comprise a local set of taxa and a focal set of traits. 
A plethora of such static datasets has been published alongside scientific articles, or as standalone data publications [see @kleyer08 for a review on plant data; for animal data, see e.g., @gossner15; see Supplementary Material A, Table A1].<!--  @sokol11;  @renner14; @vanderplas17; @price13; -->

Today, the online publication of such data is greatly facilitated by *file hosting services* (e.g. Figshare.com, see Table 1), which warrant long-term accessibility and citeability via DOIs, and govern data sharing via licensing.
These platforms offer publicly accessible *repositories* at low-cost or for free, which makes them attractive for small and intermediate-sized research projects that cannot dedicate extra resources for data management. 
However, although open for download, the trait datasets on data repositories might be stored in proprietary (e.g. .xlsx, .docx) or binary (e.g. .pdf) data formats which make extraction tedious and dependent on commercial software, putting the long-term and open accessibility of these data at risk. 
Most importantly, these platforms enable public hosting of data with very low quality thresholds regarding *metadata* documentation and data standardisation. As a result, published datasets greatly vary in structure, nomenclature and metadata quality. 

For trait data, there are common issues arising from the variability of data structures. 
For instance, the column descriptions and terminology applied to taxa and traits are mostly project specific and rarely chosen to allow translation into larger database initiatives. Furthermore, metadata varies in its level of detail, e.g. for documenting descriptions of variables, measurement procedures or sampling context [@kattge11a]. 
In terms of structure, trait data usually are reported in a species$\times$traits wide-table format. In this intuitive *data table*, each row contains a species (or taxon) for which multiple traits are reported in columns. <!-- Also, sources of  measurement bias are documented (methodology, observer ID, n of observations).  -->
Similarly, when reporting raw data, researchers place observations of individual organisms (i.e. the *occurrences*) in rows with multiple trait measurements applied to the same individual across multiple columns. 
Co-variates on the taxon, the individual specimen (e.g. sex or lifestage) or context of observation (e.g. time and place of sampling) would be placed in additional columns, and would further expand the two-dimensional data table. The resolution or scope of these co-variates varies greatly depending on the research question and observation context. 
Variability in the number and meaning of columns in these data tables requires tedious manual adjustments when merging multiple datasets [@wickham14]. A task that is taken up by data curators, who compile heterogeneous data into comprehensive databases (see next section). 

<!--To remedy these problems, data management initiatives that arise from scientific communities and projects, like the KNB (the Knowledge Network for Biocomplexity; https://knb.ecoinformatics.org/) or GFBio [the German Federation on for Biological Data; gfbio.org; @diepenbroek14], offer means for small projects to meet the high scientific criteria of data management.-->
<!--A global overview of existing trait data for all taxa and trait types is difficult to obtain. Therefore, in an attempt to collate a list of existing distributed datasets, we initiated a living spreadsheet (https://goo.gl/QxzfHy) which lists published trait datasets, their regional and taxonomic focus, the number and scope of traits covered, their location on the internet and the terms of use (see Appendix A for an excerpt of this list). We invite data owners and users to add further trait datasets to this spreadsheet.  -->
<!-- FDS: this might be a task for community  building and decentralised registries, as discussed on the OpenTraits platform, e.g. FENNEC or MAD package do that, or a registry for earning badges of data quality would achieve that as a by-product. I think we should not aim for solving this registry problem here. -->

## Database initiatives

<!-- community database backend, but static publication --> 
In the past two decades, many distributed trait datasets have been aggregated and harmonised into greater collections with particular taxonomic or regional focus [e.g. @kleyer08; @oliveira17, see Supplementary Material A, Table A1].
While mostly concerned with issues of heterogeneity in units or factor levels, and aiming for high taxonomic coverage, few of these *databases* apply a standardised terminology for taxa or traits that would allow them to be efficiently linked to other databases.<!--and aiming for taxonomic completeness, the products of these initiatives employ a project specific structure and vocabulary. -->
Documentation of metadata and methodology differs in the level of detail, depending on the research focus. 
Just as the individual datasets described above, many of these databases are published as static data tables on journal websites or open-access file hosting platforms, but they may be updated irregularly.

As they deal with much larger amounts of data, initiatives that form around natural history museum collections are traditionally more concerned with standardisation. 
The amount of morphological measurements data extracted from museum collections and herbaria is likely to skyrocket in the near future due to digitisation efforts [@vollmar10; @blagoderov12] supported by new technology for scanning and pattern recognition [@younis18; @mantle12; @strobel18] and citizen science initiatives (e.g. www.markmybird.org), 
For example, the VertNet database compiled and harmonised large quantities of vertebrate trait data from collections [@guralnick16]. The resulting data are published as versioned data tables which are updated as new data sources become available (vertnet.org). 

<!-- database backend, selective download, user access management --> 
Specialised *online portals* have been created to attract data submissions from a defined research field and take care of data harmonisation, thereby greatly facilitating data synthesis.
<!--For example, traits on crop plants have been collected in agronomy databases like  BetyDB [https://www.betydb.org/; @lebauer17].-->
For example, by aiming for a universal framework for plant traits, the TRY database [@kattge11]  attracted more data submissions and downloads than any other trait-data platform. The *online database* enables selective data download and user permission and rights management. As a community effort, TRY serves as a network for consensus building on trait definitions [@garnier17] and measurement methodology [@perez-harguindeguy13] (see next section). 
For animal trait data, however, a single unified portal and harmonising scheme is still lacking. The fact that animals span multiple trophic levels and possess diverse body structures makes this a more complex task than for plants [@moretti17]. 

Nonetheless, initiatives for particular groups of animals do exist: examples are the BETSI database on soil invertebrate traits [http://betsi.cesab.org/; @pey14], the Carabids.org web portal  (http://www.carabids.org/), or the Coral Trait Database [@madin16, see Supplementary Material A, Table A1]. 
The role of online portals and database initiatives in standardising data and making them more accessible is paramount. 
Trait-data portals incentivise data submissions by offering increased data visibility and usage, while providing data-use policies that secure author attribution and, potentially, co-authorship of associated articles. However, maintaining centralised database infrastructures is costly and requires long-term funding [@bach12]. 

## Thesauri and Ontologies for traits

<!--Microbial ecologists also make frequent use of trait-based approaches to assess genomic function and describe functional diversity at the community level [@krause14; @fierer14;  @fierer12]. Here, 'operational taxonomic units' (OTUs) are derived from metagenomic analysis [@torsvik02; @langille13]. Databases are also used to interpret OTUs in terms of their functional role [e.g. the KEGG orthology, @kanehisa12]. -->

A major challenge in trait-data standardisation is the lack of widely accepted and unambiguous trait definitions [@williams06; @kissling18]. Previous standard definitions of trait *concepts* range from listings of selected definitions in *terminologies*, over well-defined *method handbooks* and comprehensive *thesauri*, to formalized definitions of trait concepts in *ontologies*. 
The initiatives behind method handbooks, thesauri and ontologies are primarily concerned with building community consensus for trait definitions. 

Very general classes of traits are defined within the list of GeoBON Essential Biodiversity Variables [@pereira13; @kissling18] aiming for a list of functional indicators for ecosystem health.
Assigning a more detailed and unambiguous methodological protocol for a trait, including the units to use or the ordinal or factor levels to be assigned, is essential for standardising its measurement process. Efforts to develop handbooks for measurement protocols provide such a methodological standardisation for plants  [@cornelissen03; @perez-harguindeguy13] or invertebrates [@moretti17], but are of limited use in harmonising trait data that pre-date or ignore this standard [@kattge11a].

A thesaurus provides a "controlled vocabulary designed to clarify the definition and structuring of key terms and associated concepts in a specific discipline" [@laporte13; @garnier17]. 
In thesauri, hierarchies of concepts can be formalized by linking each term to broader or narrower terms, or to synonyms. For example, the definition of 'femur length of first leg, left side' is narrower than 'femur length' which is narrower than 'leg trait' which is narrower than 'locomotion trait'. 
Being publicly available, it is also possible to refer to these defined terms via globally unique *Uniform Resource Identifiers (URIs)*. For example, a measurement of fruit mass could be linked to the definition of the term within the TOP Thesaurus of plant traits [@garnier17] via its URI ‘http://top-thesaurus.org/annotationInfo?viz=1&&trait=Fruit_mass’.

Ontologies link defined terms by formally describing the relationships between them and providing a semantic model of the concepts used within a domain of research, with the objective of enabling the computational interpretation of data [@williams06; @walls12; @walls14]. 
Ontologies define terms based on other well-defined terms from published ontologies. The Planteome Trait Ontology (TO) definition of the concept 'seed size' contains references to other globally defined terms: "A seed morphology trait (TO:0000184) which is the size of a seed (PO:0009010)." 
Furthermore, trait definitions may refer to related terms or synonyms defined in other trait ontologies or other scientific ontologies, like units as defined by the Units of Measurement Ontology [@gkoutos12]. 
By providing ontologies in a formalized file format, like OWL or RDF, a machine-readable web of definitions is spun across the Internet allowing researchers and search engines to relate independent trait measurements with each other and connect them to the wider *semantic web* of online data [@berners-lee01; @page08; @williams06; @walls12]. 

Comprehensive trait thesauri have been developed in the TOP Thesaurus of plant traits, which is employed in the TRY database [@garnier17], and in the Thesaurus for Soil Invertebrate Trait-based Approaches [T-SITA, http://t-sita.cesab.org/, @pey14]. 
Ontologies of trait definitions have been developed for plants [e.g. the Plant Ontology, @jaiswal05; the Flora Phenotype Ontology, @hoehndorf16], as well as for animals [e.g. the Hymenoptera Anatomy Ontology, @yoder10; the vertebrate trait ontology, @park13]. 
The existing thesauri and ontologies for traits differ widely in terms of hierarchical depth and detail, as well as in curation efforts and measures for peer-reviewed quality control. 
<!--Thus, in its simplest form,  a trait thesaurus would assign trait names with A) an unambiguous definition of the trait and B) an expected format (e.g. units or legit factor levels) of measured values or reported facts. It might additionally provide semantic relationships between terms for deriving a hierarchical or tree-based classification of traits. --><!-- BKR: Isn’t that the same? Or can you have multiple parents for one node? FDS: agree. simplify phrasing. -->
Meta-ontology initiatives, like OBO Foundry [@smith07, http://www.obofoundry.org/], Ontobee (http://www.ontobee.org/), Bioportal (https://bioportal.bioontology.org/), or the GFBio Terminology service [@karam16, https://terminologies.gfbio.org/], provide centralised hosting for advanced trait ontologies and offer harmonized web services computational access. 

To conclude, there is already a suite of globally available thesauri and ontologies for traits available. However, definitions in some domains are better covered than others and the interlinkage and accesibility of ontologies can be much improved to fulfil semantic web standards. 
While these defined vocabularies are widely used in biodiversity data management, distributed trait-data of smaller projects published in general-purpose file servers make little use of them. 

## Trait-data structures 

While trait thesauri and ontologies typically define traits for focal groups of organisms, they do not specify the  format or structure in which trait data should be stored and linked to further standard terminologies, such as standard taxonomy nomenclatures. 

To make sense of trait data in the context of more general databases, a consensus definition of trait data has been provided by  @garnier17: 
The define trait data to follow an entity-quality model (EQ), where a trait is defined as 'an entity having a quality'. 
More specifically, a trait dataset contains information on quantitative *measurements* or qualitative *facts* (i.e. trait values) describing the physical phenotypic characteristics relating to fitness and performance (i.e. traits) observed on a biological entity (i.e. an individual specimen, or its parts) that can be assigned to a biological taxon (i.e. a species or higher-level taxon). 
The entity at which the trait was assessed may differ in organisational scale -- depending on the scientific question -- and could be a sub-sample or bodypart, an individual specimen, an entire species or a higher-level taxon (e.g. a genus).  

These relationships between a trait measurement and an individual organism as an entity of a particular taxon have been formalised in the schema for biological collection records [ABCD Schema; @holetschek12] and the Darwin Core Standard for biodiversity data [DwC; @wieczorek12]. 
These terminology frameworks specify terms and classes to describe the general structure of biodiversity databases. The terms may be used for  defining columns in data tables that contain measurement values, units and factor levels, taxon names, variables such as sex or life stage, ancillary information of time and date of observation, and methodological details. The terminologies provided by these standards are quite universal and would even cover most use cases of trait data. 
A suite of terminology extensions links to and expands the capacities of DwC [@wieczorek12].  
For example, the Global Biodiversity Information Facility (GBIF, www.gbif.org) refers to DwC terms for defining its database content. 

<!--  DF: am not sure, where to best write this comment, but I think it belongs somewhere in this chapter. The Trait Data Standard is a standard on how to store trait data measurements. In this chapter you mention what databases for trait data there are (usually specific for certain taxa) and what ontologies there for the different traits, however it is missing the section on how trait data measurements have been and can be stored prior to your standard. For instance the ABCD standard has a huge section for MeasurementsAndFacts. And while it certainly is not primarily made to capture trait measurements, it can certainly be used for it (at least on a unit level, combining measurements for higher taxa is not possible). Darwin Core has similar, though not as extensive, fields for measurements, as they were modelled based on ABCD. While I agree that your standard is useful and necessary, it certainly is not the first standardized way to describe trait measurements.  -->

Specifically designed for plant traits, @kattge11a proposed a generic database structure that covers most potential use cases of trait-based ecology. This data structure is built around a central data table that contains observations of individual plants (entities) linked to several measurements of traits (qualities) via identifiers. The observations are also linked to a taxonomy and ancillary descriptors of the observation context, like location or experimental treatment. 
The structure of @kattge11a can be mapped to the  Extensible Observation Ontology [OBOE, @madin07; @schildhauer16], which formalizes observations andd measurements of traits into a machine-readable ontology, thus being easily integrated into larger database management systems. 

In a similar vein, the Encyclopedia of Life (EOL) project has proposed the database framework TraitBank [@parr16] for major physiological and life-history traits of all kingdoms of life, which is to date the most general approach of an integrated structure for trait data.
The framework employs established terms provided by the DwC, relates trait definitions to ontologies for phenotypic or anatomical terms, and maps taxa to global identifiers in taxonomic hierarchies of name service providers to capture synonyms, misspellings and controversies [@parr16, http://eol.org/info/cp_archives]. 
Additional layers of information cover bibliographic references, multimedia archives and ecological interactions. 
TraitBank invites data submissions to the EOL database in a structured *Darwin Core Archive* [DwC-A, @robertson09a]. 

All of these structures make  use of stable URIs to refer to taxon concepts. The difficulties with keeping taxonomic references intact along with continuous changes in taxonomy consensus are a central challenge of biodiversity data management and are beyond the scope of this review [@franz16]. Initiatives that aim at providing a stable reference for taxon concepts are for instance the EOL Catalogue of Life [http://www.catalogueoflife.org/, @roskov18], the GBIF Backbone Taxonomy [@gbifsecretariat17], or the EDIT Platform for Cybertaxonomy (https://cybertaxonomy.eu/). 


## Closing gaps in the trait-data lifecycle

In sum, we attest a gap in the trait-data structures developed for data curators and managers and the data input by data providers. Hardly any of the aforementioned trait datasets or databases for birds, amphibians, mammals or invertebrates employs the described standard terminologies, ontologies or semantic web standards. As it stands, re-using these data in larger compilations is error-prone and inefficient and the potential for a broad synthesis is diminished. 

One likely reason for this is complexity: the proposed data structures are designed for multi-layered, relational databases rather than for standalone datasets for which a two-dimensional data table may suffice. In the eyes of the data-provider, in most cases, ancillary co-factors can be appended as extra columns to the dataset.
The other reason is lack of awareness of the need for trait-data standardisation among data providers: many providers are not trained in the demands of biodiversity data-management. In addition, complying with what may be non-intuitive data structures is an investment without clear incentive or immediate pay-off, and hardly affordable for small and intermediate-size research projects. 

By filling this gap, data-brokering services [e.g. the German Federation for Biological Data; gfbio.org; @diepenbroek14] or data management systems for scientific projects (e.g. KNB and its open-source database back-end Metacat, https://knb.ecoinformatics.org/; Diversity Workbench, www.diversityworkbench.net; BExIS, http://bexis2.uni-jena.de/) are likely to gain importance. These services simplify and direct the standardised upload of research data and descriptive metadata into reliable and interlinked data infrastructures. The goal of such initiatives is to facilitate data re-use by providing standardisation of data, for instance by mapping unambiguous terminologies and ontologies for biodiversity data, and clarify conditions of data re-use. 

Another solution for data-users to access trait data in a structured way is offered by decentralised tools and toolchains to facilitate the use and analysis of trait-data. 
For instance the R-package 'traits' [@chamberlain17] contains functions to extract trait data via several open API interfaces including Birdlife, EOL TraitBank or BetyDB. The package  'TR8' provides similar access to plant traits from a list of databases [including LEDA, BiolFlor and Ellenberg values; @bocci15] and aggregates them into a species$\times$traits wide-table. 
<!--The package 'NatDB' [@pearse18] collates distributed trait data and harmonizes them into a unified format, and assists in browsing and filtering these data. -->
FENNEC [@ankenbrand18], an online tool or self-hosted service that can extract trait information from multiple sources for a target species community.

<!-- To remediate this, incentives for data standardisation before upload to file hosting services should be created. This could be implementations of data quality standards for trait-data as requirements for journal publications. Data publications should also be 
The recognition of data publications as a fully valid scientific contribution becomes also more likely if data are published according to the FAIR guiding principles for scientific data (data should be findable, accessible, interoperable and reusable; @). In this sense, trait-datasets will be findable if they are provided with explicit metadata and keywords, labelling them as trait-data for search engines (such as the Google Dataset Search). Accessibility and reusability are warranted by releasing datasets under public domain dedication or open licenses (such as Creative Commons 0 or Attribution license; @). Interoperability and 
-->

Interlinkage and accessibility of ontologies can be much improved to fulfil semantic web standards. 
A more widespread implementation of ontologies would advance the possibilities to aggregate datasets into databases and reduce noise and uncertainty. 
To achieve this, the use of ontologies and thesauri must be incentivised and facilitated for individual researchers.
For example, the accessibility of ontologies will increase if open APIs are provided as a way to extract the definitions and higher-level trait hierarchies programmatically via software tools. 
Software then can assist researchers in linking own data to globally defined concepts.

To bring all these tools for datamanagement together, a common vocabulary and definition of consensus concepts and exchance formats is required that connects to all steps of the data life-cycle.
Despite centralised data management efforts, however, no unified definition of traits as a data type with a well-defined exchange format or data structure has emerged across the multiple sub-disciplines of trait-based research. 
In the following chapter, we propose the a unified vocabulary for trait-data that can form the basis of such an exchange format. The simplicity of this standard terminology will lower the thresholds and offer high pay-off in the visibility and reuse of published data. 
By establishing this as a best-practice standard in trait-based research, trait-data will eventually fulfil the FAIR guiding principles for scientific data [@wilkinson16]. 

<!--
Depending on the research objective, trait-data are stored in a range of different table formats. In the subsequent paragraphs, we discuss the most common formats  in relation to the use cases in trait-based ecology. 

### a) Species wide-table

<!-- AO: You argue somehow that matrix format I a result of lit. review and so on. I would not agree with it. Your example also “just” points to aggregated data which menas one row is on species (aggregated) value.
Your example is (for me) not really a matrix. In a matrix you have a relation between two variables. In the first column and and on the upper row normally the values of one variable are written. An example would be species x body length. Then for body_length some classifications are written like (0-1, 1-2, 2-3,…) Then in the matrix the observation/abundance/… is noted. E.g. spec1 x “0-1” = 5, spec1 x “1-2” = 3, and so on. Your wexample hat already different variables (=traits) with different units as columns. That is a meaningfull representation of variables and values in a normal data table. Maybe it is good her to show the enormous number of different investigated traits and point to the resulting big (many columns = wide) tables.
--><!--
If trait data have been collated at the species level from different literature sources or from expert knowledge, or as aggregated measurements collated from raw data, they usually are reported in a species$\times$trait wide-table format (Fig. 2*a*), with a column of values for each trait recorded and a row for each species (or taxon) for which data were available. 
The wide-table format is widely used for the production of lookup-tables at the species level, which may either report qualitative facts (from literature or  expert knowledge) or aggregated quantitative traits (averages or ranges of trait values for this taxon). <!-- AO: You say it already 2 sentences before --><!--
This format needs to clearly differentiate missing data and traits not applicable to a given taxon (often filled in as 'NA' or dummy numbers, e.g. '999'). It may store additional information (e.g. on variation of means or literature source) in secondary columns or accompanying datasheets. 

Quantitative traits reported at this resolution are often used to calculate community weighted means or functional diversity metrics at the community level [@laliberte10a]. Also at this level of information, missing entries, e.g. about the trait values of a rarely described species, may be inferred from these tables when combined with phylogenies ['gap filling', @swenson14; @penone14; @taugourdeau14] and using Bayesian approaches [@schrodt15]. <!--However, these matrices do not allow estimates of within-species variation or assessments of the quality of the reported measurement or fact [@guralnick16]. --><!--AO: Why not? I can filter for a specific species and get all respective rows. But this is not a result a using a data format, it is a result of the data itself, FDS: no, since only one row per species exists; true that data itself don't cover this variation. -->
<!--

### b) Occurrence wide-table

In these occurrence wide-tables (Fig. 2*b*), each row is centered around a single physical occurrence of a species, e.g. phenotypic variation of traits within species would usually be recorded per observation where each replicate taken on a distinct individual specimen would be recorded in a row, with the respective trait values stored in columns. Those data are common in investigations of evolutionary trade-offs and trait correlation, or of intra-specific variation along environmental gradients. Occurrence data would also capture phenotypic variation arising from morphotypes and sexual dimorphism. This format is the most intuitive for recording own empirical measurements and therefore is common for measured quantitative data but rarely found for reported qualitative facts obtained from the literature.

### c.) measurement long-table

Both wide-table formats report the different trait values stored in multiple columns. Each trait column represents one value that applies to the entity in question (i.e. to a species  or an occurrence). As the column names and the width of the table varies with the number of traits included, merging data from different sources requires user-defined mapping and manual harmonization of these structures. 

A computationally more effective method that allows for high flexibility is the storage of traitdata in long-table formats [@wickham14; @parr16]. In these, each row is reserved for a single measurement or fact of a specific trait, referenced to a single specimen  (i.e., one occurrence)  which is assigned to a specific taxon. 
<!-- FDS: also refer to kattge 11 -->
<!--AO: Useful for data sharing, ease of data harmonization, aggregation: probably yes. But not necessarily computational more effective at all. E.g. I start to mix up numeric values with descriptive values. I start to mix up different units. Now it is more complicated to retrieve associated values (like all length measurements in mm) again to make e.g. quality stuff. Ergo less effective from that perspective. But it has some advantages which may overrule dis. See below 
AO: For the whole proofing why to use such a long format I think that GBIF and DwC has already given some sentences. I strongly think their work should be considered and can be used as reference!
We can simply argue that a well established routine should now also executed on trait level!
Furthermore for DwC exists also the “extended measurement or facts extension” which is useful for putting e.g. traits to species.

--><!--
This allows repeated measurements on a single individual to be stored by linking the data from separate rows via a unique specimen identifier. Also, multivariate trait measurements can be recorded in this format by linking multiple rows via a unique measurement identifier.  Long-table datasets purport multiple advantages for data manipulation (e.g. filtering, sub-setting and aggregating data), visualization (e.g. plot measured values by factor variable or taxon) and statistical modelling (e.g. ANOVA for testing difference of trait value by sex) [@wickham14]. As long-table fromats draw from a defined set of columns,  merging of datasets is much easier. 
<!-- BKR: Is that really true? Of course, you have less problems on the schema or structure level. But, on the other hand, the differences (e.g. different terms for the same trait,…) now happen on the instance level (that is – in the data; in the rows). Thus, at least with respect to integration, you just move the problem from one level to the other.  To ease integration, one would need to agree on the entries in the features column
AO: Maybe describe some more advantages here like: it is possible to ensure that the items/values are always taken from a provided lookup-table e.g. for taxon, trait_names, units, more?
We may have something like a dimension, the data source (measurement, lit. review, …) and some more criteria which are beneficial for researchers  and data aggregation?
Again: I think we find here papers regarding DWC and GBIF (and OBIS) as source of proofing the facts.
-->
<!--
The long-table format therefore is ideal for storing raw data of trait measurements in huge databases, and allows the entries to be linked via the respective unique identifiers to additional information. Each row of the dataset can therefore be interpreted as a statement of an 'entity x having a qualitative/quantitative feature y' (Fig. 2*c*). 

<!-- Cat: here I guess that a figure showing few rows of a species x trait matrix and a long-format one with links to other information (metadata,..) woud help quite a lot. By experience some users are not be familiar with the interest of using the long-table format.-->
<!-- Malte: I agree (see my comment above), such a figure / would definitely help convince people (or at least not to lose them). -->
<!--
\begin{figure}[t]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig2.png}
\caption{Formats used for trait datasets.  a) taxon-level trait data compiled from literature or aggregated from measurements are often published as a compiled species $\times$ traits matrix; b) wide-tables report measurements on the specimen-level, i.e. one physical occurrence of a taxon; c) a measurement long-table is a well defined and tidy data format, reporting one single measurement per row and relating it to a trait definition and taxon via unambiguous identifiers. Additional identifiers relate each row to other layers of information on d) the specimen (occurrence) or e) the origin or confidence on the reported measurement or fact.}

\end{figure}
-->

<!--The trade-off is data-consistency and readability *vs.* avoidance of content duplication. -->

<!--When traits are assessed on a higher level of organisation, e.g. microbial functional traits assessed at the community level taken from a soil sample, key data columns would be placed in the same table as the core data are traits. Here, location or measurement information are the primary focus of the investigation and must be contained in the . 
<!-- PM: I get lost here-->
<!--A general recommendation, whether a column describes ancillary data or is part of the central dataset is ill advised. Therefore, our glossary of terms and its extensions should be used to describe the scientific data according to the study context. -->
<!-- FDS: shorten this and refer to existing standards for database use. the star-structure of ABCD, Kattge et al , and Garnier .-->
<!--while the existing trait data standards may cover all of the aspects required in the various use-cases of trait-based research. <!-- examples for blind spots; PM: Vague- not sure what is missing here- be more specific --> 

# An ecological trait-data standard vocabulary

As a response to the challenges outlined above, we propose a versatile vocabulary for trait-based ecological research. 
The standard vocabulary is accessible at https://terminologies.gfbio.org/terms/ets/pages/.
The aim of this vocabulary is to cover the variety of trait-based approaches and their different degrees of measurement details. 
Rather than describing a data structure for relational databases, the vocabulary is intended as a more inclusive terminology, that can be used in three major use cases: 
1. by data providers: for publication of standardised primary data on open-access data repositories, or for labelling project-specific data for local use and exchange with collaborators, e.g. in two-dimensional data tables or project databases
2. by data users and curators: as a consensus vocabulary when compiling data from distributed sources into databases, e.g. to map standardised columns and refer to taxa and trait definitions in a uniform way, and
3. by data managers: as a data exchange format of between databases, web services and software tools, e.g. when providing database queries via a web service or defining input and output formats of software packages.

By using this standard vocabulary, authors can ensure that the description of trait measurements uploaded into distributed data repositories will be unambiguous and generally applicable. It will facilitate interoperability of published data and enable their re-use for future data aggregation initiatives, data synthesis and ensure long-term accessibility. By adopting the standard vocabulary, trait-data will support the FAIR principles for scientific data [@wilkinson16]. 
We describe best practice principles and give examples for the application of the vocabulary in Supplementary Material B. 

In designing this vocabulary, we drew on the combined expertise of empirical biodiversity researchers (data providers), biodiversity synthesis researchers (data users and curators), and biodiversity informatics researchers (data managers).
We paid particular attention to existing data standards [particularly @madin07; @kattge11; @kattge11a; @garnier17; @parr16] to ensure compatibility of our proposed data structure with major trait databases and existing standards for biodiversity data management. 

<!--
## Why use the vocabulary

The incentives for using a unified vocabulary for trait data might not be obvious for data providers, who are mainly concerned with a particular study site, organism group and research question. We won't discuss here how sharing trait data will improve science on broader questions and larger scales, by providing a data basis for downstream analysis. However, it appears to be broad consensus that an open biodiversity science is crucial for an evidence-based decision making and conservation policy on regional and global scales. 

Instead, this section points on the technical aspects and advantages of readying own trait data to link with the 'semantic web' of biodiversity data and knowledge. 
By using a constrained vocabulary with globally accessible definitions of terms, data can be accessed more easily by other researchers and datasets from different sources can be combined into aggregated datasets.
Also it will be easier to develop computational methods and software tools that access and handle the data, based on the standard vocabulary. -->

A trait dataset typically contains data according to the Entity-Quality model described above [@garnier17]: each entry describes a trait value (i.e. quality) observerd on an individual or population (i.e. entity), of a biological taxon. 
The trait values may be quantitative measurements or qualitative facts. 
Quantitative measurements are values obtained either by direct morphological, physiological or behavioural observations on single specimens (Fig. 1*a*), by aggregating replicated measurements on multiple entities (Fig. 1*b*) or by estimating the means or ranges for the respective taxon as reported in the literature or other published sources (e.g. databases, Fig. 1*c*). 
This encompasses a wide range of data types, including numeric, integer, intervals or ratios, or even values from an ordinal scale. Qualitative facts are assignments of an entity to a categorical level, e.g. of a behavioural or life-history trait (Fig. 1*d*). 
Beyond these core observations, further information might be available that further specify the taxon concept applied, provide detail on the measurement method, or that place the reported measurement in a broader observation context (including geolocation and date information).
These information may be useful for future analysis of the causal reasons of trait variation and should always be published along with the core data.

\begin{figure}[t]
\centering
\includegraphics[width=0.85\textwidth, angle=0]{figures/fig1.png}
\caption{Types of ecological trait data assume different entities or reported qualities: a) morphometric or morphological measurements of individual body features (lengths, areas, volumes, weights) or other quantities related to life history (e.g. reproductive rates, life spans); b) aggregated trait values are reported as means taken on multiple measures of members of a taxon; c) quantitative traits may be extracted from literature or existing databases, referring to the entire taxon (or a subset, e.g. a sex) as the entity of description; d) qualitative traits are categorical, ordinal or binary descriptors of the entire species or higher taxonomic level (also called 'facts').}

\end{figure}
<!-- DF : The “variables” for the taxa etc should only be reused for the different examples when they refer to something similar: “a) Entity: an individual X of taxon Y”, “b) Entity: taxon Y”, “c) Entity: males of taxon Z” and “d) Entity: genus U”. Alternatively, since these are already example, why use variables here in the first place and not just specific example for the taxa as well: “a) Entity: an individual X of taxon Tettigonia viridissima” -->

Our vocabulary thus provides core terms to reflect the minimal Entity-Quality structure of trait data. 
We further offer three extensions to contain ancillary information on different layers of the core data ("Measurement or Fact", and "Occurrence", "Taxon"; see section on extensions below). 
All terms may be applied to describe columns of a data table, or to label content of a database (see Box 1 / Supplementary Material B for best practice principles and examples for publishing primary data).
Additional terms are provided for metadata and for providing trait concepts (see section on metadata below).  
Some of the terms expand and refine terms of DwC and its extensions (http://rs.tdwg.org/dwc/terms/) or the Glossary of EOL (http://eol.org/info/516). Wherever our vocabular refines or duplicates existing terms, we indicate this in the fields 'refines' or 'identical', respectively. The vocabulary can be mapped onto other existing ontologies (e.g. @madin07) or database structures (e.g. @kattge11a).
Aspects not covered by the terminology may draw from terms provided by DwC and it's extensions or be added as user-defined columns (which should be clearly specified in the meta-information accompanying the dataset). 


## Specification of core terms

For trait data according to the Entity-Quality model, the following content is required at minimum (Fig. 2 *a&b*):
1. a value (column `traitValue`) and -- for numeric values -- a standard unit (`traitUnit`); 
2. a descriptive trait name (`traitName`) that links to a standardised definition (i.e. a concept); 
3. the scientific taxon name for which the measurement or fact was obtained (`scientificName`) that links to an accepted taxon concept. 

<!--### Relate to global taxon and trait definitions -->
The trait name and taxon names as would ideally use unambiguous terms assigning both to clearly-defined concepts. 
However, to ensure compatibility with existing databases or analytical code, it might be necessary to use abbreviations or user-specific identifiers for `scientificName`  and `traitName` instead. 
In this case, it is essential to relate the user-defined names to a consensus standard of taxon names as well as a look-up table of traits. 
This is achieved by adding globally valid Uniform Resource Identifiers (URIs) for taxon (`taxonID`) and trait definitions (`traitID`), complemented by the human-readable verbatim accepted names (`ScientificNameStd` and `traitNameStd`, respectively, Fig. 2 *c*). 
For example, referring to GBIF Backbone Terminology, for *Bellis perennis*, the `taxonID` would be  `https://www.gbif.org/species/3117424`; the `traitID` for 'fruit mass' according to TOP Thesaurus of plant traits would be  `http://top-thesaurus.org/annotationInfo?viz=1&&trait=Fruit_mass`.  
<!-- units and factor levels-->
Similarly, standardisation of units can be achieved by relying on SI convention or by relating units to unambigous concepts via URIs provided by ontologies [@steinberg16; @keil18]. 
For categorial, ordinal or binary traits, the factor levels should conform to expected levels as defined in the trait concept, or be unambiguously defined in the metadata of the dataset (see section metadata below). 
The vocabulary complements the user-defined values in dataset-specific units and factor levels with standardised entries (`traitValueStd` and `traitUnitStd`). 
<!-- double record -->
By suggesting for a double record of both user-specific and standardised entries, we acknowledge the fact that most authors have their own schemes for standardisation which may refer to different scientific community standards  [as practiced in TRY; @kattge11]. This redundancy of data allows for continuity for data providers while also enabling quality checks and comparability for data curators.

## Extensions for additional data layers

Beyond measurement units or higher taxon information, further information might complement the core data which are related to the individual specimen, the reported fact, measurement or sampling event. 
We propose three extensions of the vocabulary that should be used to describe this information (Fig. 2*d*):

- The `Taxon` extension provides further terms for specifying the taxonomic resolution of the observation and to ensure the correct reference in case of synonyms and homonyms.  (https://terminologies.gfbio.org/terms/ets/pages/#extension-taxon)
- The `MeasurementOrFact` extension  provides terms to describe information at the level of single measurements or reported facts, such as the original literature reference for the reported value, the method of measurement or statistical method of aggregation. It provides important information that allows for the tracking of potential sources of noise or bias in measured data (e.g. variation in measurement method) or aggregated values (e.g. statistical method), as well as the source of reported facts (e.g. literature source or expert reference). (https://terminologies.gfbio.org/terms/ets/pages/#extension-measurement-or-fact)
- The `Occurrence` extension contains vocabulary to describe information on the observation context of individual specimens, such as sex, life stage or age. This also includes the method of sampling and preservation, as well as the date and geographical location, which provide an important resource to analyse trait variation due to differences in space and time. (https://terminologies.gfbio.org/terms/ets/pages/#extension-occurrence)

<!--Many terms of these extensions refine or copy terms of the DwC and their own [Taxon](http://rs.tdwg.org/dwc/terms/index.htm#Taxon),  [MeasurementOrFact](http://rs.tdwg.org/dwc/terms/index.htm#measureindex) and [Occurrence](http://rs.tdwg.org/dwc/terms/index.htm#Occurrence) extensions and EOL TraitBank's use of those terms (http://eol.org/info/structured_data_archives).
 If no taxonID has been provided, in order to be unambiguous about the taxon referred to by the trait observation, it is highly recommended to keep an additional reference to higher taxonomic hierarchies (using columns `kingdom`, `order` etc.) to avoid taxonomic homonyms, and provide the authority (???) and key of reference (???).  
Furthermore, it is best practice to keep a record of the reported taxon rank (in column `taxonRank`) for traits recorded at the family or genus level. This information can then be used for trait inference at higher taxonomic resolution. <!-- using hierarchical probabilistic matrix factorisation [@schrodt15]. -->
These additional layers of information can either be added as extra columns to the core dataset or kept in separate data sheets, thus avoiding redundancy and duplication of content. A unique identifier links to these other datasheets, encoding each individual occurrence of a species (`occurrenceID`), single measurements or reported facts (`measurementID`), locations of sampling (`locationID`) and sampling campaigns (`eventID`). 
Some data-types may directly refer to existing global identifiers for occurrence IDs, e.g. a GBIF URI or a stable identifier references the precise specimen from which the measurement was taken [@groom17; @guntsch17].

Identifiers also can be used to provide a structure within a two-dimensional data table, e.g. to link rows of measurements on the same individual (using `occurrenceID`) or observations at the same location (using `locationID`). The  values of multivariate measurements can be labelled using `measurementID`. 

## Specification of Metadata

Higher-level information about structure, authorship and data ownership and terms of use should be considered when sharing and working with trait datasets.
Our metadata vocabulary (https://ecologicaltraitdata.github.io/ETS/#metadata-vocabulary) provides terms to label these information. 
In the case of primary measurement data, this information applies to the entire trait dataset, and should be stored along with the published data as metadata [e.g. in a separate file, possibly applying the ecological metadata language, EML, @knb11]. 
Whenever data from different sources are compiled into a trait database, these information must be provided at the measurement level to maintain reference to the original data provider and conditions of re-use of these data. This can be achieved by appending the information as columns to the core dataset, or via an unambiguous `datasetID` and a descriptive `datasetName`. 

Wherever possible, the field `traitID` should point to a publicly available, unambiguous trait definition in a published ontology (using its URI). 
If such an external reference does not exist, trait data should always be accompanied by a dataset-specific list of trait concepts. 
Such a controlled vocabulary would, in its simplest form, assign trait names with an unambiguous definition of the trait and an expected format of measured values or reported facts (e.g. units or legit factor levels). Ideally, this definition refers to or refines terms from published trait ontologies.
By providing a minimal vocabulary for trait lists (see https://ecologicaltraitdata.github.io/ETS/#terms-for-trait-definitions), we hope to facilitate the unambiguous definition of traits for trait datasets.
This vocabulary might also prove useful for the future publication of trait ontologies. 
<!--See our online reference for recommendations how to prepare a traitlist for  (https://ecologicaltraitdata.github.io/ETS/thesauri.html). We also provide an example thesaurus on Github (http://ecologicaltraitdata.github.io/TraitDataList/), which has been developed to capture traits assessed for multiple animal taxa within the German Biodiversity Exploratories project [@fischer10]. -->
<!--Open access to science data from publicly funded research is a declared goal of international and national research organisations [@emerson15; @allianceofgermanscienceorganisations10; @theroyalsocietysciencepolicycentre12] and increasingly required by journals and funding agencies to warrant reproducibility of science and foster public benefit. 
Also, factual data such as trait measurements, being not a product of a creative act, do not qualify as intellectual property and copyright does not apply [@carroll15; @creativecommons].
However, there is still no broad consensus achieved in the scientific community whether data from public funding must be generally dedicated to the public domain, or whether the investment in production must be acknowledged by attribution or co-authorship. -->
<!-- cP: Maybe we should put in the appendix an example of dataset+metadata? This should probably go in the metadata section 
FDS: to consider!! -->

In order to ensure traceability, the metadata of any dataset that employs this vocabulary should refer to the specific online version that was used to build the dataset, e.g. "Schneider, F.D., Jochum, M., Le Provost, G., Penone, C., Ostrowski, A. and Simons, N.K., 2018 Ecological Traitdata Standard v0.9, DOI:  10.5281/zenodo.1441132, URL: https://terminologies.gfbio.org/terms/ets/pages/".
Wherever referring to individual terms of the vocabulary in publications or metadata, this should be done via their URIs. 
The definitions of terms are hosted on the GFBio Terminology Service [@karam16, https://terminologies.gfbio.org/] and can be accessed programmatically (i.e. via the API; https://terminologies.gfbio.org/api/terminologies/). 

\begin{figure}[t]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig2.png}
\caption{Formats used for trait datasets.  a) taxon-level trait data compiled from literature or aggregated from measurements are often published as a compiled species $\times$ traits wide-table; b) observation long-tables are a well defined and tidy data format, reporting one single measurement per row and c) relating it to a standard trait definition and accepted taxon name using unambiguous identifiers. Additional identifiers relate each row to other layers of information on d) the taxon resolution, the specimen (occurrence) or the origin or confidence on the reported measurement or fact.}

\end{figure}

# Discussion

To serve the demand for the standardisation and harmonisation of ecological trait data which has arisen from a growing number of distributed datasets of different research contexts, we propose a versatile vocabulary for the publication of new datasets, for the creation of data compilations, as well as for the exchange and handling of trait data in the context of a 'semantic web'.

<!-- community effort and consensus -->
While it is clear, that a single vocabulary standard is not able to fulfil the needs of all research fields working with traits, we believe that our proposal captures a common core concept of trait data. 
Consensus building on how traits are to be used and evaluated is currently under way in several fields with their taxonomic focus and project-specific questions [@kissling18; @garnier17; @pey14; @moretti17]. While community discussions on trait definitions and measurement practices are leading to a better quality of data, naturally, the heterogeneity in trait-data quality and detail across fields will remain. 
With our proposal of an Ecological Trait-data Standard vocabulary, we provide a starting point for the development of a joint language and terminology around traits as a cross-sectoral topic of ecological and evolutionary research. 
Therefore, we invite researchers to contribute to future versions of the standard vocabulary and develop own applications and ontologies that interact with it.
The standard vocabulary will be released in subsequent versions and published as a stable reference on the GFBio Terminology Service (https://terminologies.gfbio.org/terms/ets/pages/). 

<!-- ETS investment and pay-off for data providers, incentives -->
For individual researchers and small research projects, the investment of time and resources in trait-data standardisation before publication seems currently not to be yielding a direct benefit.
However, data released according to consensus standards, especially if published under open-access licenses, will be more easily re-used in compilations and synthesis studies. 
A well-defined vocabulary for metadata will also ensure that authorship and terms of use are appropriately documented along the data life-cycle.
However, for these incentives to take effect, data publications and data citations must become viewed as a valid scientific contribution to the community, and  recognized in the professional evaluation of individual researchers [@costello09; @roche15].
To mitigate the cost of data standardisation efforts, supporting tools and infrastructures should be developed to mitigate the cost of data providers for readying trait data for the 'semantic web' of biodiversity data.

<!-- ETS solving Data compilation issues --> 
At the community level, shifting the task of standardisation from the data-user side to the data-owner side yields great gain in accuracy and reduces the risk of misinterpretation. For instance, measurement results depend very much on the precise methodology used and often systematic biases could be corrected for when providing an undambiguous definition. 
On the other hand, plausibility checks and evaluation of statistical methods, e.g. for aggregating trait values to the species level, can only be done in comparison across a wide array of datasets. Currently, these "big data" volumes are only available in centralised databases.
However, to establish a best-practice of data aggregation, an exploration and evaluation of different methods for quality assessment and quality control should be subject to a community discussion. 
This is only possible with large quantities of distributed data being available in a harmonised way. Our ecological trait-data standard vocabulary facilitates such a community-driven comparison. 

<!-- ETS to build toolchain and semantic web -->
Our standard vocabulary represents an important building block for a unified mode of data exchange between web services and software packages and thus facilitates the development of a software toolchain for the trait-data lifecycle. 
Having well-defined terms is also a key precondition for developing exchange formats between large database initiatives and biodiversity data archives. 
Even further downstream, readying the primary data for the semantic web via references to ontologies and data standards will ease the application of automatised big-data mining and machine-learning techniques. Without clearly defined terms and concepts, handling of large amounts of trait-data by computational assistance systems for scientific analysis ('e-Science') will be massively hampered [@wilkinson16]. 

# Conclusion

To date, there is a rich distributed body of independently published trait datasets, with different focus on particular organism groups, ecosystem types or regions. These distributed data are heterogeneous in form and description and initiatives to harmonise, compile and analyse these data require significant amounts of funding.

By using a standard vocabulary with globally accessible definitions of terms,  would allow distributed trait data to be more easily re-used and harmonised into aggregated datasets. 
The biggest challenge in future standardisation of trait data may be consensus building for standard terms, the establishment of incentives and the development of tools for a user-side standardisation and publication of data.
This requires significant effort, but it returns  great scientific benefit by enabling data-heavy synthesis for a general understanding of biodiversity and ecosystem functioning. 

# Acknowledgements

Thanks to all respondents to an internal online survey on trait data for the Biodiversity Exploratories project and to <!--Matthias Biber, Kristin Bohn,--> Diana Bowler, Klaus Birkhofer, Runa Boeddinghaus, Markus Fischer, Jens Kattge, Catrin Westphal for comments on the manuscript drafts and pre-print, as well as the ecological trait-data standard vocabulary<!-- and the pre-releases of the R-package-->. We are grateful to the organisers and participants of the Open Traits workshop in New Orleans, USA, in August 2018.

We thank the managers of the three Exploratories, Kirsten Reichel-Jung, Katrin Lorenzen, Juliane Vogt, Miriam Teuscher, and all former managers for their work in maintaining the plot and project infrastructure; 
Christiane Fischer and Jule Mangels for giving
support through the central office, and Markus Fischer, Eduard Linsenmair,
Dominik Hessenmöller, Daniel Prati, Ingo Schöning, François Buscot,
Ernst-Detlef Schulze, Wolfgang W. Weisser and the late Elisabeth Kalko
for their role in setting up the Biodiversity Exploratories project. 
The work has been partly funded by the DFG Priority Program
1374 "Infrastructure-Biodiversity-Exploratories" (DFG-Refno.). <!--Field work
permits were issued by the responsible state environmental offices of
Baden-Württemberg, Thüringen, and Brandenburg (according to § 72
BbgNatSchG).-->

MMG obtained funding from Swiss National Science Foundation (SNF 310030E-173542/1);
MJ was supported by the German Research Foundation within the framework of the Jena Experiment (FOR 1451) and by the Swiss National Science Foundation.

# Authors' contributions

FDS, AO, CP, and NKS conceived the idea and developed the vocabulary for the trait data standard with significant contributions of MJ and GLP;<!--NKS authored the example list of traits; FDS developed the R package 'traitdataform';--> CP and FDS curated the living spreadsheet; AG and DF implemented the vocabulary in the GFBio terminology service; all authors contributed critically to the structure and content of the manuscript and gave final approval for publication.

# Data Accessibility

The online reference for the Ecological Trait-data Standard Vocabulary described in this paper is https://ecologicaltraitdata.github.io/ETS/, stable DOI representing all versions: 10.5281/zenodo.1041732. Any future development of the vocabulary is coordinated via https://github.com/EcologicalTraitData/ETS/.  

Supplementary Material A - static excerpt of a living spreadsheet (https://goo.gl/QxzfHy) which lists published trait datasets, their regional and taxonomic focus, the number and scope of traits covered, their location on the internet and the terms of use. We invite data owners and users to add further trait datasets to this spreadsheet.  

Supplementary Material B - contains three worked examples of how to  1.) apply the core ETS vocabulary to species$\times$traits wide-trables, 2.) convert data to long-table format using identifiers 3.) apply ETS extensions and identifiers to include co-variate information on different layers of the dataset.

<!--The development website for the R-package 'traitdataform' is https://github.com/EcologicalTraitData/traitdataform.
<!--An example of a published trait thesaurus for the Biodiversity Exploratories (www.biodiversity-exploratories.de) can be found at https://ecologicaltraitdata.github.io/TraitDataList/.-->


# References 

\setlength{\parindent}{0pt}
\setlength{\parskip}{5pt plus 2pt minus 1pt}
\setlength{\emergencystretch}{3em} 

\footnotesize