---
title: "Towards an Ecological Trait-data Standard"
author:
- name: Florian D. Schneider \footnotemark[1] \footnotemark[2] \footnotemark[3]
- name: Malte Jochum \footnotemark[4]
- name: Gaëtane Le Provost \footnotemark[3]
- name: Andreas Ostrowski \footnotemark[6]
- name: Caterina Penone \footnotemark[4]
- name: David Fichtmüller \footnotemark[5]
- name: Martin Gossner \footnotemark[8]
- name: Anton Güntsch \footnotemark[5]
- name: Pete Manning \footnotemark[3]
- name: Birgitta König-Ries\footnotemark[6]
- name: Nadja K. Simons \footnotemark[7]
csl: amnat.csl
fontsize: 11pt
geometry: margin=1.3in
output:
  pdf_document:
    fig_caption: yes
    fig_crop: no
    keep_tex: yes
    template: pretty.template
  html_document:
    fig_caption: yes
    fig_width: 4
    highlight: null
    keep_md: yes
    theme: united
  word_document:
    fig_caption: yes
    fig_width: 6
bibliography: div05.bib
---

\renewcommand{\thefootnote}{\fnsymbol{footnote}}

\footnotetext[1]{Corresponding author: florian.dirk.schneider@gmail.com}
\footnotetext[2]{Institute of Linguistics and Literary Studies, Technische Universität Darmstadt, Darmstadt, Germany}
\footnotetext[3]{Senckenberg Biodiversity and Climate Research Centre (BiK-F) Frankfurt am Main, Germany}
\footnotetext[4]{Institute of Plant Sciences, University of Bern, Bern, Switzerland}
\footnotetext[5]{Botanic Garden and Botanical Museum Berlin, Freie Universität Berlin, Berlin, Germany}
\footnotetext[6]{Department of Mathematics and Computer Science, Friedrich-Schiller-Universität Jena, Jena, Germany}
\footnotetext[8]{Forest Entomology, Swiss Federal Research Institute WSL, Birmensdorf, Switzerland} <!--  Zürcherstrasse 111, CH-8903 -->
\footnotetext[7]{Chair for Terrestrial Ecology, Department of Ecology and Ecosystemmanagement,
Technische Universität München, Freising, Germany}

\linenumbers

# Abstract

1. Trait-based approaches are widespread throughout ecological research, offering great potential for trait data to deliver general and mechanistic conclusions. Accordingly, a wealth of trait data is available for many organism groups, but, due to a lack of standardisation, these data come in heterogeneous formats.
2. We review current initiatives and infrastructures for standardising trait data and discuss the importance of standardisation for trait data hosted in distributed open-access repositories.
3. In order to facilitate the standardisation and harmonisation of distributed trait datasets, we propose a general and simple vocabulary as well as a simple data structure for storing and sharing ecological trait data. 
4. Additionally, we provide an R-package that enables the transformation of any tabular dataset into the proposed format. This also allows trait datasets from heterogeneous sources to be harmonised and merged, thus facilitating data compilation for any particular research focus. 
5. With these decentralised tools for trait-data harmonisation, we intend to facilitate the exchange and analysis of trait data within ecological research and enable global syntheses of traits across a wide range of taxa and ecosystems. 

# Key-words: 

functional ecology, species traits, semantic web, ontologies, data standard

# Introduction

<!-- CP:  I think we should develop more on what are the current problems/limitations (quickly because it’s then expanded in the paper) so it becomes evident that the standard+package fill a real need.

That’s the structure I try to follow:
1. Traits are important in ecology, evolution and conservation (i.e.  It is important to mention that traits are important in research and give some examples of questions that traits can help to solve but maybe not spend too long on it (most people know that they are important).

2. To increase the scale of studies (spatial, temporal, taxonomical) we need to put together trait data from different sources. When we do this we can answer new questions and find generalities. So there is a growing interest in assembling trait data from different sources and taxa (e.g. give the number of papers using TRY).

3. There is a problem of standardization that makes it difficult to match trait datasets from different sources. Different reasons: no traits definitions, no common structure.

4. One solution are databases. But these are costly and depend on the will of few persons/funding. They also cover few taxonomic groups. Matching between different databases is still difficult because of different definitions and structures. Most data are still uploaded separately.
We need a solution to make their use easy. 
-->
<!-- GLP: 
 Add a general paragraph on the fact that there have been some initiatives for the development of trait data base for plants (e.g. TRY), why is it important (provide quick access to plant trait values, allowing comparisons even between ecosystems and biomes), now the development of trait data base for other organisms (from microbes to mammals). Say also something on the development of general protocols for measuring traits (e.g. Moretti et al. 2016; Freschet et al. 2017) ?

Importance of data standardization. This is the main idea here and so I wonder if some idea of the first point (Need for data standardization) should not be merge here with the last paragraph of the introduction ? As there is also some repetitions between those 2 parts.

I wonder if you should speak about ‘Functional traits’here or just Traits ? As you are not only considered Functional traits in this paper (e.g. life history traits) ? Maybe remove the definition no ? 
-->
<!-- PM: Thuis is quite complex- e.g. ‘taxonomic impediment’etc. I actually think you can simplify this and built these references into the more general text above by adding a copule of more examples. I don’t think you need to say much about the merits of the trait based approach here- it’s covered elsewhere. 

Huge number of references here- I think you can strip it down to a few key ones, and reviews. Overall I think this section is quite a lot more complex than it needs to be – I think all you need to say is that it’s very popular, general and mechanistic and so adopted throughout eco and evo, with no sign of slowing down- the TRY database download stats can provide a quantitative measure of this- see their site. 
-->

<!-- 1. Traits are important in ecology, evolution and conservation -->

Functional traits are phenotypic  (i.e. morphological, physiological, behavioral) characteristics that are related to the fitness and performance of an organism [@mcgill06; @violle07]. Because trait-based approaches allow studying both patterns and mechanisms [@lavorel02; @diaz16], recent years have seen a proliferation of trait-based research in a wide range of fields.
Trait-based studies have been conducted in a wide range of thematic areas ranging from the evolutionary basis of individual-level properties [@salguero-gomez16] to global patterns of biodiversity [@diaz16] and ecosystem functioning [@allan15; @bello10]. 
The trait framework relates losses of ecosystem function to changes in  the functional composition of species assemblages [@mouillot13; @perovic15]. This offers the mechanistic background to relate biodiversity to climate change or local anthropogenic land use [@diaz11; @lavorel12; @allan15]. <!-- ;@wood15;  @diaz11; @jennings09; @hopfenmuller14; @mouillot13;  @vandewalle10; @perovic15-->
Using traits is also a promising means of bypassing taxonomic impediment, i.e. the fact that a majority of species are yet undescribed and little is known of their interactions with the environment and other organisms. This is because functional traits allow us to infer the ecological role of organisms  from their apparent features, regardless of their taxonomic identity [@leprovost17; @duarte11; @schrodt15].

<!-- Comparative studies of traits inform researchers about physiological and evolutionary trade-offs, and help identifying economic strategies of organisms beyond taxonomy [@menezes10; @forister15;  @siefert15; @gross09; @wright04]. Inferring an unknown functional role or ecological function of individuals from their apparent features [@duarte11; @schrodt15] is a promising venue to bypass taxonomic impediment, i.e. the fact that a majority of species are yet undescribed and little is known of their interactions with the environment and other organisms. 
Mutual 'trait-matching' of co-evolved organisms enables the inference of ecological interactions from observed features of live specimens [ @kalinkat13; @dehling14a] which may be used to inform ecological networks [@weiner14; @miranda13; @digel11; @schleuning15] and parametrise ecological population-dynamic models [@schneider12; @kalinkat13]. These analyses are also a promising venue for a multitrophic community ecology aiming to unravel dependencies and synchrony in trait spaces between interacting functional groups [@soliveres16]. <!-- GLP: Maybe not the best example as they are not looking at trait diversty and species interactions ? 
Put instead:
Gravel et al. 2016 The meaning of functional trait composition of food webs for ecosystem functioning
And Coux et al. 2016 Linking species functional roles to their network roles
?
-->
<!--
Since traits are so closely related to ecological mechanisms, they have proven useful for scaling individual level physiology and behaviour to functional diversity  of communities [@laliberte10a; @petchey09; @villeger08a; @hillerislambers12;@diaz98; @lavorel02], processes at the ecosystem scale [@fontaine05; @simons16; @gossner16; @moretti13;  @kuhsel15;  @birkhofer14], as well as macro-patterns of biodiversity at the global scale [@diaz16; e.g. @reich97]. <!-- Traits can be used to predict extinction risk [@fritz10; @pearson14] and susceptibility to environmental toxins [@rubach11].--> <!-- GLP: What do you mean by processes ? I’m not sure that the literature cited is always appropriate here ?
-->
<!--
Most importantly, the trait framework relates key ecosystem services for human well-being [@diaz11; @lavorel12; @allan15] back to changes in biodiversity in consequence to global change or local anthropogenic land use [ @diaz07; @gossner16; @mangels17; @jennings09; @simons16;  @mouillot13;  @vandewalle10;  @birkhofer15;@gamez-virues15; @birkhofer17]. 
--> 
<!-- In recent years, trait-based ecology has thus grown into a major field of ecological research, which continues to integrate a rapidly increasing number of fundamental and applied research questions. 
Given the expected future challenges in dealing with climate change and managing land use,  global trait synthesis is an interdisciplinary, cross-topic field with great potential. -->


<!-- 2. assemble trait data into databases -->

Many issues in trait-based research arise when compiling datasets from several sources. Data may differ in taxonomic nomenclature and resolution (e.g. reported on species level or aggregated on higher taxonomical orders), the scale and place of the study context, or the accurracy of the methodology applied in measurements. These differences are not always documented in the metadata accompanying a dataset. 
All of these factors render trait data extremely heterogeneous and make the task of data compilation time-consuming or even prohibitive.
However, fully exploiting the potential of trait-based approaches relies heavily on the broad availability and compatibility of trait data to achieve sufficient taxonomic and regional coverage, both of present-day taxa ase well as in evolutionary deep-time. 

<!-- 3. trait databases have been developed, with standard structure and terms --> 
To this end, the number of available trait datasets is increasing rapidly. 
In the past, trait data have been standardised and compiled in centralised databases for specific organism groups and regional scope, often centred around particular research questions [e.g. PanTHERIA, @jones09; TRY, @kattge11; AmphiBio, @oliveira17]. 
<!--These databases are then published as static data tables (i.e. product of a one-time effort as it is the case of AmphiBIO), or in the form of sophisticated online portals, which attract data submissions from a focal community of researchers and provide research networks for consensus building in data resolution and measurement methods (e.g. TRY, www.try-db.org). Furthermore,--> These initiatives map heterogeneous data into a common scheme and, importantly, also offer access control and data usage policies. As such, they protect the rights of the original data providers while simplifying data queries for synthesis researchers. 
Besides initiatives aiming at assembling data, other tools to enable the compatibility of data across databases are being developed. These include semantic-web standards [@wieczorek12; @page08] and ontologies of standard terms [@walls12; @garnier17].
<!-- Solving the numerous issues of data assemblage and harmonisation in centralised infrastructures requires a huge effort in costs and personnel. -->
<!-- 4. distributed trait data must be integrated --> 
Meanwhile, open-science reaches the mainstream: it has become the declared goal of an open biodiversity knowledge management (http://www.bouchoutdeclaration.org/) and is increasingly demanded by journals and public research funding [@emerson15; @allianceofgermanscienceorganisations10; @theroyalsocietysciencepolicycentre12; @swan12; @allison15]. As a result, an increasing number of individual research projects publish their primary data on file hosting services like Figshare.com, Dryad (datadryad.org), Researchgate.net, or Zenodo.org, where no data standards are forced upon the uploaded material. <!-- The data standards mentioned above, that have been proposed for relational trait-databases, do hardly apply for these individual datasets.-->
It is likely that trait data will become increasingly available, but a lack of data and metadata standardisation will hamper the efficient re-use and synthesis of published datasets. 

In this paper, we review existing trait databases and online portals, as well as initiatives for standardisation. We discuss current practice and the importance of data standards for trait-based research, and we identify current deficits in standardisation from a pragmatic view of data providers and data users. Based on these considerations, we propose a minimal structure and vocabulary for describing trait datasets, that builds upon and is compatible with existing terminology standards for biodiversity data. Finally, we present an R package that assists the harmonisation of trait data from distributed sources. 
With this easy-to-use terminology and toolset, we hope to  convince trait-data providers and trait-data users about the general importance of trait-data standardisation and lay out the roadmap towards an accessible ecological trait data standard.


<!-- FDS: Why 'ecological' trait data? what is special about this subject? Environmental context, community context, space and time, attributes of functional role in the ecosystem and energetic demands, interactions (not covered right now?), correlation and economic trade-offs in trait evolution -->

<!-- PM: Can I suggest a glossary of terms – this would be really helpful for people who are not data management specialists

FDS: see Garnier et al. but may contain: 

-->
\newpage

\tiny

Term | Definition 
----- | ---------
Term | A word that describes a particular *concept* as part of the specialised vocabulary of a field
Concept | An idea, notion or object that is made explicit in an information context by *name*, definition, *URI* or other reference (https://www.w3.org/TR/skos-reference/)
Controlled vocabulary | A list of *terms* that gives all valid consensus terms for a praticular context, while no unlisted entries are accepted
Terminology | The  body of *terms* and *concepts* used with a particular application in a subject of study, usually formalized in a *thesaurus* or *ontology*
Data standard | A published set of instructions and *terminologies* for storing and exchanging data content of a particular type (e.g. trait data), that is recognised by a large proportion of members of the application context
Thesaurus | *Controlled vocabulary* that provides key *terms* with their associated *concepts* for a specific field or domain of interest [@laporte13] 
Ontology | *Controlled vocabulary* that (opposed to a *thesaurus*) relates *concepts* to each other by cross-references, e.g. defines a hierarchy of terms; thus a formal model of the objects and their relationships in a domain of interest [@gruber95]
Semantic web | An extension of the world wide web that aims for machine-readable meaning of information via well-defined *data standards*, *ontologies* and exchange protocols [@berners-lee01]; the World Wide Web Consortium (W3C) defines standards, i.e. specifications of protocols and technologies for the semantic web (http://www.w3.org/standards/semanticweb/)
Dataset | A set of measurements and observations; often originating from a single experimental set-up or study context; can be considered as being internally homogeneous across all data entries
Data table | A two-dimensional spread-sheet containing data organised in rows and columns; in most cases these data are considered 'static', i.e. they are not altered or filtered across time
Database | A suite of data compiled from multiple *datasets*, i.e. from multiple study contexts or observation types; may take the form of a two-dimensional data table, but mostly is organised in into *relational databases* using database software; 
Relational database | Usage in this paper: Two or more *data tables* that are related by common information contained in one or more columns; common information is usually labelled by *identifiers* (IDs)
Online database | A *relational database* that is made accessible on the internet; offering forms for filtering and downloading subsets of the data; some online databases offer access via a *webservice* and an *API* that can be addressed computationally 
Identifier (ID) | A unique label that relates entries within and across *datasets*; is used to connect *data tables* into a *relational database*; can be user-specific or, as a *URI*, point to a globally valid *ontology* or *thesaurus*  
Uniform Resource Identifier (URI) | An unambiguous pointer to a unique resource on the internet; used to refer to single terms of a *thesaurus* or *ontology*; an example is 'http://t-sita.cesab.org/BETSI_vizInfo.jsp?trait=Body_length'
Webservice | An exchange protocol to access *online databases* directly and programmatically, i.e. by calls from a software tool
Application Programming Interface (API) |  A set of clearly defined methods of communication between  software components, e.g. client software and the *webservice* of an online databases; APIs are usually documented on the website of a database provider
Online Portal | A website designed as a platform for the exchange of information, e.g. trait data; a portal may include a communication forum, data upload forms, a database access point, and advanced user management for data access
File repository | A short-term storage of datasets or long-term archiving on *file-hosting services*; online repositories make data available for public access, provide *metadata* and (not always) facilitate citations via DOIs (Digital Object Identifiers) 
File-hosting service | An online platform that hosts *datasets* or entire *repositories* and provides access to a wide audience on the internet; examples in biology are Figshare.com, Dryad (datadryad.org), Researchgate.net, or Zenodo.org
Metadata | Data documentation of the higher level information or instructions; describe the content, context, quality, structure, provenance and accessibility of a data object [@michener97]
Darwin Core Standard (Dwc) | Body  of *terminologies* providing terms intended to facilitate the sharing of information about biological diversity (http://rs.tdwg.org/dwc/)
Darwin Core Archive (DwC-A) | A file archive, or *repository* structure, that contains *metadata* (specified using Ecological Metadata Language, EML) and primary data combined into a *relational database* via *identifier* columns. 
Method handbook | A listing of consensus methodology that is to be applied to acquire a particular measure, thus formalising the precise *concepts* of measures. 
Occurrence | A single observation instance of a taxon, i.e. an organism at a particular place at a particular time (http://rs.tdwg.org/dwc/terms/Organism)

Table: Glossary of terms from the biodiversity data-management context as they are used in this paper; draws from @garnier17. 

<!--
Measurement |  
Aggregated Value |
Value | 
Fact | 
-->

\normalsize

\newpage

# A review of initiatives for trait-data standardisation

In this section, we review four types of initiatives that are of relevance for trait-data standardisation (see Glossary in Table 1 for italicised terms):

1. Initiatives that provide *trait datasets* which have been assembled out of a particular research interest, either by measurement or collated from the literature. 
2. Initiatives that aim to harmonise trait data from the literature or from direct measurement into *trait databases* and make those data widely available. 
3. Initiatives that aim at the standardisation and development of consensus measurement methods and definitions for traits, and provides standard *terminologies* in the form of *thesauri* and *ontologies*.
4. Initiatives that aim to leverage *relational database* structures and *semantic web* technology to link trait data to a wider set of biodiversity data. 

We discuss these initiatives separately although often they are developed in conjunction to serve a particular database project, as for instance in the case of the TRY plant database [@kattge11; @kattge11a] and the  Thesaurus of Plant Traits [TOP; @garnier17]. 
We show how the degree of trait-data standardisation in existing datasets spans this entire spectrum and which tools and standards are applied to achieve harmonisation of data from multiple, distributed sources. 
The objective of this review is to raise awareness for the generic structure of trait data and aid researchers to share and publish own datasets in an appropriate form. 

## Trait datasets 

In the field of comparative biology, morphological traits related to plant flower, leaf and stem traits or bird wing and beak measurements, as well as life-history traits such as Ellenberg values for plants or ecological parameters of animals (e.g. reproductive traits, feeding biology, dispersal or body size) have been measured for decades, and have been published in regular journal articles or books. 
With the rise of ecological trait-based research, individual measurements and information available from species descriptions have been compiled into project-specific datasets that typically comprise a local set of taxa and a focal set of traits. 
A plethora of such static datasets has been published along with scientific articles or as standalone data publications [see @kleyer08 for a review on plant data; on animal data, see e.g., @gossner15; @ricklefs17].<!--  @sokol11;  @renner14; @vanderplas17; @price13; -->
Today, the online publication of such data is greatly facilitated by  *file hosting services* (e.g. Figshare.com), which warrant long-term accessibility and citability via DOIs, and support Public Domain dedication or Creative Commons licenses.
These platforms offer publicly accessible repositories at low-cost or for free, which makes them attractive for small and intermediate sized research projects that cannot dedicate extra resources for data management. 
However, although open for manual access, the trait datasets on data repositories might be stored in proprietary (e.g. .xlsx, .docx) or binary (e.g. .pdf) data formats which make a programmatical extraction tedious and dependent on commercial software, putting the long-term and open accessibility of these data at risk. 
Most importantly, these platforms enable public hosting of data with very low thresholds for *metadata* documentation and data standardisation. 
<!-- As a result, datasets greatly vary in structure and terminology. -->

For trait data, there are typical issues arising from the variability of data structures. 
For instance, the column descriptions and terminology applied to taxa and traits are mostly project specific, and rarely chosen to allow translation into larger database initiatives. Furthermore, metadata varies in its detail, e.g. for documenting descriptions of variables, measurement procedures or sampling context [@kattge11a]. 
In terms of structure, trait data usually are reported in a species$\times$traits wide-table format. In this format, each row contains a species (or taxon) for which multiple traits are reported in columns. 
Similarly, when reporting raw data, researchers place observations of individual organisms in rows with multiple trait measurements applied to the same individual across multiple columns. 
Variability in the number and meaning of columns in these *data tables* requires tedious manual adjustments when merging multiple datasets [@wickham14].

<!--To remedy these problems, data management initiatives that arise from scientific communities and projects, like the KNB (the Knowledge Network for Biocomplexity; https://knb.ecoinformatics.org/) or GFBio [the German Federation on for Biological Data; gfbio.org; @diepenbroek14], offer means for small projects to meet the high scientific criteria of data management.-->

A global overview of existing trait data for all taxa and trait types is difficult to obtain. Therefore, in an attempt to collate a list of existing distributed datasets, we initiated a living spreadsheet (https://goo.gl/QxzfHy) which lists published trait datasets, their regional and taxonomic focus, the number and scope of traits covered, their location on the internet and the terms of use (see Appendix A for a current excerpt of this list). We invite data owners and users to add further trait datasets to this spreadsheet.  
As it stands, the decentralisation and the lack of data standardisation of low-threshold online repositories renders the compilation of data into larger collections inefficient and reduces the potential of many published datasets to be re-used and combined into broad synthesis analysis.

## Database initiatives

<!-- community database backend, but static publication --> 
In the past two decades, many distributed trait datasets have been aggregated and harmonised into greater collections with particular taxonomic or regional focus [e.g. @klotz02; @iversen17; @kleyer08; @myhrvold15; @kissling14; @jones09; @oliveira17, see Appendix A table A1].<!--[@wright04; ] -->
While mostly concerned with issues of heterogeneity in units or factor levels, and aiming for high taxonomic coverage, few of these datasets apply a standardised terminology for taxa or traits that would allow them to be efficiently related to other databases.<!--and aiming for taxonomic completeness, the products of these initiatives employ a project specific structure and vocabulary. -->
Documentation of metadata and methodology differs in the level of detail, depending on the research focus of the initiative. 
Just as the individual datasets described above, many of these databases are published as static data tables on low-threshold file hosting platforms and are updated irregularly.

As they deal with much larger amounts of data, initiatives that form around natural history museum collections are more concerned with standardisation. 
Concerning organism traits, with the digitisation efforts that are currently undertaken in many museum collections [@vollmar10; @blagoderov12a], supported by citizen science crowdsourcing (e.g. www.markmybird.org), data on body measurements are likely to grow exponentially in the near future. 
For example, the VertNet database compiled and harmonized large quantities of vertebrate trait data with the aim of mobilising measurements from collections [@guralnick16]. The resulting data are published as versioned data tables which are updated as new data sources become available. 

<!-- database backend, selective download, user access management --> 
More specialised trait-database platforms have been created to cover certain trait types (e.g. floral traits, seed traits, root traits or wood density traits), interaction types (e.g. pollination traits or feeding relationships), or a specific environmental and experimental context of the trait observation (e.g. location or climatic data).
Such database initiatives attract data submissions from a defined research field and take care of the harmonisation process and thereby greatly facilitate data synthesis.
<!--For example, traits on crop plants have been collected in agronomy databases like  BetyDB [https://www.betydb.org/; @lebauer17].-->
For example, by aiming for a universal framework for plant traits, the TRY database [@kattge11]  attracted more data submissions and downloads than any other trait data platform. The *online database* enables selective data download and user permission and rights management. As a community effort, TRY serves as a network for consensus building on trait definitions [@garnier17] and measurement methodology [@perez-harguindeguy13] (see next section). 
Microbial ecologists also make frequent use of trait-based approaches to assess genomic function and describe functional diversity at the community level [@krause14; @fierer14;  @fierer12]. Here, 'operational taxonomic units' (OTUs) are derived from metagenomic analysis [@torsvik02; @langille13]. Databases are also used to interpret OTUs in terms of their functional role [e.g. the KEGG orthology, @kanehisa12]. 
For animals, a single unified platform and harmonising scheme for animal trait data is still lacking. The reason for this may be that harmonising trait data on animals, which span multiple trophic levels and possess diverse body plans, is a more complex task than for plants [@moretti17]. Nonetheless, initiatives for particular groups of animals, such as the BETSI database collects traits on soil invertebrates [http://betsi.cesab.org/; @pey14], and the Carabids.org web portal collects traits of carabid beetles (http://www.carabids.org/), already exist. 

Regarding open access, few of these centralised databases comply with the criteria demanded by journals and funding agencies for primary data publication. The platforms incentivise data submissions by offering increased data visibility and usage, while providing data use policies that secure author attribution and potentially co-authorship.
With the proactive turn towards open access data (as stated in the Bouchot Declaration; http://www.bouchoutdeclaration.org/), it may be necessary to find other incentives for data submission. 

<!-- AO: In a sense of data availability to the public : yes. But do we discuss it here? We want to show how to make trait data sharing easier due to better data format, or?Not the topic of this article to discuss open vs. non-open data, or?
bKR: I am not quite sure what point you are trying to make it, the paragraph sort of goes back and forth between advantages and drawbacks of the different approaches. 
FDS: I think its part of the rationale for a simple data standard. More open data mean that we need more awareness for data standardisation. 
 -->

## Thesauri and Ontologies for traits

A major challenge in trait-data standardisation is the lack of widely accepted and unambiguous trait definitions. Previous standard definitions of trait *concepts* range from listings of selected definitions in *glossaries*, over well-defined methodological *handbooks* and comprehensive *thesauri*, to relational definitions of trait concepts in *ontologies*. 
While glossaries may be seen as specific for a study context, the initiatives behind method handbooks, thesauri and ontologies are primarily concerned with consensus building on trait definitions in a wider community. 

Very general classes of traits are defined within the list of GeoBON Essential Biodiversity Variables [@pereira13].
Assigning a more detailled and unambiguous methodological protocol to a trait, including the units to use or the ordinal or factor levels to be assigned, is key for standardising the physical process of measuring. Efforts to develop handbooks for measurement protocols provide such a methodological standardisation for plants  [@cornelissen03; @perez-harguindeguy13] or invertebrates [@moretti17], but obviously are of limited use in harmonising trait data that predate or ignore this standard [@kattge11a].

A thesaurus provides a "controlled vocabulary designed to clarify the definition and structuring of key terms and associated concepts in a specific discipline" [@laporte13; @garnier17]. 
Expanding on this, ontologies link the defined terms by formally defining the relationships between them, with the objective of enabling a computational interpretation of data. 
Being publicly available, it is also possible to refer to these defined terms via globally unique *Uniform Resource Identifiers (URIs)* within own datasets. For example, a measurement of seed size could be linked to the Planteome Trait Ontology (TO) definition of 'seed size' by referencing 'http://browser.planteome.org/amigo/term/TO:0000391'. 
Ontologies define terms based on other well-defined terms from published ontologies. The TO definition of the concept 'seed size' contains references to other globally defined terms: "A seed morphology trait (TO:0000184) which is the size of a seed (PO:0009010)." 
Furthermore, trait definitions may refer to related terms or synonyms defined in other trait ontologies or other scientific ontologies, like units as defined by the Units of Measurement Ontology [@gkoutos12]. 
This way, each trait definition may link to a broader or narrower term. For example, the definition of 'femur length of first leg, left side' is narrower than 'femur length' which is narrower than 'leg trait' which is narrower than 'locomotion trait'. 
By providing this interlinkage of trait ontologies, a machine-readable web of definitions is spun across the Internet which allows researchers and search engines to relate independent trait measurements with each other and connect it to the wider *semantic web* of online data [@berners-lee01; @page08]. 
The distinction of thesauri and ontologies is not truely binary. Rather they mark idealised ends of a spectrum. While thesauri may contain defined relations between terms within the standard, ontologies relate most terms to other defined concepts, and also link those to other standards. 

Comprehensive trait thesauri have been developed in the TOP Thesaurus of plant traits, which is employed in the TRY database [@garnier17], and in the Thesaurus for Soil Invertebrate Trait-based Approaches [T-SITA, http://t-sita.cesab.org/, @pey14]. 
Ontologies of trait definitions have been developed for plants [e.g. the Plant Ontology, @jaiswal05; the Flora Phenotype Ontology, @hoehndorf16], as well as for animals [e.g. the Hymenoptera Anatomy Ontology, @yoder10; the vertebrate trait ontology, @park13]. 
The existing thesauri and ontologies for traits differ widely in terms of hierarchical depth and detail, as well as in curation efforts and measures for peer-reviewed quality control. 

<!--Thus, in its simplest form,  a trait thesaurus would assign trait names with A) an unambiguous definition of the trait and B) an expected format (e.g. units or legit factor levels) of measured values or reported facts. It might additionally provide semantic relationships between terms for deriving a hierarchical or tree-based classification of traits. --><!-- BKR: Isn’t that the same? Or can you have multiple parents for one node? FDS: agree. simplify phrasing. -->
 
Meta-ontology initiatives, like Planteome.org, offer access to multiple published ontologies and build platforms for their collaborative development [@walls12]. 
For general biodiversity data, the OBO Foundry (http://www.obofoundry.org/), Ontobee (http://www.ontobee.org/), Bioportal (https://bioportal.bioontology.org/), or the GFBio Terminology service (https://terminologies.gfbio.org/), provide centralised hosting for advanced trait ontologies and offer webservices for computational access.

To conclude, there is already a suite of globally available thesauri and ontologies for traits that emerged from standardisation efforts of methodologies and community consensus processes. However, definitions in some domains are better covered than others. Interlinkage and accessibility of ontologies can be much improved to fulfil semantic web standards.
Most importantly, while these defined vocabularies are widely used in biodiversity data management, distributed data repositories of smaller project contexts hardly make use of them. 
A more widespread implementation of ontologies would advance the possibilities to aggregate datasets into databases and reduce noise and uncertainty. 
To achieve this, the use of ontologies and thesauri must be incentivised and facilitated for individual researchers.
For example, the accessibility of ontologies will increase if open *Application Programming Interfaces (APIs)* are provided as a way to extract the definitions and higher-level trait hierarchies programmatically via software tools. 
Software then can assist researchers in linking own data to globally defined concepts.


## Trait-data structures for the semantic web

While trait thesauri and ontologies typically define traits for focal groups of organisms, they do not specify the  format or structure in which trait data should be stored and linked to further standard terminologies, such as standard taxonomy nomenclatures.  
To make sense of trait data in the context of more general databases, a consensus definition of trait data is necessary.  

Trait data have been defined by @garnier17 to follow an entity-quality model (EQ), where a trait observation is 'an entity having a quality'. 
More specifically, a trait dataset contains information on quantitative *measurements* or qualitative *facts* (i.e. trait values) describing the physical phenotypic characteristics relating to fitness and performance (i.e. traits) observed on a biological entity (i.e. an individual specimen, or parts of an individual specimen) that can be assigned to a biological taxon (i.e. a species or higher-level taxon). 
We are expanding on this definition: quantitative measurements are values obtained either by direct morphological, physiological or behavioural observations on single specimens (Fig. 1*a*), by aggregating replicated measurements on multiple entities (Fig. 1*b*) or by estimating the means or ranges for the respective taxon as reported in the literature or other published sources (e.g. databases, Fig. 1*c*). Qualitative facts are assignments of an entity to a categorical level, e.g. of a behavioural or life-history trait (Fig. 1*d*). 
The entity or observation (i.e. the *occurrence*) to which the reported measurement or fact applies may differ in organisational scale -- depending on the scientific question -- and could be a sub-sample or bodypart, an individual specimen, an entire species or a higher-level taxon (e.g. a genus).  

\begin{figure}[t]
\centering
\includegraphics[width=0.85\textwidth, angle=0]{figures/fig1.png}
\caption{Types of ecological trait data assume different entities or reported quantities. a) morphometric or morphological measurements of individual body features (lengths, areas, volumes, weights) or other quantities related to life history (e.g. reproductive rates, life spans); b) aggregated traits are reported as means taken on multiple measures of members of a taxon; c) quantities may be extracted from literature or existing databases, referring to the entire taxon (or a subset, e.g. a sex) as the entity of description; d) Qualitative traits are categorical or binary descriptors of the entire species or higher taxonomic group.}

\end{figure}
<!-- DF : The “variables” for the taxa etc should only be reused for the different examples when they refer to something similar: “a) Entity: an individual X of taxon Y”, “b) Entity: taxon Y”, “c) Entity: males of taxon Z” and “d) Entity: genus U”. Alternatively, since these are already example, why use variables here in the first place and not just specific example for the taxa as well: “a) Entity: an individual X of taxon Tettigonia viridissima” -->

These relationships between a trait observation and an individual organism as an occurrence of a particular taxon have been formalised in the schema for biological collection records [ABCD Schema; @holetschek12] and the Darwin Core Standard for biodiversity data [DwC; @wieczorek12]. For example, the Global Biodiversity Information Facility (GBIF, www.gbif.org) applies these terms.
These frameworks specify terms and classes to describe the general structure of biodiversity databases, for example by defining names for columns that contains measurement values, units, taxon names, variables such as sex or life stage, ancillary information of time and date of observation, and methodological details. The terminologies provided by these standards are quite universal and even cover most use cases of trait data. An entire ecosystem of data standards links to and expands the capacities of DwC [@wieczorek12].
<!--  DF: am not sure, where to best write this comment, but I think it belongs somewhere in this chapter. The Trait Data Standard is a standard on how to store trait data measurements. In this chapter you mention what databases for trait data there are (usually specific for certain taxa) and what ontologies there for the different traits, however it is missing the section on how trait data measurements have been and can be stored prior to your standard. For instance the ABCD standard has a huge section for MeasurementsAndFacts. And while it certainly is not primarily made to capture trait measurements, it can certainly be used for it (at least on a unit level, combining measurements for higher taxa is not possible). Darwin Core has similar, though not as extensive, fields for measurements, as they were modelled based on ABCD. While I agree that your standard is useful and necessary, it certainly is not the first standardized way to describe trait measurements.  -->

Specifically designed for plant traits, @kattge11 proposed a generic database structure that covers most potential use cases of trait-based ecology. This data structure is built around a central data table that contains observations, i.e. a single event of measurement on the same individual plant specimen at the same point in time. This structure emphasises the fact that multiple trait data are measured on the same individual organisms and used to analyse correlations between these multiple traits. Identifiers link the measurements (qualities) to the same observation (entity), each measurement being well defined by additional standard tables. The observations are also linked to a taxonomy and ancillary descriptors of the observation context, like location or experimental treatment. This structure can be implemented in any relational database management system. 

In a similar vein, the Encyclopedia of Life (EOL) project has proposed the database framework TraitBank [@parr16] for major physiological and life-history traits of all kingdoms of life, which is to date the most general approach of an integrated structure for trait data.
The framework employs established terms provided by the DwC, relates trait definitions to trait ontologies for phenotypic or anatomical terms, and maps taxa to global identifiers in taxonomic hierarchies of name service providers to capture synonyms, misspellings and controversies [@parr16, http://eol.org/info/cp_archives]. 
Additional layers of information capture bibliographic reference, multimedia archives and ecological interactions. 
TraitBank invites data submissions to the EOL database in a structured Darwin Core Archive [DwC-A, @robertson09a], a zip-file with annotated text-files that is also preffered for observation data in GBIF [@gbif17,  http://tools.gbif.org/dwca-assistant/]. 
The archive also integrates the general framework for metadata of the Ecological Metadata Language [EML, @knb11]. 
The difficulties with keeping taxonomic references intact along with continuous changes in taxonomy consensus are a central challenge of biodiversity data management and are beyond the scope of this review [@franz16]. Initiatives that aim at providing a stable reference for taxa are for instance the EOL Catalogue of Life [http://www.catalogueoflife.org/, @roskov18], the GBIF Backbone Taxonomy [@gbifsecretariat17], or the EDIT Platform for Cybertaxonomy (https://cybertaxonomy.eu/). 

These proposed standards are responses to a demand from biodiversity data managers for more structured input from the research community. 
However, hardly any of the aforementioned trait datasets for birds, amphibians, or mammals employs such ontologies or semantic web standards. 
One reason for this is most certainly complexity: the data structures are designed for multi-layered, relational databases rather than for standalone datasets for which a two-dimensional data table may suffice. In the eyes of the data-provider, in most cases, ancillary co-factors can be appended as extra columns to the dataset. 
<!--If the traits are assembled for a specific research question with just few ancillary co-factors, it may be safer to append these as columns to the two-dimensional data table, thus facilitating an analysis of correlations between traits and co-factors further down the road. This way, if datasets of different sources are merged, the related information is readily available without the risk of breaking the reference to an external datasheet. -->
The other reason is lack of awareness for the need for trait-data standardisation among data providers: many providers are not trained in the demands of biodiversity data-management and complying with what may be non-intuitive data structures is an investment without clear incentive or immediate pay-off, and hardly affordable for small and intermediate-size research projects. 

By filling this gap, data-brokering services [e.g. the German Federation for Biological Data; gfbio.org; @diepenbroek14] or data management systems for scientific projects (e.g. BExIS, http://bexis2.uni-jena.de/; Diversity Workbench, www.diversityworkbench.net; KNB and its open-source database back-end Metacat, https://knb.ecoinformatics.org/) are likely to gain importance. These services simplify and direct the standardised upload of research data and descriptive metadata into reliable and interlinked data infrastructures. One goal of such initiatives is to facilitate data publications and standardisation for researchers, for instance by providing terminologies and ontologies for biodiversity data, and by consulting on publication licenses. <!-- AG:  EDIT Platform for Cybertaxonomy (cybertaxonomy); NEON Dataportal http://data.neonscience.org/home;   -->

## Conclusion of review

Initiatives for standardisation (e.g. ontologies and data standards) and platforms for data management (e.g. database and data management platforms) provide great visibility and improve interconnectedness of datasets, but raise relatively high thresholds for data and metadata preparation. 
Low-threshold repositories offer the hosting of scientific primary data attracting a wealth of heterogeneous trait datasets, but data harmonisation of these distributed data sets is currently laborious. 
The goal must be to better integrate these distributed data into the global biodiversity data-management ecosystem by creating awareness for data standardisation on the side of data providers. 
We propose the development of  tools and vocabularies that impose low thresholds and offer high payoff in the visibility and interconnectedness of published data. 

<!--
Depending on the research objective, trait-data are stored in a range of different table formats. In the subsequent paragraphs, we discuss the most common formats  in relation to the use cases in trait-based ecology. 

### a) Species wide-table

<!-- AO: You argue somehow that matrix format I a result of lit. review and so on. I would not agree with it. Your example also “just” points to aggregated data which menas one row is on species (aggregated) value.
Your example is (for me) not really a matrix. In a matrix you have a relation between two variables. In the first column and and on the upper row normally the values of one variable are written. An example would be species x body length. Then for body_length some classifications are written like (0-1, 1-2, 2-3,…) Then in the matrix the observation/abundance/… is noted. E.g. spec1 x “0-1” = 5, spec1 x “1-2” = 3, and so on. Your wexample hat already different variables (=traits) with different units as columns. That is a meaningfull representation of variables and values in a normal data table. Maybe it is good her to show the enormous number of different investigated traits and point to the resulting big (many columns = wide) tables.
--><!--
If trait data have been collated at the species level from different literature sources or from expert knowledge, or as aggregated measurements collated from raw data, they usually are reported in a species$\times$trait wide-table format (Fig. 2*a*), with a column of values for each trait recorded and a row for each species (or taxon) for which data were available. 
The wide-table format is widely used for the production of lookup-tables at the species level, which may either report qualitative facts (from literature or  expert knowledge) or aggregated quantitative traits (averages or ranges of trait values for this taxon). <!-- AO: You say it already 2 sentences before --><!--
This format needs to clearly differentiate missing data and traits not applicable to a given taxon (often filled in as 'NA' or dummy numbers, e.g. '999'). It may store additional information (e.g. on variation of means or literature source) in secondary columns or accompanying datasheets. 

Quantitative traits reported at this resolution are often used to calculate community weighted means or functional diversity metrics at the community level [@laliberte10a]. Also at this level of information, missing entries, e.g. about the trait values of a rarely described species, may be inferred from these tables when combined with phylogenies ['gap filling', @swenson14; @penone14; @taugourdeau14] and using Bayesian approaches [@schrodt15]. <!--However, these matrices do not allow estimates of within-species variation or assessments of the quality of the reported measurement or fact [@guralnick16]. --><!--AO: Why not? I can filter for a specific species and get all respective rows. But this is not a result a using a data format, it is a result of the data itself, FDS: no, since only one row per species exists; true that data itself don't cover this variation. -->
<!--

### b) Occurrence wide-table

In these occurrence wide-tables (Fig. 2*b*), each row is centered around a single physical occurrence of a species, e.g. phenotypic variation of traits within species would usually be recorded per observation where each replicate taken on a distinct individual specimen would be recorded in a row, with the respective trait values stored in columns. Those data are common in investigations of evolutionary trade-offs and trait correlation, or of intra-specific variation along environmental gradients. Occurrence data would also capture phenotypic variation arising from morphotypes and sexual dimorphism. This format is the most intuitive for recording own empirical measurements and therefore is common for measured quantitative data but rarely found for reported qualitative facts obtained from the literature.

### c.) measurement long-table

Both wide-table formats report the different trait values stored in multiple columns. Each trait column represents one value that applies to the entity in question (i.e. to a species  or an occurrence). As the column names and the width of the table varies with the number of traits included, merging data from different sources requires user-defined mapping and manual harmonization of these structures. 

A computationally more effective method that allows for high flexibility is the storage of traitdata in long-table formats [@wickham14; @parr16]. In these, each row is reserved for a single measurement or fact of a specific trait, referenced to a single specimen  (i.e., one occurrence)  which is assigned to a specific taxon. 
<!-- FDS: also refer to kattge 11 -->
<!--AO: Useful for data sharing, ease of data harmonization, aggregation: probably yes. But not necessarily computational more effective at all. E.g. I start to mix up numeric values with descriptive values. I start to mix up different units. Now it is more complicated to retrieve associated values (like all length measurements in mm) again to make e.g. quality stuff. Ergo less effective from that perspective. But it has some advantages which may overrule dis. See below 
AO: For the whole proofing why to use such a long format I think that GBIF and DwC has already given some sentences. I strongly think their work should be considered and can be used as reference!
We can simply argue that a well established routine should now also executed on trait level!
Furthermore for DwC exists also the “extended measurement or facts extension” which is useful for putting e.g. traits to species.

--><!--
This allows repeated measurements on a single individual to be stored by linking the data from separate rows via a unique specimen identifier. Also, multivariate trait measurements can be recorded in this format by linking multiple rows via a unique measurement identifier.  Long-table datasets purport multiple advantages for data manipulation (e.g. filtering, sub-setting and aggregating data), visualization (e.g. plot measured values by factor variable or taxon) and statistical modelling (e.g. ANOVA for testing difference of trait value by sex) [@wickham14]. As long-table fromats draw from a defined set of columns,  merging of datasets is much easier. 
<!-- BKR: Is that really true? Of course, you have less problems on the schema or structure level. But, on the other hand, the differences (e.g. different terms for the same trait,…) now happen on the instance level (that is – in the data; in the rows). Thus, at least with respect to integration, you just move the problem from one level to the other.  To ease integration, one would need to agree on the entries in the features column
AO: Maybe describe some more advantages here like: it is possible to ensure that the items/values are always taken from a provided lookup-table e.g. for taxon, trait_names, units, more?
We may have something like a dimension, the data source (measurement, lit. review, …) and some more criteria which are beneficial for researchers  and data aggregation?
Again: I think we find here papers regarding DWC and GBIF (and OBIS) as source of proofing the facts.
-->
<!--
The long-table format therefore is ideal for storing raw data of trait measurements in huge databases, and allows the entries to be linked via the respective unique identifiers to additional information. Each row of the dataset can therefore be interpreted as a statement of an 'entity x having a qualitative/quantitative feature y' (Fig. 2*c*). 

<!-- Cat: here I guess that a figure showing few rows of a species x trait matrix and a long-format one with links to other information (metadata,..) woud help quite a lot. By experience some users are not be familiar with the interest of using the long-table format.-->
<!-- Malte: I agree (see my comment above), such a figure / would definitely help convince people (or at least not to lose them). -->
<!--
\begin{figure}[t]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig2.png}
\caption{Formats used for trait datasets.  a) taxon-level trait data compiled from literature or aggregated from measurements are often published as a compiled species $\times$ traits matrix; b) wide-tables report measurements on the specimen-level, i.e. one physical occurrence of a taxon; c) a measurement long-table is a well defined and tidy data format, reporting one single measurement per row and relating it to a trait definition and taxon via unambiguous identifiers. Additional identifiers relate each row to other layers of information on d) the specimen (occurrence) or e) the origin or confidence on the reported measurement or fact.}

\end{figure}
-->

<!--The trade-off is data-consistency and readability *vs.* avoidance of content duplication. -->

<!--When traits are assessed on a higher level of organisation, e.g. microbial functional traits assessed at the community level taken from a soil sample, key data columns would be placed in the same table as the core data are traits. Here, location or measurement information are the primary focus of the investigation and must be contained in the . 
<!-- PM: I get lost here-->
<!--A general recommendation, whether a column describes ancillary data or is part of the central dataset is ill advised. Therefore, our glossary of terms and its extensions should be used to describe the scientific data according to the study context. -->
<!-- FDS: shorten this and refer to existing standards for database use. the star-structure of ABCD, Kattge et al , and Garnier .-->
<!--while the existing trait data standards may cover all of the aspects required in the various use-cases of trait-based research. <!-- examples for blind spots; PM: Vague- not sure what is missing here- be more specific --> 

# An ecological trait-data standard vocabulary

As a response to the challenges outlined above, we propose a versatile vocabulary for trait-based ecological research.
The aim of the vocabulary is to cover the variety of trait-based approaches and their different degrees of measurement detail. 
Rather than describing a data structure for relational databases, the vocabulary is intended as a more inclusive terminology, that can be used in simple two-dimensional datasets as well as in the exchange of data between web services in the semantic web.
By using this standard vocabulary, authors can ensure that the description of trait measurements that are uploaded to distributed data repositories will be unambiguous and generally applicable. It will facilitate re-use of data for future data aggregation initiatives and data synthesis and ensure long-term accessibility.

In designing this vocabulary, we drew on the combined expertise of empirical biodiversity researchers (data providers), biodiversity synthesis researchers (data users), and biodiversity informatics researchers (data managers).
We paid particular consideration to the work of @kattge11, @kattge11a, and @garnier17, as well as @parr16 to ensure compatibility of our proposed data structure with major trait databases and existing standards for biodiversity data management. Here, the use of identifiers ('IDs') for the individual measurement observations ('measurementID'), specimens ('occurrenceID'), sampling events ('eventID'), or taxa ('taxonID') is key to map two-dimensional data onto the structure of relational databases. 
Besides being used for the publication of datasets, the standard vocabulary could be imposed in webservices or download tools, e.g. APIs that provide direct access to online databases. 
The vocabulary proposed is intended to form the foundations of a standard nomenclature that can be expanded and corrected by the wider community of researchers using trait-based approaches in ecology.
<!--
## Why use the vocabulary

The incentives for using a unified vocabulary for trait data might not be obvious for data providers, who are mainly concerned with a particular study site, organism group and research question. We won't discuss here how sharing trait data will improve science on broader questions and larger scales, by providing a data basis for downstream analysis. However, it appears to be broad consensus that an open biodiversity science is crucial for an evidence-based decision making and conservation policy on regional and global scales. 

Instead, this section points on the technical aspects and advantages of readying own trait data to link with the 'semantic web' of biodiversity data and knowledge. 
By using a constrained vocabulary with globally accessible definitions of terms, data can be accessed more easily by other researchers and datasets from different sources can be combined into aggregated datasets.
Also it will be easier to develop computational methods and software tools that access and handle the data, based on the standard vocabulary. -->


## How to apply the standard vocabulary

We suggest that any trait dataset that is published on online repositories should draw its column names and field entries from the defined vocabulary where possible.
The core vocabulary lists and defines terms that describe a dataset according to the Entity-Quality model described above [@garnier17]: each entry describes a trait value (i.e. quality) observed on an individual or population (i.e. entity), of a biological taxon.
When applying the vocabulary, it is implicit to use a two-dimensional observation long-table format for the data (Fig. 2 *b*), rather than a species$\times$traits matrix (Fig. 2 *a*). As the long-table format draws from a defined set of columns, merging datasets is easier. Long-table datasets also purport multiple advantages for data manipulation [e.g. filtering, sub-setting and aggregating data, @wickham14]. 

Well-defined identifiers ('IDs') are key elements to structure the datasets and relate them to complementing datasets, if necessary (Fig. 2 *c* & *d*).
For instance, for occurrence level data where multiple trait measurements are reported for each individual specimen, the same user-defined entry for 'occurrenceID' would link several measurements across the rows of the dataset. 
Similarly, multivariate measurements, for instance gas chromatography data or x-y-z data of morphometric landmarks could be linked via a 'measurementID'. 
In literature data, summarized traits are usually given at the taxon level instead of the individual organism (e.g. reported as means or factorials) and a 'taxonID' is the key identifier.
In larger compilations, a 'datasetID' allow to trace data origin to the primary source. 
Beyond being just of structural use for the dataset, identifiers are capable of linking own data to consensus taxonomy and trait terminology via Uniform Resource Identifiers (URIs), which point to external terminology services (see above for resources). 
Two-dimensional spreadsheets are however limited in the number and complexity of co-variates they can contain. As such, for datasets containing multi-layered information on observations, traits, taxa and environmental context, the use of relational datatabase structures may be indicated, like the generic trait database structure proposed by @kattge11a or the TraitBank structure proposed by @parr16. 
The trade-off is user-side readability and handling in a single table *vs.* avoidance of content duplication and redundancy in a relational database. 
The standard vocabulary proposed here may still be applied to describe columns within the individual data tables of relational databases. 

\begin{figure}[t]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig2.png}
\caption{Formats used for trait datasets.  a) taxon-level trait data compiled from literature or aggregated from measurements are often published as a compiled species $\times$ traits matrix; b) observation long-tables are a well defined and tidy data format, reporting one single measurement per row and c) relating it to a standard trait definition and accepted taxon name using unambiguous identifiers. Additional identifiers relate each row to other layers of information on d) the taxon resolution, the specimen (occurrence) or the origin or confidence on the reported measurement or fact.}

\end{figure}

For reasons of long-term accessibility, data should not be uploaded in proprietary spreadsheet formats (like '.xlsx') but rather in comma-separated text files  ('.csv' or '.txt') that are compatible with all computing platforms and internationalisation settings by applying a unified character encoding (e.g. UTF-8 or ASCII).

In order to ensure traceability, the metadata of any dataset that employs this vocabulary should refer to the specific online version that was used to build the dataset, e.g. "Schneider, F.D., Jochum, M.,  LeProvost, G., Penone, C., Ostrowski, A. and Simons, N.K., 2017 Ecological Traitdata Standard v0.6, DOI: 10.5281/zenodo.1048956, URL: https://ecologicaltraitdata.github.io/ETS/v0.6/". In addition to this versioned online reference, the dataset should also cite this paper for an explanation of the rationale.
Wherever referring to individual terms of the vocabulary in publications or metadata, this should be done via their global identifiers, which will be hosted by the GFBio Terminology Service [@karam16, https://terminologies.gfbio.org/] and can be accessed programmatically (i.e. via the API; **in preparation!**). 
Wherever our glossary refines or duplicates existing terms from other ontologies for biological data, like the Glossary of EOL (http://eol.org/info/516) and Darwin Core (http://rs.tdwg.org/dwc/terms/), we indicate this in the fields 'refines' or 'identical', respectively.

## Terms of the standard vocabulary
  
The online reference to the standard vocabulary can be found at https://ecologicaltraitdata.github.io/ETS/.
The core terms describe minimal trait data according to the Entity-Quality model. 
Beyond these core observations, further information might be available that are related to the taxonomic assignment, or that put the reported fact, measurement or sampling event in a broader observation context (including geolocation and date information). These information can be useful for future analysis of the causal reasons of trait variation and should always be published along with the core data. For this case, we offer three extensions of the core vocabulary ("Taxon", "Measurement or Fact", and "Occurrence") that expand and refine terms of the Darwin Core Extensions (see below). This information may simply be added as extra columns to the core dataset.
Additional terms are provided for metadata and for lists of trait definitions (i.e. trait thesauri, see below). 
The scope of the vocabulary may not yet cover all aspects of morphological and evolutionary perspectives. 
Also, information about interactions between species are not within the scope of the Entity-Quality Model, but may easily be combined with trait data by using other Extensions of the Darwin Core.
Therefore, we invite researchers to contribute to the next iterations of the standard vocabulary and develop own applications and ontologies that interact with it. 


### Specification of core terms

To qualify as trait data according to the definition provided above, where each row is the reported measurement or fact for a single observation, the following columns are required at minimum  (Fig. 2 *b*):
1. a value (column `traitValue`) and -- for numeric values -- a standard unit (`traitUnit`); 
2. a descriptive trait name (`traitName`) that links to a well-defined definition; 
3. the scientific taxon name for which the measurement or fact was obtained (`scientificName`). 
For these core values, unambiguous and self-explanatory vocabularies for trait names and taxa are recommended. 
However, to ensure compatibility with existing databases or analytical code, it might be necessary to use abbreviations or user-specific identifiers for `scientificName`  and `traitName` instead. 
<!--### Relate to global taxon and trait definitions -->
In this case, it is essential to relate the user-defined names to a consensus standard of taxon names as well as a look-up table of traits. 
This is achieved by adding globally valid Uniform Resource Identifiers (URIs) for taxon (`taxonID`) and trait definitions (`traitID`), complemented by the human-readable verbatim accepted names (`ScientificNameStd` and `traitNameStd`, respectively). 
For example, referring to GBIF Backbone Terminology, for *Bellis perennis*, the taxonID would be  'https://www.gbif.org/species/3117424'; the `traitID` for 'fruit mass' according to TOP Thesaurus of plant traits would be  'http://top-thesaurus.org/annotationInfo?viz=1&&trait=Fruit_mass'.  
By allowing for a double record of both user-specific and standardised entries, we acknowledge the fact that most authors have their own schemes for standardisation which may refer to different scientific community standards  [as practiced in TRY; @kattge11]. This redundancy of data allows for continuity for data owners while also ensuring quality checks and comparability for the data user.

### Extensions for additional data layers

Beyond measurement units or higher taxon information, further information might be available that may not be core data, but are related to the individual or specimen, or to the reported fact, measurement or sampling event. 
The data standard provides three extensions of the vocabulary that should be used to describe this information (Fig. 2 *d*):

- The `Taxon` extension (http://ecologicaltraitdata.github.io/ETS/#extension-taxon) provides further terms for specifying the taxonomic resolution of the observation and to ensure the correct reference in case of synonyms and homonyms.
- The `MeasurementOrFact` extension (https://ecologicaltraitdata.github.io/ETS/#extension-measurement-or-fact) provides terms to describe information at the level of single measurements or reported facts, such as the original literature from where the value is cited, the method of measurement or statistical method of aggregation. It provides important information that allows for the tracking of potential sources of noise or bias in measured data (e.g. variation in measurement method) or aggregated values (e.g. statistical method applied), as well as the source of reported facts (e.g. literature source or expert reference).
- The `Occurrence` extension (https://ecologicaltraitdata.github.io/ETS/#extension-occurrence) contains vocabulary to describe information on the level of individual specimens, such as sex, life stage or age. This also includes the method of sampling and preservation, as well as date and geographical location, which provides an important resource to analyse trait variation due to differences in space and time.

Many terms of these extensions refine or copy terms of the DwC and their own [Taxon](http://rs.tdwg.org/dwc/terms/index.htm#Taxon),  [MeasurementOrFact](http://rs.tdwg.org/dwc/terms/index.htm#measureindex) and [Occurrence](http://rs.tdwg.org/dwc/terms/index.htm#Occurrence) extensions and EOL TraitBank's use of those terms (http://eol.org/info/structured_data_archives).
<!-- If no taxonID has been provided, in order to be unambiguous about the taxon referred to by the trait observation, it is highly recommended to keep an additional reference to higher taxonomic hierarchies (using columns `kingdom`, `order` etc.) to avoid taxonomic homonyms, and provide the authority (???) and key of reference (???).  
Furthermore, it is best practice to keep a record of the reported taxon rank (in column `taxonRank`) for traits recorded at the family or genus level. This information can then be used for trait inference at higher taxonomic resolution. <!-- using hierarchical probabilistic matrix factorisation [@schrodt15]. -->
These additional layers of information can either be added as extra columns to the core dataset or kept in separate data sheets (published separately or as part of a Darwin Core Archive), thus avoiding redundancy and duplication of content. A unique identifier would link to these other datasheets, encoding each individual occurrence of a species (`occurrenceID`), single measurements or reported facts (`measurementID`), locations of sampling (`locationID`) and sampling campaigns (`eventID`). 
Some data-types may directly refer to existing global identifiers for occurrence IDs, e.g. a GBIF URI or a museum collection code references the precise specimen from which the measurement was taken [@groom17; @guntsch17].

### Specification of traits

Wherever possible, the column `traitID` should point to a public trait definition, ideally as part of an unambiguous ontology according to semantic-web standards. 
If no globally available trait definition exists as a reference, trait datasets should always be accompanied by a dataset-specific glossary or thesaurus of traits. Such a 'controlled vocabulary' would, in its simplest form, assign trait names with an unambiguous definition of the trait and an expected format  of measured values or reported facts (e.g. units or legit factor levels). 
By providing a minimal vocabulary for trait thesauri (see https://ecologicaltraitdata.github.io/ETS/#terms-for-thesauri), we hope to facilitate the publication of trait thesauri developed for a specific project context, which can be referenced in the trait dataset. The thesaurus might accompany the core data in a separate file (e.g. within a Darwin Core Archive), or be published on any other stable internet address (ideally providing stable URIs). 
<!-- PM: Is this possible in practice?- in plant traits slightly different methods makes some traits comparable conceptually but not in practice, and there are lots of very minor variations of similar traits- see the vast list of traits in TRY for example -->
See our online reference for recommendations how to publish a minimal thesaurus compliant with semantic web standards (https://ecologicaltraitdata.github.io/ETS/thesauri.html). We also provide an example thesaurus on Github (http://ecologicaltraitdata.github.io/TraitDataList/), which has been developed to capture traits assessed for multiple animal taxa within the German Biodiversity Exploratories project [@fischer10]. 


### Specification of Metadata terms

Information about the authorship and ownership of the data and the terms of use should be considered when sharing and working with trait datasets.
We define a vocabulary (https://ecologicaltraitdata.github.io/ETS/#metadata-vocabulary) that allows trait data to be related to authors and owners, while also stating a bibliographic reference and license model. 
In the case of primary measurement data, this information applies to the entire trait dataset, and should be stored along with the published data as metadata (e.g. in a separate metadata file, possibly applying the ecological metadata language, EML). 
In cases where individual data from different sources are compiled into a trait database, these information must be provided at the measurement level. This can be achieved by appending the information as columns to the core dataset, or via an unambiguous `datasetID` and a descriptive `datasetName`. 

<!--Open access to science data from publicly funded research is a declared goal of international and national research organisations [@emerson15; @allianceofgermanscienceorganisations10; @theroyalsocietysciencepolicycentre12] and increasingly required by journals and funding agencies to warrant reproducibility of science and foster public benefit. 
Also, factual data such as trait measurements, being not a product of a creative act, do not qualify as intellectual property and copyright does not apply [@carroll15; @creativecommons].
However, there is still no broad consensus achieved in the scientific community whether data from public funding must be generally dedicated to the public domain, or whether the investment in production must be acknowledged by attribution or co-authorship. -->
<!-- cP: Maybe we should put in the appendix an example of dataset+metadata? This should probably go in the metadata section 
FDS: to consider!! -->

# Computational tools for producing compliant data

To access data from public databases, the R-package 'traits' [@chamberlain17] contains functions to extract trait data via several open API interfaces including Birdlife, EOL TraitBank or BetyDB. The package  'TR8' provides similar access to plant traits from a list of databases [including LEDA, BiolFlor and Ellenberg values; @bocci15] and aggregates them into a species$\times$traits matrix. However, none of these packages provide the option to harmonisation trait data into a unified scheme. 
To close this gap, we developed the R package 'traitdataform', which assists the production of data compliant with the trait data standard proposed above. 
There are two major use cases for the package:

1. preparing trait datasets for publication on public hosting services and project databases, and
2. automating the harmonisation of trait datasets from different sources by moulding them into a unified format. 

A comprehensive documentation of the package can be found on its Github repository (https://github.com/EcologicalTraitData/traitdataform) and the documentation website (http://EcologicalTraitData.github.io/traitdataform/). 
The package is under continuous open source development and invites participation in development, comments or bug reports via the Github Issue page (https://github.com/EcologicalTraitData/traitdataform/issues). 

The key function of the package is `as.traitdata()` which moulds a species-trait-matrix or occurrence table into a measurement long-table format (Fig. 3). This function also maps column names into terms provided in the trait data standard and adds metadata as attributes to the output object. This example converts an own file 'data.csv' into a dataset of long-table structure that employs the standard vocabulary for core data:

```
library("traitdataform")
dataset <- as.traitdata(read.csv("path/to/data.csv"), 
             traits = c("body_length", "antenna_length", 
                        "metafemur_length"),
             units = "mm",
             taxa = "name_correct", 
             keep = c(locationID = "location")
             )

```

\begin{figure}[h]
\centering
\includegraphics[width=1\textwidth, angle=0]{figures/fig3.png}
\caption{Process chart of the functions provided within the R package 'traitdataform' to apply the standard vocabulary to any trait-data table.}

\end{figure}

The parameter 'traits' lists column names that contain trait values. The column containing taxon names is given in parameter 'taxa'. Note that the parameter 'keep' specifies and renames any data that should be maintained in the output. The parameter 'units' is used to specify the input units of measurement. 
In order to map user-provided names to unambiguous and globally unique identifiers, the function `standardize.taxonomy()` matches scientific taxon names automatically to the GBIF Backbone Taxonomy and adds the column `taxonID` to the core data  (Fig. 3). 
<!-- BKR: In the example below, this looks like a manual matching. What is done automatically here and which information does the user have to provide?  FDS: taxon is matched automatically, traits not. lack of trait APIs and webservices. --> 
<!-- Similarly, the `standardize.traits()` function matches user-provided trait names onto a given trait vocabulary and links it to an existing trait ontology or a user-provided table of trait definitions, i.e. a thesaurus of traits. The same function harmonises trait values by converting values into target units and unified factor levels. 
<!-- BKR: Have you thought about linking to unit ontologies here?
AO: This paper is a good reference and gives an idea of which ontology could be used;
http://www.semantic-web-journal.net/content/comparison-and-evaluation-ontologies-units-measurement
FDS: might be referred to in the section about thesauri -->

The following example harmonises traits based on a minimal thesaurus, referencing trait names with globally valid URIs defined by the BETSI thesaurus of soil invertebrate traits:

```
traitlist <- as.thesaurus(
   body_length = as.trait("body_length", 
     expectedUnit = "mm", valueType = "numeric", 
     identifier = "http://t-sita.cesab.org/BETSI_vizInfo.jsp?trait=Body_length"), 
   antenna_length = as.trait("antenna_length", 
     expectedUnit = "mm", valueType = "numeric", 
     identifier = "http://t-sita.cesab.org/BETSI_vizInfo.jsp?trait=Antenna_length"),
   metafemur_length = as.trait("metafemur_length", 
     expectedUnit = "mm", valueType = "numeric", 
     identifier = "http://t-sita.cesab.org/BETSI_vizInfo.jsp?trait=Femur_length")
)

datasetStd <- standardize.traits(dataset, thesaurus = traitlist)
```

The function `as.thesaurus()` provides a structured object that is required by the function `standardize.traits()`  (Fig. 3). Other ways of defining a thesaurus object are documented in the package vignette and function documentation (`?as.thesaurus`). Future iterations of the R package will aim at automatizing the generation of trait thesaurus objects from globally available Ontologies. 
The package functions form a tool-chain where each function can be piped as an input into the next. A wrapper function `standardize()` applies all functions sequentially, making transferring and harmonizing trait data as simple as:

```
datasetStd <- standardize(read.csv("path/to/data.csv"),
            thesaurus = traitlist,
            taxa = "name_correct", 
            units = "mm"
            )
```

Datasets that have been produced by these functions can easily be appended using the function `rbind()` of R base, while maintaining any available metadata information as separate column entries. 
To merge datasets with additional information on the occurrence or measurement level, secondary data tables can be added as columns of the core dataset according to a unique identifier using the function `merge()`. This enables an easy handling of data sources that originate in a relational database format.

Since the intention of the package is also to simplify the harmonisation of published trait data, the package offers direct access to trait datasets that have been released in the Public Domain or under Creative Commons licenses. <!--See a full list of linked datasets on the package  website or by calling `data(package = "traitdataform")`.--> We invite users and authors of datasets to add further data to the package and thereby contribute to this registry for distributed trait datasets. 

# Conclusion

<!--The practical importance of these frameworks for data providers and data users is often not obvious and not appropriately considered when storing or uploading data to low-threshold  file-publishing services. 

The investment is only paying out in the long run. Citations of trait datasets are higher when they are easier to access and relate to other data due to standard structures and vocabularies. 
Along with the push towards open access publication of primary data, the scientific community must come up with ways to incentivize and require data standardization. 

More compatibility and availability of trait thesauri could greatly improve the possibilities for a comparison of trait data from different sources.-->
To serve the demand for simple ways to standardise and harmonise ecological trait data, we propose a versatile vocabulary for simple, two-dimensional datasets as well as for the exchange and handling of trait data in the context of a 'semantic web'.
With the R-package 'traitdataform', we also present a toolbox in R to transfer and harmonise data into this scheme. 

It appears to be broad consensus that an open biodiversity science is crucial for an evidence-based decision making and conservation policy on regional and global scales.
In times of increasing demand for open research data and international platforms for biodiversity data management, the development of meaningful terminologies for the standardisation of biodiversity data is more than essential: defined ontologies enable researchers to relate published datasets to each other to achieve a greater synthesis, thereby paving the way for a better mechanistic understanding of the relationship between drivers, communities and functions and providing new insights on global biodiversity patterns. 
Moreover it might be also a step towards a more predictive ecology as a broader set of available traits might enable more hypothesis based trait-based approaches. 
In terms of data science, machine-readable, ontology-based data ease the application of big-data mining and machine-learning techniques.

To date, a rich distributed body of independently published trait datasets focus on particular organism groups, ecosystem types or regions. 
However, these distributed data are heterogeneous in form and description and initiatives to harmonise and compile these data require significant amounts of funding and personel. 
To support the long-term rewards of standardisation efforts, incentives should be sought to mitigate the cost of readying trait data for the 'semantic web' of biodiversity data and knowledge. This can be software tools or supporting infrastructures.
The tools proposed here help to standardise trait datasets before upload to central as well as distributed data repositories. 
By using a constrained vocabulary with globally accessible definitions of terms,  distributed trait data can be accessed more easily by other researchers and harmonised into aggregated datasets.
Also, it will ease the exchange of data between databases and facilitate the development computational methods and software tools that access and handle the data, based on the standard vocabulary. 
We also encourage the advancement of trait thesauri into more interrelated and complete ontologies. 
The biggest challenge in community efforts of standardisation of traits may be the investment in consensus building which leads to an acceptance and establishment of the methodological and conceptual definitions of traits.
This requires significant effort, but it returns  great scientific benefit by enabling synthesis on our general understanding of biodiversity and ecosystem function. 

# Acknowledgements

Thanks to all respondents to an internal online survey on trait data for the Biodiversity Exploratories project and to Matthias Biber, Kristin Bohn, Diana Bowler, Klaus Birkhofer, Runa Boeddinghaus, Catrin Westphal, Markus Fischer and Jens Kattge for comments on the manuscript drafts, the trait data standard vocabulary and the pre-releases of the R-package. 

We thank the managers of the three Exploratories, Kirsten Reichel-Jung, Juliane Vogt, Miriam Teuscher, and all former managers for their work in maintaining the plot and project
infrastructure; Christiane Fischer and Jule Mangels for giving
support through the central office, andMarkus Fischer, Eduard Linsenmair,
Dominik Hessenmöller, Daniel Prati, Ingo Schöning, François Buscot,
Ernst-Detlef Schulze, Wolfgang W. Weisser and the late Elisabeth Kalko
for their role in setting up the Biodiversity Exploratories project. 
The work has been partly funded by the DFG Priority Program
1374 "Infrastructure-Biodiversity-Exploratories" (DFG-Refno.). <!--Field work
permits were issued by the responsible state environmental offices of
Baden-Württemberg, Thüringen, and Brandenburg (according to § 72
BbgNatSchG).-->

M.J. was supported by the German Research Foundation within the framework of the Jena Experiment (FOR 1451) and by the Swiss National Science Foundation.

# Authors' contributions

FDS, AO, CP, and NKS conceived the idea and developed the vocabulary for the trait data standard with significant contributions of MJ and GLP (forming the first tier of the author list); NKS authored the example list of traits; FDS developed the R package 'traitdataform'; CP and FDS curated the living spreadsheet. All contributing authors appear in alphabetical order in a second tier of the author list. AG and DF implemented the vocabulary in the GFBio terminology service. All authors contributed critically to the structure and content of the manuscript and gave final approval for publication.

# Online Resources

The online reference for the Ecological Trait-data Standard Vocabulary described in this paper is https://ecologicaltraitdata.github.io/ETS/, stable DOI representing all versions: 10.5281/zenodo.1041732.

Any future development of the vocabulary and the R-package is coordinated via https://github.com/EcologicalTraitData/. 

The development website for the R-package 'traitdataform' is https://github.com/EcologicalTraitData/traitdataform.

An example of a published trait thesaurus for the Biodiversity Exploratories (www.biodiversity-exploratories.de) can be found at https://ecologicaltraitdata.github.io/TraitDataList/.

The living spreadsheet on existing trait datasets and databases can be found at https://goo.gl/QxzfHy. 

# References 

\footnotesize